diff --git a/config/external/lgpllibs/moz.build b/config/external/lgpllibs/moz.build
index b906062d03f8a..25762cb909d4c 100644
--- a/config/external/lgpllibs/moz.build
+++ b/config/external/lgpllibs/moz.build
@@ -10,7 +10,3 @@
 
 GeckoSharedLibrary("lgpllibs")
 SHARED_LIBRARY_NAME = "lgpllibs"
-
-# For some reason, mingw builds export way too many symbols.
-if CONFIG["TARGET_OS"] == "WINNT" and CONFIG["CC_TYPE"] != "clang-cl":
-    SYMBOLS_FILE = "lgpllibs.symbols"
