diff --git a/dom/canvas/WebGLContextValidate.cpp b/dom/canvas/WebGLContextValidate.cpp
index fc13eb2ea5e95..dd1295d61f2a6 100644
--- a/dom/canvas/WebGLContextValidate.cpp
+++ b/dom/canvas/WebGLContextValidate.cpp
@@ -25,6 +25,10 @@
 #include "mozilla/StaticPrefs_webgl.h"
 #include "nsPrintfCString.h"
 
+#if defined(MOZ_WIDGET_COCOA)
+#  include "nsCocoaFeatures.h"
+#endif
+
 ////////////////////
 // Minimum value constants defined in GLES 2.0.25 $6.2 "State Tables":
 const uint32_t kMinMaxVertexAttribs = 8;            // Page 164
@@ -483,6 +487,16 @@ bool WebGLContext::InitAndValidateGL(FailureReason* const out_failReason) {
     gl->fEnable(LOCAL_GL_PROGRAM_POINT_SIZE);
   }
 
+#ifdef XP_MACOSX
+  if (gl->WorkAroundDriverBugs() && gl->Vendor() == gl::GLVendor::ATI &&
+      !nsCocoaFeatures::IsAtLeastVersion(10, 9)) {
+    // The Mac ATI driver, in all known OSX version up to and including
+    // 10.8, renders points sprites upside-down. (Apple bug 11778921)
+    gl->fPointParameterf(LOCAL_GL_POINT_SPRITE_COORD_ORIGIN,
+                         LOCAL_GL_LOWER_LEFT);
+  }
+#endif
+
   if (gl->IsSupported(gl::GLFeature::seamless_cube_map_opt_in)) {
     gl->fEnable(LOCAL_GL_TEXTURE_CUBE_MAP_SEAMLESS);
   }
