diff --git a/dom/media/platforms/moz.build b/dom/media/platforms/moz.build
index 77803e3379e66..f99006c044062 100644
--- a/dom/media/platforms/moz.build
+++ b/dom/media/platforms/moz.build
@@ -90,11 +90,16 @@ if CONFIG["MOZ_APPLEMEDIA"]:
     ]
     UNIFIED_SOURCES += [
         "apple/AppleATDecoder.cpp",
+        "apple/AppleCMLinker.cpp",
+        "apple/AppleCVLinker.cpp",
         "apple/AppleDecoderModule.cpp",
         "apple/AppleEncoderModule.cpp",
-        "apple/AppleUtils.cpp",
+        'apple/AppleUtils.cpp',
+        'apple/AppleVDADecoder.cpp',
+        'apple/AppleVDALinker.cpp',
         "apple/AppleVTDecoder.cpp",
         "apple/AppleVTEncoder.cpp",
+        "apple/AppleVTLinker.cpp",
     ]
     LOCAL_INCLUDES += [
         "/media/libyuv/libyuv/include",
@@ -103,11 +108,14 @@ if CONFIG["MOZ_APPLEMEDIA"]:
         "-framework AudioToolbox",
         "-framework CoreMedia",
         "-framework VideoToolbox",
+        "-framework VideoDecodeAcceleration",
         # For some unknown reason, the documented method of using weak_import
         # attribute doesn't work with VideoToolbox's functions.
         # We want to lazily load _VTRegisterSupplementalVideoDecoderIfAvailable
         # symbol as it's only available in macOS 11 and later.
         "-Wl,-U,_VTRegisterSupplementalVideoDecoderIfAvailable",
+        # Same for VTIsHardwareDecodeSupported available from macOS 10.13.
+        "-Wl,-U,_VTIsHardwareDecodeSupported",
     ]
 
 include("/ipc/chromium/chromium-config.mozbuild")
