diff --git a/gfx/layers/ipc/CompositorBridgeChild.cpp b/gfx/layers/ipc/CompositorBridgeChild.cpp
index 5753b34b927f5..7bc5661549f29 100644
--- a/gfx/layers/ipc/CompositorBridgeChild.cpp
+++ b/gfx/layers/ipc/CompositorBridgeChild.cpp
@@ -71,6 +71,7 @@ CompositorBridgeChild::CompositorBridgeChild(CompositorManagerChild* aManager)
       mIdNamespace(0),
       mResourceId(0),
       mCanSend(false),
+      mWindowOverlayChanged(false),
       mActorDestroyed(false),
       mPaused(false),
       mForceSyncFlushRendering(false),
@@ -534,6 +535,7 @@ already_AddRefed<CanvasChild> CompositorBridgeChild::GetCanvasChild() {
 
 void CompositorBridgeChild::EndCanvasTransaction() {
   if (auto* cm = gfx::CanvasManagerChild::Get()) {
+    mWindowOverlayChanged = false;
     cm->EndCanvasTransaction();
   }
 }
