diff --git a/toolkit/components/remote/nsMacRemoteServer.mm b/toolkit/components/remote/nsMacRemoteServer.mm
index 126c9d3863ce3..dc807d80b941d 100644
--- a/toolkit/components/remote/nsMacRemoteServer.mm
+++ b/toolkit/components/remote/nsMacRemoteServer.mm
@@ -36,7 +36,7 @@ CFDataRef messageServerCallback(CFMessagePortRef aLocal, int32_t aMsgid,
     NSDictionary* dict =
         [NSKeyedUnarchiver unarchiveObjectWithData:(NSData*)aData];
     if (dict && [dict isKindOfClass:[NSDictionary class]]) {
-      NSArray* args = dict[@"args"];
+      NSArray* args = [dict objectForKey:@"args"];
       if (!args) {
         NS_ERROR("Wrong parameters passed to the Remote Server");
         return;
@@ -68,7 +68,7 @@ CFDataRef messageServerCallback(CFMessagePortRef aLocal, int32_t aMsgid,
       // in the current instance.
       cmdLine->Run();
 
-      NSNumber* raise = dict[@"raise"];
+      NSNumber* raise = [dict objectForKey:@"raise"];
       if (!raise || [raise boolValue]) {
         // Activating the application brings the most recent window to the
         // foreground.
