diff --git a/widget/cocoa/nsChildView.h b/widget/cocoa/nsChildView.h
index 37fd9a9cb8d3a..b3089614d7e9e 100644
--- a/widget/cocoa/nsChildView.h
+++ b/widget/cocoa/nsChildView.h
@@ -71,6 +71,17 @@ class WidgetRenderingContext;
 @end
 
 @interface NSView (Undocumented)
+// Draws the title string of a window.
+// Present on NSThemeFrame since at least 10.6.
+// _drawTitleBar is somewhat complex, and has changed over the years
+// since OS X 10.6.  But in that time it's never done anything that
+// would break when called outside of -[NSView drawRect:] (which we
+// sometimes do), or whose output can't be redirected to a
+// CGContextRef object (which we also sometimes do).  This is likely
+// to remain true for the indefinite future.  However we should
+// check _drawTitleBar in each new major version of OS X.  For more
+// information see bug 877767.
+- (void)_drawTitleBar:(NSRect)aRect;
 
 // Undocumented method of one or more of NSFrameView's subclasses.  Called
 // when one or more of the titlebar buttons needs to be repositioned, to
@@ -104,7 +115,7 @@ class WidgetRenderingContext;
                            NSPasteboardItemDataProvider,
                            NSStandardKeyBindingResponding> {
  @private
-  // the nsCocoaWindow that created the view. It retains this NSView, so
+  // the nsChildView that created the view. It retains this NSView, so
   // the link back to it must be weak.
   nsCocoaWindow* mGeckoChild;
 
@@ -172,7 +183,10 @@ class WidgetRenderingContext;
 
   // Whether this uses off-main-thread compositing.
   BOOL mUsingOMTCompositor;
-
+  // The mask image that's used when painting into the titlebar using basic
+  // CGContext painting (i.e. non-accelerated).
+  // Always null if nsCocoaFeatures::OnMavericksOrLater() is true.
+  CGImageRef mTopLeftCornerMask;
   // Subviews of self, which act as container views for vibrancy views and
   // non-draggable views.
   NSView* mVibrancyViewsContainer;      // [STRONG]
@@ -245,6 +259,8 @@ class WidgetRenderingContext;
 
 - (void)scrollWheel:(NSEvent*)anEvent;
 
+- (void)setUsingOMTCompositor:(BOOL)aUseOMTC;
+
 - (NSEvent*)lastKeyDownEvent;
 
 + (uint32_t)sUniqueKeyEventId;
@@ -274,5 +290,4 @@ class ChildViewMouseTracker {
   static NSWindow* sWindowUnderMouse;
   static NSPoint sLastScrollEventScreenLocation;
 };
-
 #endif  // nsChildView_h_
