diff --git a/widget/cocoa/nsClipboard.mm b/widget/cocoa/nsClipboard.mm
index fd4cc807ae712..7478ba1143abd 100644
--- a/widget/cocoa/nsClipboard.mm
+++ b/widget/cocoa/nsClipboard.mm
@@ -64,7 +64,10 @@
     case nsIClipboard::kGlobalClipboard:
       return [NSPasteboard generalPasteboard];
     case nsIClipboard::kFindClipboard:
-      return [NSPasteboard pasteboardWithName:NSPasteboardNameFind];
+      if (@available(macOS 10.13, *)) {
+          return [NSPasteboard pasteboardWithName:NSPasteboardNameFind];
+      }
+      return [NSPasteboard pasteboardWithName:NSFindPboard];
     default:
       return nil;
   }
