diff --git a/xpcom/base/MacStringHelpers.mm b/xpcom/base/MacStringHelpers.mm
index 6d83429987098..3ee192ec67a6c 100644
--- a/xpcom/base/MacStringHelpers.mm
+++ b/xpcom/base/MacStringHelpers.mm
@@ -18,8 +18,8 @@ void CopyNSStringToXPCOMString(const NSString* aFrom, nsAString& aTo) {
   }
 
   NSUInteger len = [aFrom length];
-  if (len > std::numeric_limits<nsAString::size_type>::max()) {
-    aTo.AllocFailed(std::numeric_limits<nsAString::size_type>::max());
+  if (len > NSUIntegerMax) {
+    aTo.AllocFailed(NSUIntegerMax);
   }
 
   aTo.SetLength(len);
