diff --git a/third_party/rust/metal/src/pipeline/render.rs b/third_party/rust/metal/src/pipeline/render.rs
index b55be15c2ddbb..4e8f23dc5ef05 100644
--- a/third_party/rust/metal/src/pipeline/render.rs
+++ b/third_party/rust/metal/src/pipeline/render.rs
@@ -218,7 +218,7 @@ foreign_obj_type! {
 
 impl ArgumentArrayRef {
     pub fn object_at(&self, index: NSUInteger) -> Option<&ArgumentRef> {
-        unsafe { msg_send![self, objectAtIndexedSubscript: index] }
+        unsafe { msg_send![self, objectAtIndex: index] }
     }
 
     pub fn count(&self) -> NSUInteger {
@@ -745,7 +745,7 @@ impl RenderPipelineColorAttachmentDescriptorArrayRef {
         &self,
         index: NSUInteger,
     ) -> Option<&RenderPipelineColorAttachmentDescriptorRef> {
-        unsafe { msg_send![self, objectAtIndexedSubscript: index] }
+        unsafe { msg_send![self, objectAtIndex: index] }
     }
 
     pub fn set_object_at(
@@ -755,7 +755,7 @@ impl RenderPipelineColorAttachmentDescriptorArrayRef {
     ) {
         unsafe {
             msg_send![self, setObject:attachment
-                   atIndexedSubscript:index]
+                   atIndex:index]
         }
     }
 }
