diff --git a/accessible/mac/mozAccessible.mm b/accessible/mac/mozAccessible.mm
index 6f88631034..1378fd513c 100644
--- a/accessible/mac/mozAccessible.mm
+++ b/accessible/mac/mozAccessible.mm
@@ -1034,7 +1034,7 @@ static bool ProvidesTitle(const Accessible* aAccessible, nsString& aName) {
           static_cast<MOXTextMarkerDelegate*>([self moxTextMarkerDelegate]);
       NSMutableDictionary* userInfo =
           [[[delegate selectionChangeInfo] mutableCopy] autorelease];
-      userInfo[@"AXTextChangeElement"] = self;
+      [userInfo setObject:self forKey:@"AXTextChangeElement"];
 
       mozAccessible* webArea = [self topWebArea];
       [webArea
