diff --git a/config/rules.mk b/config/rules.mk
index 8551d9d2aa..8897eb385e 100644
--- a/config/rules.mk
+++ b/config/rules.mk
@@ -149,14 +149,16 @@ MOZ_PROGRAM_LDFLAGS += -Wl,-rpath -Wl,@executable_path/Frameworks
 endif
 endif
 
-# For Mac executables, set the @rpath to be @executable_path by default so that
-# shared libraries built with an @rpath install name in the same directory
-# as the executable can be resolved. Executables not in the same directory
-# should override the @rpath with a relative path such as @executable_path/../
-# depending on their install location.
-ifeq ($(OS_ARCH),Darwin)
-MOZ_PROGRAM_LDFLAGS += -Wl,-rpath,@executable_path
-endif
+## this is part of commit https://hg.mozilla.org/mozilla-unified/rev/1bc4ee894015268a6be66950b80705e73f17147e
+# which is being removed for backwards compatibility purcpoases
+## For Mac executables, set the @rpath to be @executable_path by default so that
+## shared libraries built with an @rpath install name in the same directory
+## as the executable can be resolved. Executables not in the same directory
+## should override the @rpath with a relative path such as @executable_path/../
+## depending on their install location.
+#ifeq ($(OS_ARCH),Darwin)
+#MOZ_PROGRAM_LDFLAGS += -Wl,-rpath,@executable_path
+#endif
 
 ifeq ($(OS_ARCH),WINNT)
 ifeq ($(CC_TYPE),clang)
@@ -274,9 +276,12 @@ endif
 # MacOS X specific stuff
 #
 
+# see
+# https://hg.mozilla.org/mozilla-unified/rev/1bc4ee894015268a6be66950b80705e73f17147e
+# for why we are setting loader path back to executable path
 ifeq ($(OS_ARCH),Darwin)
 ifneq (,$(SHARED_LIBRARY))
-_LOADER_PATH := @rpath
+_LOADER_PATH := @executable_path
 EXTRA_DSO_LDOPTS	+= -dynamiclib -install_name $(_LOADER_PATH)/$(@F) -compatibility_version 1 -current_version 1
 endif
 endif
