diff --git a/dom/media/systemservices/objc_video_capture/device_info_avfoundation.mm b/dom/media/systemservices/objc_video_capture/device_info_avfoundation.mm
index c477f95c1e..06ca7d626e 100644
--- a/dom/media/systemservices/objc_video_capture/device_info_avfoundation.mm
+++ b/dom/media/systemservices/objc_video_capture/device_info_avfoundation.mm
@@ -191,9 +191,10 @@ void DeviceInfoAvFoundation::EnsureCapabilitiesMap() {
     return;
   }
 
-  for (AVCaptureDevice* device in [RTCCameraVideoCapturer
-           captureDevicesWithDeviceTypes:[RTCCameraVideoCapturer
-                                             defaultCaptureDeviceTypes]]) {
+  for (AVCaptureDevice* device in [RTCCameraVideoCapturer captureDevices]
+        /* //again, fuck you guys   
+        captureDevicesWithDeviceTypes:[RTCCameraVideoCapturer
+                                             defaultCaptureDeviceTypes]]*/) {
     std::string uniqueId = [NSString stdStringForString:device.uniqueID];
     std::string name = [NSString stdStringForString:device.localizedName];
     auto& [_, __, capabilities] = mDevicesAndCapabilities.emplace_back(
