diff --git a/dom/system/mac/nsOSPermissionRequest.mm b/dom/system/mac/nsOSPermissionRequest.mm
index 82e9edf1ac..4c17c980e4 100644
--- a/dom/system/mac/nsOSPermissionRequest.mm
+++ b/dom/system/mac/nsOSPermissionRequest.mm
@@ -5,7 +5,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "nsOSPermissionRequest.h"
-
+#include "nsCocoaFeatures.h"
 #include "mozilla/dom/Promise.h"
 #include "nsCocoaUtils.h"
 
@@ -16,24 +16,41 @@ using mozilla::dom::Promise;
 NS_IMETHODIMP
 nsOSPermissionRequest::GetAudioCapturePermissionState(uint16_t* aAudio) {
   MOZ_ASSERT(aAudio);
-  return nsCocoaUtils::GetAudioCapturePermissionState(*aAudio);
+    if (!nsCocoaFeatures::OnMojaveOrLater()) {
+    return nsOSPermissionRequestBase::GetAudioCapturePermissionState(aAudio);
+    }
+    return nsCocoaUtils::GetAudioCapturePermissionState(*aAudio);
 }
 
 NS_IMETHODIMP
 nsOSPermissionRequest::GetVideoCapturePermissionState(uint16_t* aVideo) {
   MOZ_ASSERT(aVideo);
+ 
+  if (!nsCocoaFeatures::OnMojaveOrLater()) {
+    return nsOSPermissionRequestBase::GetVideoCapturePermissionState(aVideo);
+  }
+
+
   return nsCocoaUtils::GetVideoCapturePermissionState(*aVideo);
 }
 
 NS_IMETHODIMP
 nsOSPermissionRequest::GetScreenCapturePermissionState(uint16_t* aScreen) {
   MOZ_ASSERT(aScreen);
+  if (!nsCocoaFeatures::OnMojaveOrLater()) {
+    return nsOSPermissionRequestBase::GetScreenCapturePermissionState(aScreen);
+ }
+
   return nsCocoaUtils::GetScreenCapturePermissionState(*aScreen);
 }
 
 NS_IMETHODIMP
 nsOSPermissionRequest::RequestVideoCapturePermission(JSContext* aCx,
                                                      Promise** aPromiseOut) {
+  if (!nsCocoaFeatures::OnMojaveOrLater()) {
+    return nsOSPermissionRequestBase::RequestVideoCapturePermission(aCx, aPromiseOut);
+  }
+
   RefPtr<Promise> promiseHandle;
   nsresult rv = GetPromise(aCx, promiseHandle);
   if (NS_FAILED(rv)) {
@@ -48,6 +65,10 @@ nsOSPermissionRequest::RequestVideoCapturePermission(JSContext* aCx,
 NS_IMETHODIMP
 nsOSPermissionRequest::RequestAudioCapturePermission(JSContext* aCx,
                                                      Promise** aPromiseOut) {
+
+  if (!nsCocoaFeatures::OnMojaveOrLater()) {
+    return nsOSPermissionRequestBase::RequestAudioCapturePermission(aCx, aPromiseOut);
+  }  
   RefPtr<Promise> promiseHandle;
   nsresult rv = GetPromise(aCx, promiseHandle);
   if (NS_FAILED(rv)) {
@@ -61,5 +82,8 @@ nsOSPermissionRequest::RequestAudioCapturePermission(JSContext* aCx,
 
 NS_IMETHODIMP
 nsOSPermissionRequest::MaybeRequestScreenCapturePermission() {
+  if (!nsCocoaFeatures::OnCatalinaOrLater()) {
+    return nsOSPermissionRequestBase::MaybeRequestScreenCapturePermission();
+  }
   return nsCocoaUtils::MaybeRequestScreenCapturePermission();
 }
