diff --git a/gfx/layers/MacIOSurfaceImage.cpp b/gfx/layers/MacIOSurfaceImage.cpp
index a6f1da18b4..b1aa666d80 100644
--- a/gfx/layers/MacIOSurfaceImage.cpp
+++ b/gfx/layers/MacIOSurfaceImage.cpp
@@ -16,6 +16,10 @@
 #include "mozilla/Unused.h"
 #include "YCbCrUtils.h"
 
+#ifdef XP_MACOSX
+#  include "nsCocoaFeatures.h"
+#endif
+
 using namespace mozilla::layers;
 using namespace mozilla::gfx;
 
@@ -307,7 +311,6 @@ already_AddRefed<MacIOSurface> MacIOSurfaceRecycleAllocator::Allocate(
                  (size_t)aCbCrSize.height);
     }
 #endif
-
     return MakeAndAddRef<MacIOSurface>(surf, false, aYUVColorSpace);
   }
 
@@ -320,9 +323,12 @@ already_AddRefed<MacIOSurface> MacIOSurfaceRecycleAllocator::Allocate(
   // planar format.
   // 4:2:2 formats with 8 bit color are single planar, otherwise bi-planar.
 
+  //Lion doesn't do 10-bit colour, so we have to always create a single planar
+  //surface for 10.7 and lower lol.
   RefPtr<MacIOSurface> result;
-  if (aChromaSubsampling == gfx::ChromaSubsampling::HALF_WIDTH &&
-      aColorDepth == gfx::ColorDepth::COLOR_8) {
+  if ((aChromaSubsampling == gfx::ChromaSubsampling::HALF_WIDTH &&
+      aColorDepth == gfx::ColorDepth::COLOR_8) || 
+      !nsCocoaFeatures::OnMountainLionOrLater()) {
     result = MacIOSurface::CreateSinglePlanarSurface(
         aYSize, aYUVColorSpace, aTransferFunction, aColorRange);
   } else {
