diff --git a/gfx/layers/ipc/CompositorBridgeChild.h b/gfx/layers/ipc/CompositorBridgeChild.h
index abc3df6fce..1b2b6d0eee 100644
--- a/gfx/layers/ipc/CompositorBridgeChild.h
+++ b/gfx/layers/ipc/CompositorBridgeChild.h
@@ -147,6 +147,7 @@ class CompositorBridgeChild final : public PCompositorBridgeChild,
   static void ShutDown();
 
   FwdTransactionCounter& GetFwdTransactionCounter();
+  void WindowOverlayChanged() { mWindowOverlayChanged = true; }
 
   /**
    * Hold TextureClient ref until end of usage on host side if
@@ -230,6 +231,7 @@ class CompositorBridgeChild final : public PCompositorBridgeChild,
   // True until the beginning of the two-step shutdown sequence of this actor.
   bool mCanSend;
 
+  bool mWindowOverlayChanged;
   // False until the actor is destroyed.
   bool mActorDestroyed;
 
