diff --git a/image/decoders/icon/mac/nsIconChannelCocoa.mm b/image/decoders/icon/mac/nsIconChannelCocoa.mm
index f7ebc79e6e..fc755ba8a6 100644
--- a/image/decoders/icon/mac/nsIconChannelCocoa.mm
+++ b/image/decoders/icon/mac/nsIconChannelCocoa.mm
@@ -348,11 +348,12 @@ nsresult nsIconChannel::MakeInputStream(nsIInputStream** _retval,
 
   NSGraphicsContext* oldContext = [NSGraphicsContext currentContext];
   [NSGraphicsContext
-      setCurrentContext:[NSGraphicsContext graphicsContextWithCGContext:ctx
+      setCurrentContext:[NSGraphicsContext graphicsContextWithGraphicsPort:ctx
                                                                 flipped:NO]];
 
-  [iconImage drawInRect:NSMakeRect(0, 0, width, height)];
-
+  if(@available(macOS 10.9, *)) {
+     [iconImage drawInRect:NSMakeRect(0, 0, width, height)];
+  }
   [NSGraphicsContext setCurrentContext:oldContext];
 
   CGContextRelease(ctx);
