diff --git a/layout/base/nsDocumentViewer.cpp b/layout/base/nsDocumentViewer.cpp
index ed8b288f55..e7bd03a7b1 100644
--- a/layout/base/nsDocumentViewer.cpp
+++ b/layout/base/nsDocumentViewer.cpp
@@ -3265,10 +3265,16 @@ bool nsDocumentViewer::ShouldAttachToTopLevel() {
   if (nsIWidget::UsePuppetWidgets() || mParentWidget->IsPuppetWidget()) {
     return true;
   }
+
+  // TODO(emilio, bug 1919165): Unify this between macOS and other platforms?
+#ifdef XP_MACOSX
+  return false;
+#else
 #ifdef DEBUG
   nsIWidgetListener* parentListener = mParentWidget->GetWidgetListener();
   MOZ_ASSERT(!parentListener || !parentListener->GetView(),
              "Expect a top level widget");
+#endif
 #endif
   return true;
 }
