diff --git a/layout/base/nsLayoutUtils.cpp b/layout/base/nsLayoutUtils.cpp
index ee4e4b7e04..eefe4a8ceb 100644
--- a/layout/base/nsLayoutUtils.cpp
+++ b/layout/base/nsLayoutUtils.cpp
@@ -2397,10 +2397,34 @@ nsRect nsLayoutUtils::TransformFrameRectToAncestor(
       NSFloatPixelsToAppUnits(float(result.height), destAppUnitsPerDevPixel));
 }
 
+static LayoutDeviceIntPoint GetWidgetOffset(nsIWidget* aWidget,
+                                            nsIWidget*& aRootWidget) {
+  LayoutDeviceIntPoint offset(0, 0);
+  while (aWidget->GetWindowType() == widget::WindowType::Child) {
+    nsIWidget* parent = aWidget->GetParent();
+    if (!parent) {
+      break;
+    }
+    LayoutDeviceIntRect bounds = aWidget->GetBounds();
+    offset += bounds.TopLeft();
+    aWidget = parent;
+  }
+  aRootWidget = aWidget;
+  return offset;
+}
+
+
 LayoutDeviceIntPoint nsLayoutUtils::WidgetToWidgetOffset(nsIWidget* aFrom,
                                                          nsIWidget* aTo) {
-  auto fromOffset = aFrom->WidgetToScreenOffset();
-  auto toOffset = aTo->WidgetToScreenOffset();
+  nsIWidget* fromRoot;
+  LayoutDeviceIntPoint fromOffset = GetWidgetOffset(aFrom, fromRoot);
+  nsIWidget* toRoot;
+  LayoutDeviceIntPoint toOffset = GetWidgetOffset(aTo, toRoot);
+
+  if (fromRoot != toRoot) {
+    fromOffset = aFrom->WidgetToScreenOffset();
+    toOffset = aTo->WidgetToScreenOffset();
+  }
   return fromOffset - toOffset;
 }
 
