diff --git a/media/libsoundtouch/src/SoundTouchFactory.cpp b/media/libsoundtouch/src/SoundTouchFactory.cpp
index 1da3b88f36..7c2831c43b 100644
--- a/media/libsoundtouch/src/SoundTouchFactory.cpp
+++ b/media/libsoundtouch/src/SoundTouchFactory.cpp
@@ -9,12 +9,14 @@
 extern "C" namespace soundtouch
 {
 
+SOUNDTOUCH_API
 soundtouch::SoundTouch*
 createSoundTouchObj()
 {
   return new soundtouch::SoundTouch();
 }
 
+SOUNDTOUCH_API
 void
 destroySoundTouchObj(soundtouch::SoundTouch* aObj)
 {
