diff --git a/security/manager/ssl/osclientcerts/Cargo.toml b/security/manager/ssl/osclientcerts/Cargo.toml
index b634715bf5..3c12981f41 100644
--- a/security/manager/ssl/osclientcerts/Cargo.toml
+++ b/security/manager/ssl/osclientcerts/Cargo.toml
@@ -20,6 +20,8 @@ rsclientcerts = { path = "../rsclientcerts" }
 sha2 = "0.10.2"
 xpcom = { path = "../../../../xpcom/rust/xpcom" }
 
+whatsys = "0.3"
+
 [target."cfg(any(target_os = \"macos\", target_os = \"ios\"))".dependencies.core-foundation]
 version = "0.9"
 
