diff --git a/third_party/rust/cubeb-coreaudio/src/backend/tests/interfaces.rs b/third_party/rust/cubeb-coreaudio/src/backend/tests/interfaces.rs
index 3da50cc262..d91a257166 100644
--- a/third_party/rust/cubeb-coreaudio/src/backend/tests/interfaces.rs
+++ b/third_party/rust/cubeb-coreaudio/src/backend/tests/interfaces.rs
@@ -1308,7 +1308,8 @@ fn test_ops_input_voice_stream_init_and_destroy() {
         let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
         assert_eq!(
             stm.core_stream_data.using_voice_processing_unit(),
-            macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+            (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
         );
     });
 }
@@ -1320,7 +1321,8 @@ fn test_ops_input_voice_stream_start() {
         let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
         assert_eq!(
             stm.core_stream_data.using_voice_processing_unit(),
-            macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+            (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
         );
     });
 }
@@ -1332,7 +1334,8 @@ fn test_ops_input_voice_stream_stop() {
         let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
         assert_eq!(
             stm.core_stream_data.using_voice_processing_unit(),
-            macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+            (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
         );
     });
 }
@@ -1343,7 +1346,8 @@ fn test_ops_duplex_voice_stream_init_and_destroy() {
         let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
         assert_eq!(
             stm.core_stream_data.using_voice_processing_unit(),
-            macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+            (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
         );
     });
 }
@@ -1355,7 +1359,8 @@ fn test_ops_duplex_voice_stream_start() {
         let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
         assert_eq!(
             stm.core_stream_data.using_voice_processing_unit(),
-            macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+            (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
         );
     });
 }
@@ -1367,7 +1372,8 @@ fn test_ops_duplex_voice_stream_stop() {
         let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
         assert_eq!(
             stm.core_stream_data.using_voice_processing_unit(),
-            macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+            (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
         );
     });
 }
@@ -1382,7 +1388,8 @@ fn test_ops_duplex_voice_stream_drain() {
             let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
             assert_eq!(
                 stm.core_stream_data.using_voice_processing_unit(),
-                macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+                (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
             );
             thread::sleep(Duration::from_millis(10));
         },
@@ -1392,7 +1399,8 @@ fn test_ops_duplex_voice_stream_drain() {
 #[test]
 #[ignore]
 fn test_ops_timing_sensitive_multiple_voice_stream_init_and_destroy() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1545,7 +1553,8 @@ fn test_ops_timing_sensitive_multiple_voice_stream_init_and_destroy() {
 #[test]
 #[ignore]
 fn test_ops_timing_sensitive_multiple_duplex_voice_stream_start() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1586,7 +1595,8 @@ fn test_ops_timing_sensitive_multiple_duplex_voice_stream_start() {
 #[test]
 #[ignore]
 fn test_ops_timing_sensitive_multiple_duplex_voice_stream_params() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1673,7 +1683,8 @@ fn test_ops_timing_sensitive_multiple_duplex_voice_stream_params() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_mute() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1689,7 +1700,8 @@ fn test_ops_duplex_voice_stream_set_input_mute() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_mute_before_start() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1709,7 +1721,8 @@ fn test_ops_duplex_voice_stream_set_input_mute_before_start() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_mute_before_start_with_reinit() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1748,7 +1761,8 @@ fn test_ops_duplex_voice_stream_set_input_mute_before_start_with_reinit() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_mute_after_start() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1765,7 +1779,8 @@ fn test_ops_duplex_voice_stream_set_input_mute_after_start() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_processing_params() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1785,7 +1800,8 @@ fn test_ops_duplex_voice_stream_set_input_processing_params() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_processing_params_before_start() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1809,7 +1825,8 @@ fn test_ops_duplex_voice_stream_set_input_processing_params_before_start() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_processing_params_before_start_with_reinit() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1873,7 +1890,8 @@ fn test_ops_duplex_voice_stream_set_input_processing_params_before_start_with_re
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_processing_params_after_start() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1903,7 +1921,8 @@ fn test_ops_stereo_input_duplex_voice_stream_init_and_destroy() {
             let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
             assert_eq!(
                 stm.core_stream_data.using_voice_processing_unit(),
-                macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+                (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
             );
         },
     );
@@ -1917,7 +1936,8 @@ fn test_ops_stereo_input_duplex_voice_stream_start() {
             let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
             assert_eq!(
                 stm.core_stream_data.using_voice_processing_unit(),
-                macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+                (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
             );
             assert_eq!(unsafe { OPS.stream_start.unwrap()(stream) }, ffi::CUBEB_OK);
         },
@@ -1932,7 +1952,8 @@ fn test_ops_stereo_input_duplex_voice_stream_stop() {
             let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
             assert_eq!(
                 stm.core_stream_data.using_voice_processing_unit(),
-                macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+                (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
             );
             assert_eq!(unsafe { OPS.stream_stop.unwrap()(stream) }, ffi::CUBEB_OK);
         },
