diff --git a/widget/cocoa/MediaHardwareKeysEventSourceMacMediaCenter.mm b/widget/cocoa/MediaHardwareKeysEventSourceMacMediaCenter.mm
index 614e94a049..224f4140c2 100644
--- a/widget/cocoa/MediaHardwareKeysEventSourceMacMediaCenter.mm
+++ b/widget/cocoa/MediaHardwareKeysEventSourceMacMediaCenter.mm
@@ -207,12 +207,13 @@
                      forKey:MPMediaItemPropertyAlbumTitle];
   if (mCurrentImageUrl.IsEmpty() ||
       !IsImageIn(aMetadata.mArtwork, mCurrentImageUrl)) {
-    [nowPlayingInfo removeObjectForKey:MPMediaItemPropertyArtwork];
-
+    if(@available(macos 10.13.2, *)) {
+      [nowPlayingInfo removeObjectForKey:MPMediaItemPropertyArtwork];
+    }
     if (mFetchingUrl.IsEmpty() ||
         !IsImageIn(aMetadata.mArtwork, mFetchingUrl)) {
-      mNextImageIndex = 0;
-      LoadImageAtIndex(mNextImageIndex++);
+        mNextImageIndex = 0;
+        LoadImageAtIndex(mNextImageIndex++);
     }
   }
 
@@ -318,11 +319,10 @@
                     requestHandler:^NSImage* _Nonnull(CGSize aSize) {
                       return image;
                     }];
-            [nowPlayingInfo setObject:artwork
-                               forKey:MPMediaItemPropertyArtwork];
-            [artwork release];
-            [image release];
-
+            if(@available(macOS 10.13.2, *)) {
+              [nowPlayingInfo setObject:artwork
+                                 forKey:MPMediaItemPropertyArtwork];
+            }
             center.nowPlayingInfo = nowPlayingInfo;
 
             mFetchingUrl.Truncate();
