diff --git a/widget/cocoa/TextInputHandler.mm b/widget/cocoa/TextInputHandler.mm
index 936623d587..71478fd057 100644
--- a/widget/cocoa/TextInputHandler.mm
+++ b/widget/cocoa/TextInputHandler.mm
@@ -22,7 +22,8 @@
 #include "mozilla/TextEvents.h"
 #include "mozilla/ToString.h"
 
-#include "nsCocoaWindow.h"
+#include "nsChildView.h"
+#include "nsCocoaFeatures.h"
 #include "nsObjCExceptions.h"
 #include "nsBidiUtils.h"
 #include "nsToolkit.h"
@@ -1791,7 +1792,7 @@ inline NSRange MakeNSRangeFrom(
  *
  ******************************************************************************/
 
-TextInputHandler::TextInputHandler(nsCocoaWindow* aWidget,
+TextInputHandler::TextInputHandler(nsChildView* aWidget,
                                    NSView<mozView>* aNativeView)
     : IMEInputHandler(aWidget, aNativeView) {
   EnsureToLogAllKeyboardLayoutsAndIMEs();
@@ -1834,7 +1835,7 @@ inline NSRange MakeNSRangeFrom(
     [NSCursor setHiddenUntilMouseMoves:YES];
   }
 
-  RefPtr<nsCocoaWindow> widget(mWidget);
+  RefPtr<nsChildView> widget(mWidget);
 
   KeyEventState* currentKeyEvent = PushKeyEvent(aNativeEvent, aUniqueId);
   AutoKeyEventStateCleaner remover(this);
@@ -2049,7 +2050,7 @@ inline NSRange MakeNSRangeFrom(
     return;
   }
 
-  RefPtr<nsCocoaWindow> kungFuDeathGrip(mWidget);
+  RefPtr<nsChildView> kungFuDeathGrip(mWidget);
   mozilla::Unused << kungFuDeathGrip;  // Not referenced within this function
 
   MOZ_LOG_KEY_OR_IME(
@@ -2319,7 +2320,7 @@ inline NSRange MakeNSRangeFrom(
                               modifierFlags:modifiers
                                   timestamp:[aNativeEvent timestamp]
                                windowNumber:[aNativeEvent windowNumber]
-                                    context:nil
+                                    context:[aNativeEvent context]
                                  characters:@""
                 charactersIgnoringModifiers:@""
                                   isARepeat:NO
@@ -2565,7 +2566,7 @@ WidgetContentCommandEvent deleteCommandEvent(true, eContentCommandDelete,
   }
 
   // XXX Shouldn't we hold mDispatcher instead of mWidget?
-  RefPtr<nsCocoaWindow> widget(mWidget);
+  RefPtr<nsChildView> widget(mWidget);
   nsresult rv = mDispatcher->BeginNativeInputTransaction();
   if (NS_WARN_IF(NS_FAILED(rv))) {
     MOZ_LOG_KEY_OR_IME(LogLevel::Error,
@@ -2605,16 +2606,8 @@ WidgetContentCommandEvent deleteCommandEvent(true, eContentCommandDelete,
   // If mCurrentKeyEvent.mKeyEvent is null, the text should be inputted as
   // composition events.
   nsEventStatus status = nsEventStatus_eIgnore;
-  bool keyPressDispatched = [&]() {
-    // If text content is chrome process, OnTextChange etc will be dispatched
-    // synchronously. We don't want to dismiss text substitution panel at this
-    // point.
-    AutoRestore<bool> block(mBlockDismissTextSubstitutionPanel);
-    mBlockDismissTextSubstitutionPanel = true;
-
-    return mDispatcher->MaybeDispatchKeypressEvents(keypressEvent, status,
-                                                    currentKeyEvent);
-  }();
+  bool keyPressDispatched = mDispatcher->MaybeDispatchKeypressEvents(
+      keypressEvent, status, currentKeyEvent);
   bool keyPressHandled = (status == nsEventStatus_eConsumeNoDefault);
 
   // WebKit and text editor dismisses autocorrect panel by space, then process
@@ -2747,7 +2740,7 @@ WidgetContentCommandEvent deleteCommandEvent(true, eContentCommandDelete,
     }
   }
 
-  RefPtr<nsCocoaWindow> widget(mWidget);
+  RefPtr<nsChildView> widget(mWidget);
   nsresult rv = mDispatcher->BeginNativeInputTransaction();
   if (NS_WARN_IF(NS_FAILED(rv))) {
     MOZ_LOG_KEY_OR_IME(LogLevel::Error,
@@ -3056,7 +3049,7 @@ WidgetContentCommandEvent deleteCommandEvent(true, eContentCommandDelete,
 }
 
 bool TextInputHandler::DoCommandBySelector(const char* aSelector) {
-  RefPtr<nsCocoaWindow> widget(mWidget);
+  RefPtr<nsChildView> widget(mWidget);
 
   KeyEventState* currentKeyEvent = GetCurrentKeyEvent();
 
@@ -3117,17 +3110,9 @@ WidgetContentCommandEvent deleteCommandEvent(true, eContentCommandDelete,
     currentKeyEvent->InitKeyEvent(this, keypressEvent, false);
 
     nsEventStatus status = nsEventStatus_eIgnore;
-    // If text content is chrome process, OnTextChange etc will be dispatched
-    // synchronously. We don't want to dismiss text substitution panel at this
-    // point.
-    {
-      AutoRestore<bool> block(mBlockDismissTextSubstitutionPanel);
-      mBlockDismissTextSubstitutionPanel = true;
-
-      currentKeyEvent->mKeyPressDispatched =
-          mDispatcher->MaybeDispatchKeypressEvents(keypressEvent, status,
-                                                   currentKeyEvent);
-    }
+    currentKeyEvent->mKeyPressDispatched =
+        mDispatcher->MaybeDispatchKeypressEvents(keypressEvent, status,
+                                                 currentKeyEvent);
     currentKeyEvent->mKeyPressHandled =
         (status == nsEventStatus_eConsumeNoDefault);
     MOZ_LOG_KEY_OR_IME(
@@ -3995,7 +3980,7 @@ WidgetContentCommandEvent deleteCommandEvent(true, eContentCommandDelete,
   // Mark currentKeyEvent as "dispatched eKeyDown event" and actually do it.
   currentKeyEvent->mKeyDownDispatched = true;
 
-  RefPtr<nsCocoaWindow> widget(mWidget);
+  RefPtr<nsChildView> widget(mWidget);
 
   WidgetKeyboardEvent keydownEvent(true, eKeyDown, widget);
   // Don't mark the eKeyDown event as "processed by IME" if the composition
@@ -4644,7 +4629,7 @@ WidgetQueryContentEvent queryCharAtPointEvent(true, eQueryCharacterAtPoint,
  *
  ******************************************************************************/
 
-IMEInputHandler::IMEInputHandler(nsCocoaWindow* aWidget,
+IMEInputHandler::IMEInputHandler(nsChildView* aWidget,
                                  NSView<mozView>* aNativeView)
     : TextInputHandlerBase(aWidget, aNativeView),
       mPendingMethods(0),
@@ -4708,7 +4693,7 @@ WidgetQueryContentEvent queryCharAtPointEvent(true, eQueryCharacterAtPoint,
   ResetTimer();
 }
 
-bool IMEInputHandler::OnDestroyWidget(nsCocoaWindow* aDestroyingWidget) {
+bool IMEInputHandler::OnDestroyWidget(nsChildView* aDestroyingWidget) {
   MOZ_LOG(gIMELog, LogLevel::Info,
           ("%p IMEInputHandler::OnDestroyWidget, aDestroyingWidget=%p, "
            "sFocusedIMEHandler=%p, IsIMEComposing()=%s",
@@ -4716,7 +4701,7 @@ WidgetQueryContentEvent queryCharAtPointEvent(true, eQueryCharacterAtPoint,
            TrueOrFalse(IsIMEComposing())));
 
   // If we're not focused, the focused IMEInputHandler may have been
-  // created by another widget/nsCocoaWindow.
+  // created by another widget/nsChildView.
   if (sFocusedIMEHandler && sFocusedIMEHandler != this) {
     sFocusedIMEHandler->OnDestroyWidget(aDestroyingWidget);
   }
@@ -5044,10 +5029,8 @@ static NSTextCheckingType GetTextCheckingTypes() {
   }
 
   // Dismiss text substitution panel since this text change might be script etc
-  if (!mBlockDismissTextSubstitutionPanel) {
-    mProcessTextSubstitution = false;
-    DismissTextSubstitutionPanel();
-  }
+  mProcessTextSubstitution = false;
+  DismissTextSubstitutionPanel();
 
   // Set new text substitution data to show its panel by current changed text.
   //
@@ -5177,10 +5160,16 @@ WidgetQueryContentEvent queryTextContentEvent(true, eQueryTextContent,
   // NSTextCheckingResult.range is read only, so re-create this result object.
   NSRange candidatedRange = NSMakeRange(candidate.range.location + startFetch,
                                         candidate.range.length);
+  NSArray<NSString *> *anAlternativeString;
+  if(@available(macOS 10.8, *)) {
+    anAlternativeString = candidate.alternativeStrings;
+  } else {
+    anAlternativeString = @[];
+  }
   mCandidatedTextSubstitutionResult = [[NSTextCheckingResult
       correctionCheckingResultWithRange:candidatedRange
                       replacementString:candidate.replacementString
-                     alternativeStrings:candidate.alternativeStrings] retain];
+                     alternativeStrings:anAlternativeString] retain];
   mOriginalTextForTextSubstitution =
       Substring(queryTextContentEvent.mReply->DataRef(),
                 candidatedRange.location - startFetch, candidatedRange.length);
@@ -5230,12 +5219,17 @@ WidgetQueryContentEvent queryTextContentEvent(true, eQueryTextContent,
   if (!spellchecker) {
     return;
   }
+  NSArray<NSString *> *anotherAlternativeString;
+  if(@available(macOS 10.8, *)) {
+    anotherAlternativeString = mCandidatedTextSubstitutionResult.alternativeStrings;
+  } else {
+    anotherAlternativeString = @[];
+  }
   [spellchecker
       showCorrectionIndicatorOfType:NSCorrectionIndicatorTypeDefault
                       primaryString:mCandidatedTextSubstitutionResult
                                         .replacementString
-                 alternativeStrings:mCandidatedTextSubstitutionResult
-                                        .alternativeStrings
+                 alternativeStrings:anotherAlternativeString
                     forStringInRect:rect
                                view:mView
                   completionHandler:^(NSString* aAcceptedString) {
@@ -5279,11 +5273,6 @@ WidgetQueryContentEvent queryTextContentEvent(true, eQueryTextContent,
 void IMEInputHandler::DismissTextSubstitutionPanel() {
   NS_OBJC_BEGIN_TRY_IGNORE_BLOCK;
 
-  MOZ_LOG(gIMELog, LogLevel::Info,
-          ("%p IMEInputHandler::DismissTextSubstitutionPanel, "
-           "mProcessTextSubstitution=%s",
-           this, mProcessTextSubstitution ? "true" : "false"));
-
   NSSpellChecker* spellchecker = [NSSpellChecker sharedSpellChecker];
   if (!spellchecker) {
     return;
@@ -5331,7 +5320,7 @@ WidgetContentCommandEvent replaceTextEvent(true, eContentCommandReplaceText,
   }
 
   bool allowConsumeEvent = true;
-  if (!IsIMEComposing()) {
+  if (nsCocoaFeatures::OnCatalinaOrLater() && !IsIMEComposing()) {
     // Hack for bug of Korean IMEs on Catalina (10.15).
     // If we are inactivated during composition, active Korean IME keeps
     // consuming all mousedown events of any mouse buttons.  So, we should
@@ -5367,7 +5356,7 @@ WidgetContentCommandEvent replaceTextEvent(true, eContentCommandReplaceText,
 NS_IMPL_ISUPPORTS(TextInputHandlerBase, TextEventDispatcherListener,
                   nsISupportsWeakReference)
 
-TextInputHandlerBase::TextInputHandlerBase(nsCocoaWindow* aWidget,
+TextInputHandlerBase::TextInputHandlerBase(nsChildView* aWidget,
                                            NSView<mozView>* aNativeView)
     : mWidget(aWidget), mDispatcher(aWidget->GetTextEventDispatcher()) {
   gHandlerInstanceCount++;
@@ -5381,7 +5370,7 @@ WidgetContentCommandEvent replaceTextEvent(true, eContentCommandReplaceText,
   }
 }
 
-bool TextInputHandlerBase::OnDestroyWidget(nsCocoaWindow* aDestroyingWidget) {
+bool TextInputHandlerBase::OnDestroyWidget(nsChildView* aDestroyingWidget) {
   MOZ_LOG_KEY_OR_IME(LogLevel::Info,
                      ("%p TextInputHandlerBase::OnDestroyWidget, "
                       "aDestroyingWidget=%p, mWidget=%p",
@@ -5436,7 +5425,7 @@ WidgetContentCommandEvent replaceTextEvent(true, eContentCommandReplaceText,
                     modifierFlags:modifierFlags
                         timestamp:0
                      windowNumber:windowNumber
-                          context:nil
+                          context:[NSGraphicsContext currentContext]
                        characters:XPCOMStringToNSString(aCharacters)
       charactersIgnoringModifiers:XPCOMStringToNSString(aUnmodifiedCharacters)
                         isARepeat:NO
@@ -5702,7 +5691,7 @@ WidgetContentCommandEvent replaceTextEvent(true, eContentCommandReplaceText,
                           modifierFlags:[mKeyEvent modifierFlags]
                               timestamp:[mKeyEvent timestamp]
                            windowNumber:[mKeyEvent windowNumber]
-                                context:nil
+                                context:[mKeyEvent context]
                              characters:unhandledNSString
             charactersIgnoringModifiers:[mKeyEvent charactersIgnoringModifiers]
                               isARepeat:[mKeyEvent isARepeat]
