diff --git a/widget/cocoa/ViewRegion.h b/widget/cocoa/ViewRegion.h
index 6d4b1afe55..f972428918 100644
--- a/widget/cocoa/ViewRegion.h
+++ b/widget/cocoa/ViewRegion.h
@@ -11,7 +11,7 @@
 #include "nsRegion.h"
 #include "nsTArray.h"
 
-class nsCocoaWindow;
+class nsChildView;
 
 @class NSView;
 
@@ -30,7 +30,7 @@ class ViewRegion {
   /**
    * Update the region.
    * @param aRegion  The new region.
-   * @param aCoordinateConverter  The nsCocoaWindow to use for converting
+   * @param aCoordinateConverter  The nsChildView to use for converting
    *   LayoutDeviceIntRect device pixel coordinates into Cocoa NSRect
    * coordinates.
    * @param aContainerView  The view that's going to be the superview of the
@@ -39,7 +39,7 @@ class ViewRegion {
    * @return  Whether or not the region changed.
    */
   bool UpdateRegion(const mozilla::LayoutDeviceIntRegion& aRegion,
-                    const nsCocoaWindow& aCoordinateConverter,
+                    const nsChildView& aCoordinateConverter,
                     NSView* aContainerView, NSView* (^aViewCreationCallback)());
 
   /**
