diff --git a/widget/cocoa/nsAppShell.mm b/widget/cocoa/nsAppShell.mm
index e725a11a57..bcfe80d09a 100644
--- a/widget/cocoa/nsAppShell.mm
+++ b/widget/cocoa/nsAppShell.mm
@@ -29,7 +29,7 @@
 #include "nsObjCExceptions.h"
 #include "nsCocoaUtils.h"
 #include "nsCocoaFeatures.h"
-#include "nsCocoaWindow.h"
+#include "nsChildView.h"
 #include "nsToolkit.h"
 #include "TextInputHandler.h"
 #include "mozilla/BackgroundHangMonitor.h"
@@ -178,7 +178,13 @@ - (void)sendEvent:(NSEvent*)anEvent {
   [super sendEvent:anEvent];
 }
 
+#if defined(MAC_OS_X_VERSION_10_12) && MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_12 && \
+    __LP64__
+// 10.12 changed `mask` to NSEventMask (unsigned long long) for x86_64 builds.
 - (NSEvent*)nextEventMatchingMask:(NSEventMask)mask
+#else
+- (NSEvent*)nextEventMatchingMask:(NSUInteger)mask
+#endif
                         untilDate:(NSDate*)expiration
                            inMode:(NSString*)mode
                           dequeue:(BOOL)flag {
@@ -378,14 +384,52 @@ void RunLoopObserverCallback(CFRunLoopObserverRef aObserver,
       (XRE_GetProcessType() != GeckoProcessType_Socket);
 
   if (isNSApplicationProcessType) {
-    // This call initializes NSApplication unless:
-    // 1) we're using xre -- NSApp's already been initialized by
-    //    MacApplicationDelegate.mm's EnsureUseCocoaDockAPI().
-    // 2) an embedding app that uses NSApplicationMain() is running -- NSApp's
-    //    already been initialized and its main run loop is already running.
-    [[NSBundle mainBundle] loadNibNamed:@"res/MainMenu"
-                                  owner:[GeckoNSApplication sharedApplication]
-                        topLevelObjects:nil];
+    if(@available(macOS 10.8, *)) {
+      // This call initializes NSApplication unless:
+      // 1) we're using xre -- NSApp's already been initialized by
+      //    MacApplicationDelegate.mm's EnsureUseCocoaDockAPI().
+      // 2) an embedding app that uses NSApplicationMain() is running -- NSApp's
+      //    already been initialized and its main run loop is already running.
+      [[NSBundle mainBundle] loadNibNamed:@"res/MainMenu"
+                                    owner:[GeckoNSApplication sharedApplication]
+                          topLevelObjects:nil];
+    } else {
+      //on 10.7 and probably lower, the audio decoding (utility) process
+      //behaves weird, so we have to call it using the mainBundle approach
+      //we cannot use the mainBundle approach all the time because there
+      //are issues with the menu after closing a window. at least this approach
+      //is faithful since we launch a utility process slightly differently,
+      //as opposed to not launching it through nsBundle altogether (previous
+      //way)
+      //so we have no choice but to add this until i find a better fix.
+      if((XRE_GetProcessType() != GeckoProcessType_Utility)) {
+        nsCOMPtr<nsIFile> nibFile;
+        nsresult rv = NS_GetSpecialDirectory(NS_GRE_DIR, getter_AddRefs(nibFile));
+        NS_ENSURE_SUCCESS(rv, rv);
+
+        nibFile->AppendNative("res"_ns);
+        nibFile->AppendNative("MainMenu.nib"_ns);
+
+        nsAutoCString nibPath;
+        rv = nibFile->GetNativePath(nibPath);
+        NS_ENSURE_SUCCESS(rv, rv);
+        // Get the path of the nib file, which lives in the GRE location
+        [NSBundle loadNibFile:[NSString stringWithUTF8String:(const char*)nibPath.get()]
+         externalNameTable:[NSDictionary dictionaryWithObjectsAndKeys:[GeckoNSApplication sharedApplication], NSNibOwner,
+                                                                     nil, NSNibTopLevelObjects,
+                                                                     nil]
+         withZone:nil];
+     } else {
+       //big thanks to the @uTox team for their example that helped me bridge the old fawx code to
+       //the proper form that uses mainBundle and topLevelObjects for loading on < 10.8:
+       //https://github.com/uTox/uTox/blob/8d5cd82e3554a3108f2aff9411bbe551e1e443b0/src/cocoa/main.m#L513
+       [NSBundle loadNibFile:[[NSBundle mainBundle] pathForResource:@"res/MainMenu" ofType:@"nib"]
+                 externalNameTable:[NSDictionary dictionaryWithObjectsAndKeys:[GeckoNSApplication sharedApplication], NSNibOwner,
+                                                                    nil, NSNibTopLevelObjects,
+                                                                    nil]
+        withZone:nil];
+     }
+   }
   }
 
   mDelegate = [[AppShellDelegate alloc] initWithAppShell:this];
@@ -433,8 +477,10 @@ void RunLoopObserverCallback(CFRunLoopObserverRef aObserver,
     } else {
       screenManager.SetHelper(mozilla::MakeUnique<ScreenHelperCocoa>());
     }
-
-    InitMemoryPressureObserver();
+    if(__builtin_available(macOS 10.9, *)) {
+      //does not exist in 10.8 or lower. guarded accordingly.
+      InitMemoryPressureObserver();
+    }
   }
 
   nsresult rv = nsBaseAppShell::Init();
@@ -450,7 +496,8 @@ void RunLoopObserverCallback(CFRunLoopObserverRef aObserver,
   }
 
 #if !defined(RELEASE_OR_BETA) || defined(DEBUG)
-  if (Preferences::GetBool("security.sandbox.mac.track.violations", false)) {
+  if (nsCocoaFeatures::OnMavericksOrLater() && 
+      Preferences::GetBool("security.sandbox.mac.track.violations", false)) {
     nsSandboxViolationSink::Start();
   }
 #endif
@@ -793,7 +840,7 @@ void RunLoopObserverCallback(CFRunLoopObserverRef aObserver,
   NS_OBJC_END_TRY_IGNORE_BLOCK;
 
   if (!moreEvents) {
-    nsCocoaWindow::UpdateCurrentInputEventCount();
+    nsChildView::UpdateCurrentInputEventCount();
   }
 
   return moreEvents;
@@ -889,7 +936,9 @@ static void PinSidecarCoreTextCStringSections() {
   mTerminated = true;
 
 #if !defined(RELEASE_OR_BETA) || defined(DEBUG)
-  nsSandboxViolationSink::Stop();
+  if (nsCocoaFeatures::OnMavericksOrLater()) {
+    nsSandboxViolationSink::Stop();
+  }
 #endif
 
   // Quoting from Apple's doc on the [NSApplication stop:] method (from their
