diff --git a/widget/cocoa/nsClipboard.mm b/widget/cocoa/nsClipboard.mm
index fa84de48fc..3465332a56 100644
--- a/widget/cocoa/nsClipboard.mm
+++ b/widget/cocoa/nsClipboard.mm
@@ -66,7 +66,10 @@
     case nsIClipboard::kGlobalClipboard:
       return [NSPasteboard generalPasteboard];
     case nsIClipboard::kFindClipboard:
-      return [NSPasteboard pasteboardWithName:NSPasteboardNameFind];
+      if (@available(macOS 10.13, *)) {
+          return [NSPasteboard pasteboardWithName:NSPasteboardNameFind];
+      }
+      return [NSPasteboard pasteboardWithName:NSFindPboard];
     default:
       return nil;
   }
