diff --git a/widget/cocoa/nsMacDockSupport.mm b/widget/cocoa/nsMacDockSupport.mm
index 97c0e10160..5f850ad5b2 100644
--- a/widget/cocoa/nsMacDockSupport.mm
+++ b/widget/cocoa/nsMacDockSupport.mm
@@ -61,7 +61,7 @@ - (void)drawRect:(NSRect)aRect {
   progressFillRect.size.width *= mFractionValue;
   [NSGraphicsContext saveGraphicsState];
   [NSBezierPath clipRect:progressFillRect];
-  [[NSColor controlAccentColor] setFill];
+  [ControlAccentColor() setFill];
   [path fill];
   [NSGraphicsContext restoreGraphicsState];
 
@@ -324,11 +324,11 @@ CFURLRef _CFURLCreateFromPropertyListRepresentation(
   if (![aPersistantApp isKindOfClass:[NSDictionary class]]) {
     return nil;
   }
-  NSDictionary* tileData = aPersistantApp[kDockTileDataKey];
+  NSDictionary* tileData = [aPersistantApp objectForKey:kDockTileDataKey];
   if (![tileData isKindOfClass:[NSDictionary class]]) {
     return nil;
   }
-  NSDictionary* fileData = tileData[kDockFileDataKey];
+  NSDictionary* fileData = [tileData objectForKey:kDockFileDataKey];
   if (![fileData isKindOfClass:[NSDictionary class]]) {
     // Some special tiles may not have DockFileData but we can ignore those.
     return nil;
@@ -527,25 +527,42 @@ void RefreshDock(NSDictionary* aDockPlist) {
     [arguments addObject:mozilla::XPCOMStringToNSString(arg)];
   }
 
-  NSWorkspaceOpenConfiguration* config =
-      [NSWorkspaceOpenConfiguration configuration];
-  [config setArguments:arguments];
-  [config setCreatesNewApplicationInstance:YES];
-  [config setEnvironment:[[NSProcessInfo processInfo] environment]];
+  if (@available(macOS 10.15, *)) {
+    NSWorkspaceOpenConfiguration* config =
+        [NSWorkspaceOpenConfiguration configuration];
+    [config setArguments:arguments];
+    [config setCreatesNewApplicationInstance:YES];
+    [config setEnvironment:[[NSProcessInfo processInfo] environment]];
+    if (aLaunchOptions) {
+      bool val = false;
+      if (NS_SUCCEEDED(aLaunchOptions->GetAddsToRecentItems(&val))) {
+        [config setAddsToRecentItems:val];
+      }
+    }
 
-  if (aLaunchOptions) {
-    bool val = false;
-    if (NS_SUCCEEDED(aLaunchOptions->GetAddsToRecentItems(&val))) {
-      [config setAddsToRecentItems:val];
+    [[NSWorkspace sharedWorkspace]
+        openApplicationAtURL:[NSURL fileURLWithPath:launchPath]
+               configuration:config
+           completionHandler:^(NSRunningApplication* aChild, NSError* aError){
+           }];
+  } else {
+    NSError *error=nil;
+    unsigned options = NSWorkspaceLaunchAsync | NSWorkspaceLaunchNewInstance;
+    if (aLaunchOptions) {
+      bool val = false;
+      if (NS_SUCCEEDED(aLaunchOptions->GetAddsToRecentItems(&val))) {
+         if (!val) { //only add to the options line if we DON'T WANT IT TO SHOW in recent items 
+           options |= NSWorkspaceLaunchWithoutAddingToRecents;
+          }
+      }
     }
+    [[NSWorkspace sharedWorkspace] 
+        launchApplicationAtURL:[NSBundle mainBundle].bundleURL
+        options:options
+        configuration:@{NSWorkspaceLaunchConfigurationArguments:arguments, 
+                        NSWorkspaceLaunchConfigurationEnvironment:[[NSProcessInfo processInfo] environment]}
+        error:&error];
   }
-
-  [[NSWorkspace sharedWorkspace]
-      openApplicationAtURL:[NSURL fileURLWithPath:launchPath]
-             configuration:config
-         completionHandler:^(NSRunningApplication* aChild, NSError* aError){
-         }];
-
   return NS_OK;
 
   NS_OBJC_END_TRY_BLOCK_RETURN(NS_ERROR_FAILURE);
