diff --git a/widget/cocoa/nsTouchBar.mm b/widget/cocoa/nsTouchBar.mm
index 067ce03129..d0db157abb 100644
--- a/widget/cocoa/nsTouchBar.mm
+++ b/widget/cocoa/nsTouchBar.mm
@@ -15,6 +15,8 @@
 #include "nsTouchBarInputIcon.h"
 #include "nsWidgetsCID.h"
 
+#include "nsCocoaFeatures.h"
+
 @implementation nsTouchBar
 
 // Used to tie action strings to buttons.
@@ -85,16 +87,16 @@ - (instancetype)initWithInputs:(NSMutableArray<TouchBarInput*>*)aInputs {
         // identifier, that means updateItem fired before this initialization.
         // The input cached by updateItem is more current, so we should use that
         // one.
-        if (self.mappedLayoutItems[newInputIdentifier]) {
-          convertedInput = self.mappedLayoutItems[newInputIdentifier];
+        if ([self.mappedLayoutItems objectForKey:newInputIdentifier]) {
+          convertedInput = [self.mappedLayoutItems objectForKey:newInputIdentifier];
         } else {
           convertedInput = [[TouchBarInput alloc] initWithXPCOM:input];
           // Add new input to dictionary for lookup of properties in delegate.
-          self.mappedLayoutItems[[convertedInput nativeIdentifier]] =
-              convertedInput;
+          [self.mappedLayoutItems setObject:convertedInput forKey:[convertedInput nativeIdentifier]]; 
+        }
+        if([convertedInput nativeIdentifier]) {
+          [orderedIdentifiers insertObject:[convertedInput nativeIdentifier] atIndex:i];
         }
-
-        orderedIdentifiers[i] = [convertedInput nativeIdentifier];
       }
       [orderedIdentifiers addObject:@"NSTouchBarItemIdentifierFlexibleSpace"];
       self.customizationAllowedItemIdentifiers = [orderedIdentifiers copy];
@@ -114,7 +116,7 @@ - (instancetype)initWithInputs:(NSMutableArray<TouchBarInput*>*)aInputs {
       NSMutableArray* defaultItemIdentifiers =
           [NSMutableArray arrayWithCapacity:[aInputs count]];
       for (TouchBarInput* input in aInputs) {
-        self.mappedLayoutItems[[input nativeIdentifier]] = input;
+        [self.mappedLayoutItems setObject:input forKey:[input nativeIdentifier]];
         [defaultItemIdentifiers addObject:[input nativeIdentifier]];
       }
       self.defaultItemIdentifiers = [defaultItemIdentifiers copy];
@@ -156,7 +158,7 @@ - (NSTouchBarItem*)touchBar:(NSTouchBar*)aTouchBar
     return nil;
   }
 
-  TouchBarInput* input = self.mappedLayoutItems[aIdentifier];
+  TouchBarInput* input = [self.mappedLayoutItems objectForKey:aIdentifier];
   if (!input) {
     return nil;
   }
@@ -226,7 +228,7 @@ - (bool)updateItem:(TouchBarInput*)aInput {
   //   * It is contained within a popover, or
   //   * It simply does not exist.
   // We check for each possibility here.
-  if (!self.mappedLayoutItems[[aInput nativeIdentifier]]) {
+  if (![self.mappedLayoutItems objectForKey:[aInput nativeIdentifier]]) {
     if ([self maybeUpdateScrollViewChild:aInput]) {
       return true;
     }
@@ -269,7 +271,7 @@ - (bool)updateItem:(TouchBarInput*)aInput {
 
 - (bool)maybeUpdatePopoverChild:(TouchBarInput*)aInput {
   for (NSTouchBarItemIdentifier identifier in self.mappedLayoutItems) {
-    TouchBarInput* potentialPopover = self.mappedLayoutItems[identifier];
+    TouchBarInput* potentialPopover = [self.mappedLayoutItems objectForKey:identifier];
     if ([potentialPopover baseType] != TouchBarInputBaseType::kPopover) {
       continue;
     }
@@ -287,7 +289,7 @@ - (bool)maybeUpdatePopoverChild:(TouchBarInput*)aInput {
 
 - (bool)maybeUpdateScrollViewChild:(TouchBarInput*)aInput {
   NSCustomTouchBarItem* scrollViewButton =
-      self.scrollViewButtons[[aInput nativeIdentifier]];
+      [self.scrollViewButtons objectForKey:[aInput nativeIdentifier]];
   if (scrollViewButton) {
     // ScrollView buttons are similar to mainButtons except for their width.
     [self updateMainButton:scrollViewButton
@@ -301,12 +303,12 @@ - (bool)maybeUpdateScrollViewChild:(TouchBarInput*)aInput {
   }
   // Updating the TouchBarInput* in the ScrollView's mChildren array.
   for (NSTouchBarItemIdentifier identifier in self.mappedLayoutItems) {
-    TouchBarInput* potentialScrollView = self.mappedLayoutItems[identifier];
+    TouchBarInput* potentialScrollView = [self.mappedLayoutItems objectForKey:identifier];
     if ([potentialScrollView baseType] != TouchBarInputBaseType::kScrollView) {
       continue;
     }
     for (uint32_t i = 0; i < [[potentialScrollView children] count]; ++i) {
-      TouchBarInput* child = [potentialScrollView children][i];
+      TouchBarInput* child = [[potentialScrollView children] objectAtIndex:i];
       if (![[child nativeIdentifier]
               isEqualToString:[aInput nativeIdentifier]]) {
         continue;
@@ -320,8 +322,8 @@ - (bool)maybeUpdateScrollViewChild:(TouchBarInput*)aInput {
 }
 
 - (void)replaceMappedLayoutItem:(TouchBarInput*)aItem {
-  [self.mappedLayoutItems[[aItem nativeIdentifier]] release];
-  self.mappedLayoutItems[[aItem nativeIdentifier]] = aItem;
+  [[self.mappedLayoutItems objectForKey:[aItem nativeIdentifier]] release];
+  [self.mappedLayoutItems setObject:aItem forKey:[aItem nativeIdentifier]]; 
 }
 
 - (void)updateButton:(NSCustomTouchBarItem*)aButton
@@ -330,7 +332,7 @@ - (void)updateButton:(NSCustomTouchBarItem*)aButton
     return;
   }
 
-  TouchBarInput* input = self.mappedLayoutItems[aIdentifier];
+  TouchBarInput* input = [self.mappedLayoutItems objectForKey:aIdentifier];
   if (!input) {
     return;
   }
@@ -361,7 +363,7 @@ - (void)updateMainButton:(NSCustomTouchBarItem*)aMainButton
     return;
   }
 
-  TouchBarInput* input = self.mappedLayoutItems[aIdentifier];
+  TouchBarInput* input = [self.mappedLayoutItems objectForKey:aIdentifier];
   if (!input) {
     return;
   }
@@ -378,8 +380,10 @@ - (void)updateMainButton:(NSCustomTouchBarItem*)aMainButton
   button.imageHugsTitle = YES;
   [button.widthAnchor constraintGreaterThanOrEqualToConstant:MAIN_BUTTON_WIDTH]
       .active = YES;
-  [button setContentHuggingPriority:1.0
-                     forOrientation:NSLayoutConstraintOrientationHorizontal];
+  if(nsCocoaFeatures::OnLionOrLater()) {
+    [button setContentHuggingPriority:1.0
+                       forOrientation:NSLayoutConstraintOrientationHorizontal];
+  }
 }
 
 - (void)updatePopover:(NSPopoverTouchBarItem*)aPopoverItem
@@ -388,7 +392,7 @@ - (void)updatePopover:(NSPopoverTouchBarItem*)aPopoverItem
     return;
   }
 
-  TouchBarInput* input = self.mappedLayoutItems[aIdentifier];
+  TouchBarInput* input = [self.mappedLayoutItems objectForKey:aIdentifier];
   if (!input) {
     return;
   }
@@ -422,7 +426,7 @@ - (void)updateScrollView:(NSCustomTouchBarItem*)aScrollViewItem
     return;
   }
 
-  TouchBarInput* input = self.mappedLayoutItems[aIdentifier];
+  TouchBarInput* input = [self.mappedLayoutItems objectForKey:aIdentifier];
   if (!input || ![input children]) {
     return;
   }
@@ -457,8 +461,8 @@ - (void)updateScrollView:(NSCustomTouchBarItem*)aScrollViewItem
         .active = YES;
 
     NSCustomTouchBarItem* tempItem =
-        self.scrollViewButtons[[childInput nativeIdentifier]];
-    self.scrollViewButtons[[childInput nativeIdentifier]] = newItem;
+        [self.scrollViewButtons objectForKey:[childInput nativeIdentifier]];
+    [self.scrollViewButtons setObject:newItem forKey:[childInput nativeIdentifier]]; 
     [tempItem release];
 
     button.translatesAutoresizingMaskIntoConstraints = NO;
@@ -476,18 +480,19 @@ - (void)updateScrollView:(NSCustomTouchBarItem*)aScrollViewItem
   }
   layoutFormat =
       [layoutFormat stringByAppendingString:[NSString stringWithFormat:@"|"]];
-  NSArray* hConstraints = [NSLayoutConstraint
-      constraintsWithVisualFormat:layoutFormat
-                          options:NSLayoutFormatAlignAllCenterY
-                          metrics:nil
-                            views:constraintViews];
-  NSScrollView* scrollView = [[NSScrollView alloc]
-      initWithFrame:CGRectMake(0, 0, size.width, size.height)];
-  [documentView setFrame:NSMakeRect(0, 0, size.width, size.height)];
-  [NSLayoutConstraint activateConstraints:hConstraints];
-  scrollView.documentView = documentView;
-
+  if(nsCocoaFeatures::OnLionOrLater()) {
+    NSArray* hConstraints = [NSLayoutConstraint
+        constraintsWithVisualFormat:layoutFormat
+                            options:NSLayoutFormatAlignAllCenterY
+                            metrics:nil
+                              views:constraintViews];
+    NSScrollView* scrollView = [[NSScrollView alloc]
+        initWithFrame:CGRectMake(0, 0, size.width, size.height)];
+    [documentView setFrame:NSMakeRect(0, 0, size.width, size.height)];
+    [NSLayoutConstraint activateConstraints:hConstraints];
+    scrollView.documentView = documentView;
   aScrollViewItem.view = scrollView;
+  }
 }
 
 - (void)updateLabel:(NSTextField*)aLabel
@@ -496,7 +501,7 @@ - (void)updateLabel:(NSTextField*)aLabel
     return;
   }
 
-  TouchBarInput* input = self.mappedLayoutItems[aIdentifier];
+  TouchBarInput* input = [self.mappedLayoutItems objectForKey:aIdentifier];
   if (!input || ![input title]) {
     return;
   }
@@ -505,7 +510,7 @@ - (void)updateLabel:(NSTextField*)aLabel
 
 - (NSTouchBarItem*)makeShareScrubberForIdentifier:
     (NSTouchBarItemIdentifier)aIdentifier {
-  TouchBarInput* input = self.mappedLayoutItems[aIdentifier];
+  TouchBarInput* input = [self.mappedLayoutItems objectForKey:aIdentifier];
   // System-default share menu
   NSSharingServicePickerTouchBarItem* servicesItem =
       [[NSSharingServicePickerTouchBarItem alloc]
@@ -544,7 +549,7 @@ - (void)touchBarAction:(id)aSender {
     return;
   }
 
-  TouchBarInput* input = self.mappedLayoutItems[identifier];
+  TouchBarInput* input = [self.mappedLayoutItems objectForKey:identifier];
   if (!input) {
     return;
   }
@@ -581,7 +586,7 @@ - (void)releaseJSObjects {
   mTouchBarHelper = nil;
 
   for (NSTouchBarItemIdentifier identifier in self.mappedLayoutItems) {
-    TouchBarInput* input = self.mappedLayoutItems[identifier];
+    TouchBarInput* input = [self.mappedLayoutItems objectForKey:identifier];
     if (!input) {
       continue;
     }
