diff --git a/browser/app/moz.build b/browser/app/moz.build
index b5f5461fd0835..a34391ba1b161 100644
--- a/browser/app/moz.build
+++ b/browser/app/moz.build
@@ -124,6 +124,7 @@ if CONFIG["OS_ARCH"] == "Darwin":
     OS_LIBS += [
         "-framework CoreFoundation",
     ]
+    LDFLAGS += ["-Wl,-rpath,@executable_path/./"]
 
 if CONFIG["MOZ_SANDBOX"] and CONFIG["OS_ARCH"] == "WINNT":
     # For sandbox includes and the include dependencies those have
