diff --git a/build/RunCbindgen.py b/build/RunCbindgen.py
index bb0de610edb98..2459b8aba3f97 100644
--- a/build/RunCbindgen.py
+++ b/build/RunCbindgen.py
@@ -55,7 +55,7 @@ def generate_metadata(output, cargo_config):
     # within a tree, the Rust dependencies have been vendored in so Cargo won't
     # touch the lock file.
     if not buildconfig.substs.get("JS_STANDALONE"):
-        args.append("--frozen")
+        args.append("--offline")
 
     stdout, returncode = _run_process(args)
 
