diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
index 47061339da913..9116bdbbfc5b7 100644
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -241,7 +241,6 @@ with only_when(host_is_osx | target_is_osx):
 
     def mac_sdk_min_version():
         return "26.2"
-
     @depends(
         "--with-macos-sdk",
         host,
@@ -306,6 +305,36 @@ with only_when(host_is_osx | target_is_osx):
     set_config("MACOS_SDK_DIR", macos_sdk)
 
 
+with only_when(target_is_osx):
+    with only_when(cross_compiling):
+        option(
+            "--with-macos-private-frameworks",
+            env="MACOS_PRIVATE_FRAMEWORKS_DIR",
+            nargs=1,
+            help="Location of private frameworks to use",
+        )
+
+        @depends_if("--with-macos-private-frameworks")
+        @imports(_from="os.path", _import="isdir")
+        def macos_private_frameworks(value):
+            if value and not isdir(value[0]):
+                die(
+                    "PrivateFrameworks not found not found in %s. When using "
+                    "--with-macos-private-frameworks, you must specify a valid "
+                    "directory",
+                    value[0],
+                )
+            return value[0]
+
+    @depends(macos_private_frameworks, macos_sdk)
+    def macos_private_frameworks(value, sdk):
+        if value:
+            return value
+        return os.path.join(sdk or "/", "System/Library/PrivateFrameworks")
+
+    set_config("MACOS_PRIVATE_FRAMEWORKS_DIR", macos_private_frameworks)
+
+
 with only_when(target_is_ios):
     # iOS deployment target version
     # ==============================================================
@@ -2004,7 +2033,7 @@ def select_linker_tmpl(host_or_target):
             # ensure consistent output.
             env["LC_ALL"] = "C"
             retcode, stdout, stderr = get_cmd_output(*cmd, env=env)
-            if retcode == 1 and "Logging ld64 options" in stderr:
+            if retcode == 1 and "unknown options" in stderr:
                 kind = "ld64"
 
             elif retcode != 0:
