diff --git a/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm b/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm
index 3df9964720680..a9eab47e0238c 100644
--- a/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm
+++ b/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm
@@ -125,9 +125,9 @@ - (void)capturer:(RTCVideoCapturer* _Nonnull)capturer
     Clock* _Nonnull clock, const char* _Nullable aDeviceUniqueIdUTF8) {
   std::string uniqueId(aDeviceUniqueIdUTF8);
 
-  for (AVCaptureDevice* device in [RTCCameraVideoCapturer
-           captureDevicesWithDeviceTypes:[RTCCameraVideoCapturer
-                                             defaultCaptureDeviceTypes]]) {
+  for (AVCaptureDevice* device in [RTCCameraVideoCapturer captureDevices]
+           /* //FUCK YOU captureDevicesWithDeviceTypes:[RTCCameraVideoCapturer
+                                             defaultCaptureDeviceTypes]]*/) {
     if ([NSString stdStringForString:device.uniqueID] == uniqueId) {
       webrtc::scoped_refptr<VideoCaptureModule> module(
           new webrtc::RefCountedObject<VideoCaptureAvFoundation>(clock,
