diff --git a/gfx/2d/ScaledFontMac.cpp b/gfx/2d/ScaledFontMac.cpp
index 6bfbcfe519a66..de186c9462b2b 100644
--- a/gfx/2d/ScaledFontMac.cpp
+++ b/gfx/2d/ScaledFontMac.cpp
@@ -21,6 +21,8 @@
 #ifdef MOZ_WIDGET_UIKIT
 #  include <CoreFoundation/CoreFoundation.h>
 #endif
+#include "gfxPlatform.h"  // just for HasVariationFontSupport()
+#include "nsCocoaFeatures.h"
 #include "mozilla/gfx/Logging.h"
 
 #ifdef MOZ_WIDGET_COCOA
@@ -102,10 +104,10 @@ CTFontRef CreateCTFontFromCGFontWithVariations(CGFontRef aCGFont, CGFloat aSize,
     return ctFont.forget();
   }
 #endif
-
   // Older implementation used up to macOS 12.
   CTFontRef ctFont;
-  if (aInstalledFont) {
+  if (nsCocoaFeatures::OnSierraExactly() ||
+        (aInstalledFont && nsCocoaFeatures::OnHighSierraOrLater())) {
     AutoRelease<CFDictionaryRef> vars(CGFontCopyVariations(aCGFont));
     if (vars) {
       AutoRelease<CFDictionaryRef> varAttr(CFDictionaryCreate(
@@ -375,6 +377,17 @@ bool UnscaledFontMac::GetFontDescriptor(FontDescriptorOutput aCb,
     return false;
   }
 
+  // Bug 1690235: using a font descriptor may fail for variation fonts
+  // on systems prior to 10.15, so we return false in this case.
+  if (gfxPlatform::GetPlatform()->HasVariationFontSupport() &&
+      !nsCocoaFeatures::OnCatalinaOrLater()) {
+    CFDataRef data = CGFontCopyTableForTag(mFont, 0x66766172);  // 'fvar'
+    if (data) {
+      CFRelease(data);
+      return false;
+    }
+  }
+
   AutoRelease<CFStringRef> psname(CGFontCopyPostScriptName(mFont));
   if (!psname) {
     return false;
@@ -435,6 +448,11 @@ static void CollectVariationsFromDictionary(const void* aKey,
 
 static bool GetVariationsForCTFont(CTFontRef aCTFont,
                                    std::vector<FontVariation>* aOutVariations) {
+  // Avoid calling potentially buggy variation APIs on pre-Sierra macOS
+  // versions (see bug 1331683)
+  if (!nsCocoaFeatures::OnSierraOrLater()) {
+    return true;
+  }
   if (!aCTFont) {
     return true;
   }
@@ -508,6 +526,12 @@ ScaledFontMac::InstanceData::InstanceData(
 static CFDictionaryRef CreateVariationDictionaryOrNull(
     CGFontRef aCGFont, CFArrayRef& aCGAxesCache, CFArrayRef& aCTAxesCache,
     uint32_t aVariationCount, const FontVariation* aVariations) {
+  // Avoid calling potentially buggy variation APIs on pre-Sierra macOS
+  // versions (see bug 1331683)
+  if (!nsCocoaFeatures::OnSierraOrLater()) {
+    return nullptr;
+  }
+
   if (!aCGAxesCache) {
     aCGAxesCache = CGFontCopyVariationAxes(aCGFont);
     if (!aCGAxesCache) {
@@ -617,6 +641,12 @@ static CFDictionaryRef CreateVariationDictionaryOrNull(
 static CFDictionaryRef CreateVariationTagDictionaryOrNull(
     CTFontRef aCTFont, uint32_t aVariationCount,
     const FontVariation* aVariations) {
+  // Avoid calling potentially buggy variation APIs on pre-Sierra macOS
+  // versions (see bug 1331683)
+  if (!nsCocoaFeatures::OnSierraOrLater()) {
+    return nullptr;
+  }
+
   AutoRelease<CFArrayRef> axes(CTFontCopyVariationAxes(aCTFont));
   CFIndex axisCount = CFArrayGetCount(axes);
 
