diff --git a/image/decoders/icon/mac/nsIconChannelCocoa.mm b/image/decoders/icon/mac/nsIconChannelCocoa.mm
index 6b855d8c859dc..c71966089629c 100644
--- a/image/decoders/icon/mac/nsIconChannelCocoa.mm
+++ b/image/decoders/icon/mac/nsIconChannelCocoa.mm
@@ -347,11 +347,12 @@
 
   NSGraphicsContext* oldContext = [NSGraphicsContext currentContext];
   [NSGraphicsContext
-      setCurrentContext:[NSGraphicsContext graphicsContextWithCGContext:ctx
+      setCurrentContext:[NSGraphicsContext graphicsContextWithGraphicsPort:ctx
                                                                 flipped:NO]];
 
-  [iconImage drawInRect:NSMakeRect(0, 0, width, height)];
-
+  if(@available(macOS 10.9, *)) {
+     [iconImage drawInRect:NSMakeRect(0, 0, width, height)];
+  }
   [NSGraphicsContext setCurrentContext:oldContext];
 
   CGContextRelease(ctx);
