diff --git a/netwerk/protocol/http/nsHttpHandler.cpp b/netwerk/protocol/http/nsHttpHandler.cpp
index 814701e907747..b0d6057d1a9f6 100644
--- a/netwerk/protocol/http/nsHttpHandler.cpp
+++ b/netwerk/protocol/http/nsHttpHandler.cpp
@@ -111,6 +111,7 @@
 
 #if defined(XP_MACOSX)
 #  include <CoreServices/CoreServices.h>
+#  include "nsCocoaFeatures.h"
 #endif
 
 //-----------------------------------------------------------------------------
@@ -1189,7 +1190,8 @@ void nsHttpHandler::InitUserAgentComponents() {
 #  endif
 
 #elif defined(XP_MACOSX)
-  mOscpu.AssignLiteral("Intel Mac OS X 10.15");
+   // Always return an "Intel" UA string, even on ARM64 macOS like Safari does.
+   mOscpu = nsPrintfCString("Intel Mac OS X 10.15");
 #elif defined(ANDROID)
   mOscpu.AssignLiteral("Linux armv81");
 #elif defined(XP_IOS)
