diff --git a/security/manager/ssl/osclientcerts/src/backend_macos.rs b/security/manager/ssl/osclientcerts/src/backend_macos.rs
index af0a7370b3448..12b2fa7d5f6c0 100644
--- a/security/manager/ssl/osclientcerts/src/backend_macos.rs
+++ b/security/manager/ssl/osclientcerts/src/backend_macos.rs
@@ -23,21 +23,54 @@ use rsclientcerts_util::error::{Error, ErrorType};
 use std::collections::BTreeMap;
 use std::convert::TryInto;
 use std::os::raw::c_void;
+use std::str::*;
 use std::time::{Duration, Instant};
 use xpcom::interfaces::nsIEventTarget;
 use xpcom::{RefPtr, XpCom};
 
+
 // Normally we would generate this with a build script, but macos is
 // cross-compiled on linux, and we'd have to figure out e.g. include paths,
 // etc.. This is easier.
 include!("bindings_macos.rs");
 
+//rip cubeb-audio's macro check because we need to fallback on OS X < 10.12
+const MACOS_KERNEL_MAJOR_VERSION_MOJAVE: u32 = 18;
+// thanking the cubeb-coreaudio guy again
+
+#[derive(Debug, PartialOrd, PartialEq)]
+enum ParseMacOSKernelVersionError {
+    SysCtl,
+    Malformed,
+    Parsing,
+}
+
+//rip cubeb-audio's macro check because we need to fallback on OS X < 10.12
+fn macos_kernel_major_version() -> std::result::Result<u32, ParseMacOSKernelVersionError> {
+    let ver = whatsys::kernel_version();
+    if ver.is_none() {
+        return Err(ParseMacOSKernelVersionError::SysCtl);
+    }
+    let ver = ver.unwrap();
+    let major = ver.split('.').next();
+    if major.is_none() {
+        return Err(ParseMacOSKernelVersionError::Malformed);
+    }
+    let parsed_major = u32::from_str(major.unwrap());
+    if parsed_major.is_err() {
+        return Err(ParseMacOSKernelVersionError::Parsing);
+    }
+    Ok(parsed_major.unwrap())
+}
+
+
 #[repr(C)]
 pub struct __SecIdentity(c_void);
 pub type SecIdentityRef = *const __SecIdentity;
 declare_TCFType!(SecIdentity, SecIdentityRef);
 impl_TCFType!(SecIdentity, SecIdentityRef, SecIdentityGetTypeID);
 
+
 /// Safety: strictly speaking, it isn't safe to send `SecIdentity` across threads. The
 /// implementation handles this by wrapping `SecIdentity` in `ThreadSpecificHandles`. However, in
 /// order to implement `Drop` for `ThreadSpecificHandles`, the `SecIdentity` it holds must be sent
@@ -71,12 +104,29 @@ pub type SecTrustRef = *const __SecTrust;
 declare_TCFType!(SecTrust, SecTrustRef);
 impl_TCFType!(SecTrust, SecTrustRef, SecTrustGetTypeID);
 
+type SecKeyCreateSignatureType =
+    unsafe extern "C" fn(SecKeyRef, SecKeyAlgorithm, CFDataRef, *mut CFErrorRef) -> CFDataRef;
+type SecKeyCopyAttributesType = unsafe extern "C" fn(SecKeyRef) -> CFDictionaryRef;
+type SecKeyCopyExternalRepresentationType =
+    unsafe extern "C" fn(SecKeyRef, *mut CFErrorRef) -> CFDataRef;
 type SecCertificateCopyKeyType = unsafe extern "C" fn(SecCertificateRef) -> SecKeyRef;
 type SecTrustEvaluateWithErrorType =
     unsafe extern "C" fn(trust: SecTrustRef, error: *mut CFErrorRef) -> bool;
-
+type SecTrustEvaluateType =
+    unsafe extern "C" fn(trust: SecTrustRef, error: *mut u32) -> OSStatus;
 #[derive(Ord, Eq, PartialOrd, PartialEq)]
 enum SecStringConstant {
+    // These are available in macOS 10.12
+    SecKeyAlgorithmECDSASignatureDigestX962SHA1,
+    SecKeyAlgorithmECDSASignatureDigestX962SHA256,
+    SecKeyAlgorithmECDSASignatureDigestX962SHA384,
+    SecKeyAlgorithmECDSASignatureDigestX962SHA512,
+    SecKeyAlgorithmRSASignatureDigestPKCS1v15Raw,
+    SecKeyAlgorithmRSASignatureDigestPKCS1v15SHA256,
+    SecKeyAlgorithmRSASignatureDigestPKCS1v15SHA384,
+    SecKeyAlgorithmRSASignatureDigestPKCS1v15SHA512,
+    SecKeyAlgorithmRSASignatureDigestPKCS1v15SHA1,
+    SecAttrKeyTypeECSECPrimeRandom,
     // These are available in macOS 10.13
     SecKeyAlgorithmRSASignatureDigestPSSSHA1,
     SecKeyAlgorithmRSASignatureDigestPSSSHA256,
@@ -91,8 +141,12 @@ enum SecStringConstant {
 /// enough version of macOS, they will not be able to use client certificates
 /// from their keychain in Firefox until they upgrade.
 struct SecurityFramework<'a> {
+    sec_key_create_signature: Symbol<'a, SecKeyCreateSignatureType>,
+    sec_key_copy_attributes: Symbol<'a, SecKeyCopyAttributesType>,
+    sec_key_copy_external_representation: Symbol<'a, SecKeyCopyExternalRepresentationType>,
     sec_certificate_copy_key: Symbol<'a, SecCertificateCopyKeyType>,
     sec_trust_evaluate_with_error: Symbol<'a, SecTrustEvaluateWithErrorType>,
+    sec_trust_evaluate: Symbol<'a, SecTrustEvaluateType>,
     sec_string_constants: BTreeMap<SecStringConstant, String>,
 }
 
@@ -109,6 +163,21 @@ impl<'a> SecurityFramework<'a> {
             Ok(library) => library,
             Err(e) => return Err(error_here!(ErrorType::ExternalError, e.clone())),
         };
+        let sec_key_create_signature = unsafe {
+            library
+                .get::<SecKeyCreateSignatureType>(b"SecKeyCreateSignature\0")
+                .map_err(|e| error_here!(ErrorType::ExternalError, e.to_string()))?
+        };
+        let sec_key_copy_attributes = unsafe {
+            library
+                .get::<SecKeyCopyAttributesType>(b"SecKeyCopyAttributes\0")
+                .map_err(|e| error_here!(ErrorType::ExternalError, e.to_string()))?
+        };
+        let sec_key_copy_external_representation = unsafe {
+            library
+                .get::<SecKeyCopyExternalRepresentationType>(b"SecKeyCopyExternalRepresentation\0")
+                .map_err(|e| error_here!(ErrorType::ExternalError, e.to_string()))?
+        };
         let sec_certificate_copy_key = unsafe {
             library
                 .get::<SecCertificateCopyKeyType>(b"SecCertificateCopyKey\0")
@@ -119,8 +188,49 @@ impl<'a> SecurityFramework<'a> {
                 .get::<SecTrustEvaluateWithErrorType>(b"SecTrustEvaluateWithError\0")
                 .map_err(|e| error_here!(ErrorType::ExternalError, e.to_string()))?
         };
+        let sec_trust_evaluate = unsafe {
+            library
+                .get::<SecTrustEvaluateType>(b"SecTrustEvaluate\0")
+                .map_err(|e| error_here!(ErrorType::ExternalError, e.to_string()))?
+        };
         let mut sec_string_constants = BTreeMap::new();
         let strings_to_load = vec![
+            (
+                b"kSecKeyAlgorithmECDSASignatureDigestX962SHA1\0".as_ref(),
+                SecStringConstant::SecKeyAlgorithmECDSASignatureDigestX962SHA1,
+            ),
+            (
+                b"kSecKeyAlgorithmECDSASignatureDigestX962SHA256\0".as_ref(),
+                SecStringConstant::SecKeyAlgorithmECDSASignatureDigestX962SHA256,
+            ),
+            (
+                b"kSecKeyAlgorithmECDSASignatureDigestX962SHA384\0".as_ref(),
+                SecStringConstant::SecKeyAlgorithmECDSASignatureDigestX962SHA384,
+            ),
+            (
+                b"kSecKeyAlgorithmECDSASignatureDigestX962SHA512\0".as_ref(),
+                SecStringConstant::SecKeyAlgorithmECDSASignatureDigestX962SHA512,
+            ),
+            (
+                b"kSecKeyAlgorithmRSASignatureDigestPKCS1v15Raw\0".as_ref(),
+                SecStringConstant::SecKeyAlgorithmRSASignatureDigestPKCS1v15Raw,
+            ),
+            (
+                b"kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA256\0".as_ref(),
+                SecStringConstant::SecKeyAlgorithmRSASignatureDigestPKCS1v15SHA256,
+            ),
+            (
+                b"kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA384\0".as_ref(),
+                SecStringConstant::SecKeyAlgorithmRSASignatureDigestPKCS1v15SHA384,
+            ),
+            (
+                b"kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA512\0".as_ref(),
+                SecStringConstant::SecKeyAlgorithmRSASignatureDigestPKCS1v15SHA512,
+            ),
+            (
+                b"kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA1\0".as_ref(),
+                SecStringConstant::SecKeyAlgorithmRSASignatureDigestPKCS1v15SHA1,
+            ),
             (
                 b"kSecKeyAlgorithmRSASignatureDigestPSSSHA1\0".as_ref(),
                 SecStringConstant::SecKeyAlgorithmRSASignatureDigestPSSSHA1,
@@ -137,6 +247,10 @@ impl<'a> SecurityFramework<'a> {
                 b"kSecKeyAlgorithmRSASignatureDigestPSSSHA512\0".as_ref(),
                 SecStringConstant::SecKeyAlgorithmRSASignatureDigestPSSSHA512,
             ),
+            (
+                b"kSecAttrKeyTypeECSECPrimeRandom\0".as_ref(),
+                SecStringConstant::SecAttrKeyTypeECSECPrimeRandom,
+            ),
         ];
         for (symbol_name, sec_string_constant) in strings_to_load {
             let cfstring_symbol = unsafe {
@@ -148,8 +262,12 @@ impl<'a> SecurityFramework<'a> {
             sec_string_constants.insert(sec_string_constant, cfstring.to_string());
         }
         Ok(SecurityFramework {
+            sec_key_create_signature,
+            sec_key_copy_attributes,
+            sec_key_copy_external_representation,
             sec_certificate_copy_key,
             sec_trust_evaluate_with_error,
+            sec_trust_evaluate,
             sec_string_constants,
         })
     }
@@ -166,6 +284,71 @@ impl<'a> SecurityFrameworkHolder<'a> {
         }
     }
 
+    /// SecKeyCreateSignature is available in macOS 10.12
+    fn sec_key_create_signature(
+        &self,
+        key: &SecKey,
+        algorithm: SecKeyAlgorithm,
+        data_to_sign: &CFData,
+    ) -> Result<CFData, Error> {
+        match &self.framework {
+            Ok(framework) => unsafe {
+                let mut error = std::ptr::null_mut();
+                let result = (framework.sec_key_create_signature)(
+                    key.as_concrete_TypeRef(),
+                    algorithm,
+                    data_to_sign.as_concrete_TypeRef(),
+                    &mut error,
+                );
+                if result.is_null() {
+                    let error = CFError::wrap_under_create_rule(error);
+                    return Err(error_here!(
+                        ErrorType::ExternalError,
+                        error.description().to_string()
+                    ));
+                }
+                Ok(CFData::wrap_under_create_rule(result))
+            },
+            Err(e) => Err(e.clone()),
+        }
+    }
+
+    /// SecKeyCopyAttributes is available in macOS 10.12
+    fn sec_key_copy_attributes<T>(&self, key: &SecKey) -> Result<CFDictionary<CFString, T>, Error> {
+        match &self.framework {
+            Ok(framework) => unsafe {
+                let result = (framework.sec_key_copy_attributes)(key.as_concrete_TypeRef());
+                if result.is_null() {
+                    return Err(error_here!(ErrorType::ExternalError));
+                }
+                Ok(CFDictionary::wrap_under_create_rule(result))
+            },
+            Err(e) => Err(e.clone()),
+        }
+    }
+
+    /// SecKeyCopyExternalRepresentation is available in macOS 10.12
+    fn sec_key_copy_external_representation(&self, key: &SecKey) -> Result<CFData, Error> {
+        match &self.framework {
+            Ok(framework) => unsafe {
+                let mut error = std::ptr::null_mut();
+                let result = (framework.sec_key_copy_external_representation)(
+                    key.as_concrete_TypeRef(),
+                    &mut error,
+                );
+                if result.is_null() {
+                    let error = CFError::wrap_under_create_rule(error);
+                    return Err(error_here!(
+                        ErrorType::ExternalError,
+                        error.description().to_string()
+                    ));
+                }
+                Ok(CFData::wrap_under_create_rule(result))
+            },
+            Err(e) => Err(e.clone()),
+        }
+    }
+
     /// SecCertificateCopyKey is available in macOS 10.14
     fn sec_certificate_copy_key(&self, certificate: &SecCertificate) -> Result<SecKey, Error> {
         match &self.framework {
@@ -193,7 +376,18 @@ impl<'a> SecurityFrameworkHolder<'a> {
             Err(e) => Err(e.clone()),
         }
     }
-
+    //fallback to deprecated api
+    fn sec_trust_evaluate(&self, trust: &SecTrust) -> Result<OSStatus, Error> {
+        match &self.framework {
+            Ok(framework) => unsafe {
+                Ok((framework.sec_trust_evaluate)(
+                    trust.as_concrete_TypeRef(),
+                    std::ptr::null_mut(),
+                ))
+            },
+            Err(e) => Err(e.clone()),
+        }
+    }
     fn get_sec_string_constant(
         &self,
         sec_string_constant: SecStringConstant,
@@ -213,58 +407,6 @@ lazy_static! {
         SecurityFrameworkHolder::new();
 }
 
-fn sec_key_create_signature(
-    key: &SecKey,
-    algorithm: SecKeyAlgorithm,
-    data: &CFData,
-) -> Result<CFData, Error> {
-    let mut error = std::ptr::null_mut();
-    let signature = unsafe {
-        SecKeyCreateSignature(
-            key.as_concrete_TypeRef(),
-            algorithm,
-            data.as_concrete_TypeRef(),
-            &mut error,
-        )
-    };
-    if signature.is_null() {
-        if error.is_null() {
-            return Err(error_here!(ErrorType::ExternalError));
-        }
-        let error = unsafe { CFError::wrap_under_create_rule(error) };
-        return Err(error_here!(
-            ErrorType::ExternalError,
-            error.description().to_string()
-        ));
-    }
-    Ok(unsafe { CFData::wrap_under_create_rule(signature) })
-}
-
-fn sec_key_copy_attributes<T: TCFType>(key: &SecKey) -> Result<CFDictionary<CFString, T>, Error> {
-    let attributes = unsafe { SecKeyCopyAttributes(key.as_concrete_TypeRef()) };
-    if attributes.is_null() {
-        return Err(error_here!(ErrorType::ExternalError));
-    }
-    Ok(unsafe { CFDictionary::wrap_under_create_rule(attributes) })
-}
-
-fn sec_key_copy_external_representation(key: &SecKey) -> Result<CFData, Error> {
-    let mut error = std::ptr::null_mut();
-    let representation =
-        unsafe { SecKeyCopyExternalRepresentation(key.as_concrete_TypeRef(), &mut error) };
-    if representation.is_null() {
-        if error.is_null() {
-            return Err(error_here!(ErrorType::ExternalError));
-        }
-        let error = unsafe { CFError::wrap_under_create_rule(error) };
-        return Err(error_here!(
-            ErrorType::ExternalError,
-            error.description().to_string()
-        ));
-    }
-    Ok(unsafe { CFData::wrap_under_create_rule(representation) })
-}
-
 fn sec_identity_copy_certificate(identity: &SecIdentity) -> Result<SecCertificate, Error> {
     let mut certificate = std::ptr::null();
     let status =
@@ -336,17 +478,16 @@ impl<'a> SignParams<'a> {
     }
 
     fn new_ec_params(data: &'a [u8]) -> Result<SignParams<'a>, Error> {
-        let algorithm = unsafe {
-            CFString::wrap_under_get_rule(match data.len() {
-                20 => kSecKeyAlgorithmECDSASignatureDigestX962SHA1,
-                32 => kSecKeyAlgorithmECDSASignatureDigestX962SHA256,
-                48 => kSecKeyAlgorithmECDSASignatureDigestX962SHA384,
-                64 => kSecKeyAlgorithmECDSASignatureDigestX962SHA512,
-                _ => {
-                    return Err(error_here!(ErrorType::UnsupportedInput));
-                }
-            })
+        let algorithm_id = match data.len() {
+            20 => SecStringConstant::SecKeyAlgorithmECDSASignatureDigestX962SHA1,
+            32 => SecStringConstant::SecKeyAlgorithmECDSASignatureDigestX962SHA256,
+            48 => SecStringConstant::SecKeyAlgorithmECDSASignatureDigestX962SHA384,
+            64 => SecStringConstant::SecKeyAlgorithmECDSASignatureDigestX962SHA512,
+            _ => {
+                return Err(error_here!(ErrorType::UnsupportedInput));
+            }
         };
+        let algorithm = SECURITY_FRAMEWORK.get_sec_string_constant(algorithm_id)?;
         Ok(SignParams::EC(algorithm, data))
     }
 
@@ -370,32 +511,29 @@ impl<'a> SignParams<'a> {
             return Ok(SignParams::RSA(algorithm, data));
         }
 
-        // Handle the case where `data` is a DigestInfo.
-        if let Ok((digest_oid, hash)) = read_digest_info(data) {
-            let algorithm = unsafe {
-                CFString::wrap_under_create_rule(match digest_oid {
-                    OID_BYTES_SHA_256 => kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA256,
-                    OID_BYTES_SHA_384 => kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA384,
-                    OID_BYTES_SHA_512 => kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA512,
-                    OID_BYTES_SHA_1 => kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA1,
-                    _ => return Err(error_here!(ErrorType::UnsupportedInput)),
-                })
-            };
-            return Ok(SignParams::RSA(algorithm, hash));
-        }
-
-        // Handle the case where `data` is a TLS 1.0 MD5/SHA1 hash.
+        // Handle the case where this is a TLS 1.0 MD5/SHA1 hash.
         if data.len() == 36 {
-            let algorithm = unsafe {
-                CFString::wrap_under_get_rule(kSecKeyAlgorithmRSASignatureDigestPKCS1v15Raw)
-            };
+            let algorithm_id = SecStringConstant::SecKeyAlgorithmRSASignatureDigestPKCS1v15Raw;
+            let algorithm = SECURITY_FRAMEWORK.get_sec_string_constant(algorithm_id)?;
             return Ok(SignParams::RSA(algorithm, data));
         }
+        // Otherwise, `data` should be a DigestInfo.
+        let (digest_oid, hash) = read_digest_info(data)?;
+        let algorithm_id = if digest_oid == OID_BYTES_SHA_256 {
+            SecStringConstant::SecKeyAlgorithmRSASignatureDigestPKCS1v15SHA256
+        } else if digest_oid == OID_BYTES_SHA_384 {
+            SecStringConstant::SecKeyAlgorithmRSASignatureDigestPKCS1v15SHA384
+        } else if digest_oid == OID_BYTES_SHA_512 {
+            SecStringConstant::SecKeyAlgorithmRSASignatureDigestPKCS1v15SHA512
+        } else if digest_oid == OID_BYTES_SHA_1 {
+            SecStringConstant::SecKeyAlgorithmRSASignatureDigestPKCS1v15SHA1
+        } else {
+            return Err(error_here!(ErrorType::UnsupportedInput));
+        };
 
-        // Otherwise, `data` will be an emsa-pss-encoded digest that should be signed with raw RSA.
         Ok(SignParams::RSA(
-            unsafe { CFString::wrap_under_create_rule(kSecKeyAlgorithmRSASignatureRaw) },
-            data,
+            SECURITY_FRAMEWORK.get_sec_string_constant(algorithm_id)?,
+            hash,
         ))
     }
 
@@ -518,7 +656,7 @@ fn sign_internal(
     let sign_params = SignParams::new(key_type, data, params)?;
     let signing_algorithm = sign_params.get_algorithm();
     let data_to_sign = CFData::from_buffer(sign_params.get_data_to_sign());
-    let signature = sec_key_create_signature(key, signing_algorithm, &data_to_sign)?;
+    let signature = SECURITY_FRAMEWORK.sec_key_create_signature(key, signing_algorithm, &data_to_sign)?;
     let signature_value = match key_type {
         KeyType::EC(coordinate_width) => {
             // We need to convert the DER Ecdsa-Sig-Value to the
@@ -554,7 +692,6 @@ impl Drop for ThreadSpecificHandles {
         }
     }
 }
-
 pub struct Key {
     handles: ThreadSpecificHandles,
     cryptoki_key: CryptokiKey,
@@ -567,11 +704,11 @@ impl Key {
         let key = SECURITY_FRAMEWORK.sec_certificate_copy_key(&certificate)?;
         let key_type: CFString = get_key_attribute(&key, unsafe { kSecAttrKeyType })?;
         let key_size_in_bits: CFNumber = get_key_attribute(&key, unsafe { kSecAttrKeySizeInBits })?;
-        let sec_attr_key_type_ec =
-            unsafe { CFString::wrap_under_create_rule(kSecAttrKeyTypeECSECPrimeRandom) };
+        let sec_attr_key_type_ec = SECURITY_FRAMEWORK
+            .get_sec_string_constant(SecStringConstant::SecAttrKeyTypeECSECPrimeRandom)?;
         let (modulus, ec_params) =
             if key_type.as_concrete_TypeRef() == unsafe { kSecAttrKeyTypeRSA } {
-                let public_key = sec_key_copy_external_representation(&key)?;
+                let public_key = SECURITY_FRAMEWORK.sec_key_copy_external_representation(&key)?;
                 let modulus = read_rsa_modulus(public_key.bytes())?;
                 (Some(modulus), None)
             } else if key_type == sec_attr_key_type_ec {
@@ -600,6 +737,7 @@ impl Key {
             cryptoki_key: CryptokiKey::new(modulus, ec_params, der.bytes())?,
         })
     }
+
 }
 
 impl CryptokiObject for Key {
@@ -640,7 +778,7 @@ impl Sign for Key {
 }
 
 fn get_key_attribute<T: TCFType + Clone>(key: &SecKey, attr: CFStringRef) -> Result<T, Error> {
-    let attributes: CFDictionary<CFString, T> = sec_key_copy_attributes(key)?;
+    let attributes: CFDictionary<CFString, T> = SECURITY_FRAMEWORK.sec_key_copy_attributes(key)?;
     match attributes.find(attr as *const _) {
         Some(value) => Ok((*value).clone()),
         None => Err(error_here!(ErrorType::ExternalError)),
@@ -683,7 +821,12 @@ fn get_issuers(identity: &SecIdentity) -> Result<Vec<SecCertificate>, Error> {
     }
     // We ignore the return value here because we don't care if the certificate is trusted or not -
     // we're only doing this to build its issuer chain as much as possible.
-    let _ = SECURITY_FRAMEWORK.sec_trust_evaluate_with_error(&trust)?;
+    
+    if macos_kernel_major_version() >= Ok(MACOS_KERNEL_MAJOR_VERSION_MOJAVE) {
+        let _ = SECURITY_FRAMEWORK.sec_trust_evaluate_with_error(&trust)?;
+    } else {
+        let _ = SECURITY_FRAMEWORK.sec_trust_evaluate(&trust)?;
+    }
     let certificate_count = unsafe { SecTrustGetCertificateCount(trust.as_concrete_TypeRef()) };
     let mut certificates = Vec::with_capacity(
         certificate_count
@@ -824,5 +967,5 @@ fn find_objects(thread: &nsIEventTarget) -> Result<(Vec<CryptokiCert>, Vec<Key>)
             }
         }
     }
-    Ok((certs, keys))
+        Ok((certs, keys))
 }
