diff --git a/third_party/libwebrtc/sdk/objc/components/capturer/RTCCameraVideoCapturer.h b/third_party/libwebrtc/sdk/objc/components/capturer/RTCCameraVideoCapturer.h
index 6aeb577b13804..e6b1e274327ff 100644
--- a/third_party/libwebrtc/sdk/objc/components/capturer/RTCCameraVideoCapturer.h
+++ b/third_party/libwebrtc/sdk/objc/components/capturer/RTCCameraVideoCapturer.h
@@ -24,12 +24,9 @@ RTC_OBJC_EXPORT
 
 // Capture session that is used for capturing. Valid from initialization to dealloc.
 @property(readonly, nonatomic) AVCaptureSession *captureSession;
-
 // Returns list of available capture devices that support video capture.
-+ (NSArray<AVCaptureDevice *> *)captureDevicesWithDeviceTypes:
-    (NSArray<AVCaptureDeviceType> *)deviceTypes;
-// Returns list of default capture devices types
-+ (NSArray<AVCaptureDeviceType> *)defaultCaptureDeviceTypes;
++ (NSArray<AVCaptureDevice *> *)captureDevices;
+
 // Returns list of formats that are supported by this class for this device.
 + (NSArray<AVCaptureDeviceFormat *> *)supportedFormatsForDevice:
     (AVCaptureDevice *)device;
