diff --git a/widget/InitData.h b/widget/InitData.h
index a42021216ae7a..76dddd317584a 100644
--- a/widget/InitData.h
+++ b/widget/InitData.h
@@ -83,7 +83,7 @@ enum class PiPType : uint8_t {
 // Basic struct for widget initialization data.
 // @see Create member function of nsIWidget
 struct InitData {
-  WindowType mWindowType = WindowType::TopLevel;
+  WindowType mWindowType = WindowType::TopLevel; 
   BorderStyle mBorderStyle = BorderStyle::Default;
   PopupType mPopupHint = PopupType::Panel;
   PopupLevel mPopupLevel = PopupLevel::Top;
