diff --git a/widget/cocoa/nsMenuX.mm b/widget/cocoa/nsMenuX.mm
index ec637a0fd0402..5b58658cbdcef 100644
--- a/widget/cocoa/nsMenuX.mm
+++ b/widget/cocoa/nsMenuX.mm
@@ -797,9 +797,14 @@ static NSUserInterfaceLayoutDirection DirectionForElement(
     return;
   }
 
-  if (menuPopup->IsElement()) {
-    mNativeMenu.userInterfaceLayoutDirection =
-        DirectionForElement(menuPopup->AsElement());
+
+  //markus added these around milestone 90a1, so 
+  //anything beneath sierra is not going to handle these
+  if(@available(macOS 10.12, *)) {
+    if (menuPopup->IsElement()) {
+      mNativeMenu.userInterfaceLayoutDirection =
+          DirectionForElement(menuPopup->AsElement());
+    }
   }
 
   // Iterate over the kids
