diff --git a/widget/nsNativeTheme.cpp b/widget/nsNativeTheme.cpp
index 74d9e048055d8..828a2cce7b843 100644
--- a/widget/nsNativeTheme.cpp
+++ b/widget/nsNativeTheme.cpp
@@ -142,6 +142,16 @@ bool nsNativeTheme::IsButtonTypeMenu(nsIFrame* aFrame) {
                                            u"menu"_ns, eCaseMatters);
 }
 
+bool nsNativeTheme::IsPressedButton(nsIFrame* aFrame) {
+  ElementState state = GetContentState(aFrame, StyleAppearance::Toolbarbutton);
+  if (state.HasState(ElementState::DISABLED)) {
+    return false;
+  }
+
+  return IsOpenButton(aFrame) ||
+         state.HasAllStates(ElementState::ACTIVE | ElementState::HOVER);
+}
+
 bool nsNativeTheme::IsWidgetStyled(nsPresContext* aPresContext,
                                    nsIFrame* aFrame,
                                    StyleAppearance aAppearance) {
@@ -218,6 +228,30 @@ bool nsNativeTheme::IsVerticalMeter(nsIFrame* aFrame) {
   return false;
 }
 
+// menupopup:
+bool nsNativeTheme::IsSubmenu(nsIFrame* aFrame, bool* aLeftOfParent) {
+  if (!aFrame) return false;
+
+  nsIContent* parentContent = aFrame->GetContent()->GetParent();
+  if (!parentContent || !parentContent->IsXULElement(nsGkAtoms::menu))
+    return false;
+
+  nsIFrame* parent = aFrame;
+  while ((parent = parent->GetParent())) {
+    if (parent->GetContent() == parentContent) {
+      if (aLeftOfParent) {
+        LayoutDeviceIntRect selfBounds, parentBounds;
+        selfBounds = aFrame->GetNearestWidget()->GetScreenBounds();
+        parentBounds = parent->GetNearestWidget()->GetScreenBounds();
+        *aLeftOfParent = selfBounds.X() < parentBounds.X();
+      }
+      return true;
+    }
+  }
+
+  return false;
+}
+
 bool nsNativeTheme::QueueAnimatedContentForRefresh(nsIContent* aContent,
                                                    uint32_t aMinimumFrameRate) {
   NS_ASSERTION(aContent, "Null pointer!");
