From 2fbf9abe648084066941477000bf297ed0805bd6 Mon Sep 17 00:00:00 2001
From: aobaharuki2005 <tranbaohnth@outlook.com>
Date: Mon, 20 Apr 2026 00:23:44 +0700
Subject: [PATCH 1/2] shim sincospi and sincospif in TextRecognition.mm

---
 widget/cocoa/TextRecognition.mm | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/widget/cocoa/TextRecognition.mm b/widget/cocoa/TextRecognition.mm
index 31a170466a..e74bf23ec2 100644
--- a/widget/cocoa/TextRecognition.mm
+++ b/widget/cocoa/TextRecognition.mm
@@ -2,6 +2,14 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#if !defined(MAC_OS_VERSION_10_9) || MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_VERSION_10_9
+#include <math.h>
+extern "C" {
+  void __sincospif(float __x, float *__sinp, float *__cosp);
+  void __sincospi(double __x, double *__sinp, double *__cosp);
+} 
+#endif // !defined(MAC_OS_VERSION_10_9) || MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_VERSION_10_9
+
 #import <Vision/Vision.h>
 
 #include "mozilla/dom/Promise.h"
-- 
2.53.0

