diff --git a/accessible/mac/MOXTextMarkerDelegate.mm b/accessible/mac/MOXTextMarkerDelegate.mm
index a2f6faf020ce..efbf4ded7cef 100644
--- a/accessible/mac/MOXTextMarkerDelegate.mm
+++ b/accessible/mac/MOXTextMarkerDelegate.mm
@@ -142,7 +142,7 @@ mozAccessible* GetEditableNativeFromGeckoAccessible(Accessible* aAcc) {
   if (!caretEditable && stateChangeType == AXTextStateChangeTypeSelectionMove) {
     // If we are not in an editable, VO expects AXTextStateSync to be present
     // and true.
-    info[@"AXTextStateSync"] = @YES;
+    [info setObject:@YES forKey:@"AXTextStateSync"];
   }
 
   if (!prevCaretMarker.IsValid() || caretMarker == prevCaretMarker) {
@@ -156,12 +156,12 @@ mozAccessible* GetEditableNativeFromGeckoAccessible(Accessible* aAcc) {
   if (prevCaretEditable != caretEditable) {
     // If the caret goes in or out of an editable, consider the
     // move direction "discontiguous".
-    info[@"AXTextSelectionDirection"] =
-        @(AXTextSelectionDirectionDiscontiguous);
+    [info setObject:@(AXTextSelectionDirectionDiscontiguous) forKey:@"AXTextSelectionDirection"];
+
     if ([[caretEditable moxFocused] boolValue]) {
       // If the caret is in a new focused editable, VO expects this attribute to
       // be present and to be true.
-      info[@"AXTextSelectionChangedFocus"] = @YES;
+      [info setObject:@YES forKey:@"AXTextSelectionChangedFocus"];
     }
 
     return info;
@@ -271,9 +271,9 @@ mozAccessible* GetEditableNativeFromGeckoAccessible(Accessible* aAcc) {
   }
 
   GeckoTextMarker p1 = GeckoTextMarker::MarkerFromAXTextMarker(
-      mGeckoDocAccessible, (__bridge AXTextMarkerRef)textMarkers[0]);
+      mGeckoDocAccessible, (__bridge AXTextMarkerRef)[textMarkers objectAtIndex:0]);
   GeckoTextMarker p2 = GeckoTextMarker::MarkerFromAXTextMarker(
-      mGeckoDocAccessible, (__bridge AXTextMarkerRef)textMarkers[1]);
+      mGeckoDocAccessible, (__bridge AXTextMarkerRef)[textMarkers objectAtIndex:1]);
 
   if (!p1.IsValid() || !p2.IsValid()) {
     // If either marker is invalid, return nil.
