diff --git a/accessible/mac/mozAccessible.mm b/accessible/mac/mozAccessible.mm
index e25d94904ed4..618123243623 100644
--- a/accessible/mac/mozAccessible.mm
+++ b/accessible/mac/mozAccessible.mm
@@ -1038,7 +1038,7 @@ static bool ProvidesTitle(const Accessible* aAccessible, nsString& aName) {
           static_cast<MOXTextMarkerDelegate*>([self moxTextMarkerDelegate]);
       NSMutableDictionary* userInfo =
           [[[delegate selectionChangeInfo] mutableCopy] autorelease];
-      userInfo[@"AXTextChangeElement"] = self;
+      [userInfo setObject:self forKey:@"AXTextChangeElement"];
 
       mozAccessible* webArea = [self topWebArea];
       [webArea
