diff --git a/config/external/rlbox_wasm2c_sandbox/rlbox_wasm2c_thread_locals.cpp b/config/external/rlbox_wasm2c_sandbox/rlbox_wasm2c_thread_locals.cpp
index f4be9c6c4448..a754a23e73f0 100644
--- a/config/external/rlbox_wasm2c_sandbox/rlbox_wasm2c_thread_locals.cpp
+++ b/config/external/rlbox_wasm2c_sandbox/rlbox_wasm2c_thread_locals.cpp
@@ -15,9 +15,7 @@
 #  include "mozilla/rlbox/rlbox_wasm2c_tls.hpp"
 #  include "wasm-rt.h"
 
-#  ifndef WASM_RT_GROW_FAILED_CRASH
 #    include "nsExceptionHandler.h"
-#  endif
 
 // The MingW compiler does not correctly handle static thread_local inline
 // members. We instead TLS storage via functions. This can be removed if the
@@ -35,12 +33,8 @@ void moz_wasm2c_trap_handler(wasm_rt_trap_t code) {
 // The below function is called if a malloc in sandboxed code returns null
 // This indicates that the sandbox has run out of memory.
 void moz_wasm2c_memgrow_failed() {
-#  ifdef WASM_RT_GROW_FAILED_CRASH
-  MOZ_CRASH("wasm2c memory grow failed");
-#  else
   CrashReporter::RecordAnnotationBool(
       CrashReporter::Annotation::WasmLibrarySandboxMallocFailed, true);
-#  endif
 }
 
 // This function is called when mozalloc_handle_oom is called from within
