diff --git a/dom/media/moz.build b/dom/media/moz.build
index 391b7200a14f..8c3d7dbd3037 100644
--- a/dom/media/moz.build
+++ b/dom/media/moz.build
@@ -197,6 +197,8 @@ EXPORTS += [
     "PrincipalHandle.h",
     "QueueObject.h",
     "ReaderProxy.h",
+    "RLBoxSoundTouch.h",
+    "RLBoxSoundTouchTypes.h",
     "SeekJob.h",
     "SeekTarget.h",
     "SelfRef.h",
@@ -311,6 +313,7 @@ UNIFIED_SOURCES += [
     "MemoryBlockCache.cpp",
     "QueueObject.cpp",
     "ReaderProxy.cpp",
+    "RLBoxSoundTouch.cpp",
     "SeekJob.cpp",
     "TimeUnits.cpp",
     "Tracing.cpp",
@@ -358,7 +361,7 @@ EXTRA_JS_MODULES.media += [
 ]
 
 LOCAL_INCLUDES += [
-    "!/media/libsoundtouch/src",
+    "!/security/rlbox",
     "/caps",
     "/docshell/base",
     "/dom/base",
