diff --git a/dom/media/platforms/apple/AppleVTDecoder.h b/dom/media/platforms/apple/AppleVTDecoder.h
index 0d74fae6b6d8..ab52a74a862f 100644
--- a/dom/media/platforms/apple/AppleVTDecoder.h
+++ b/dom/media/platforms/apple/AppleVTDecoder.h
@@ -11,6 +11,7 @@
 #include <CoreMedia/CoreMedia.h>          // For CMVideoFormatDescriptionRef
 #include <VideoToolbox/VideoToolbox.h>    // For VTDecompressionSessionRef
 
+#include "AppleVDADecoder.h"
 #include "AppleDecoderModule.h"
 #include "PerformanceRecorder.h"
 #include "PlatformDecoderModule.h"
