diff --git a/dom/media/platforms/moz.build b/dom/media/platforms/moz.build
index 237cbc9f0444..81db4d807fb7 100644
--- a/dom/media/platforms/moz.build
+++ b/dom/media/platforms/moz.build
@@ -93,11 +93,16 @@ if CONFIG["MOZ_APPLEMEDIA"]:
     ]
     UNIFIED_SOURCES += [
         "apple/AppleATDecoder.cpp",
+        "apple/AppleCMLinker.cpp",
+        "apple/AppleCVLinker.cpp",
         "apple/AppleDecoderModule.cpp",
         "apple/AppleEncoderModule.cpp",
-        "apple/AppleUtils.cpp",
+        'apple/AppleUtils.cpp',
+        'apple/AppleVDADecoder.cpp',
+        'apple/AppleVDALinker.cpp',
         "apple/AppleVTDecoder.cpp",
         "apple/AppleVTEncoder.cpp",
+        "apple/AppleVTLinker.cpp",
     ]
     LOCAL_INCLUDES += [
         "/media/libyuv/libyuv/include",
@@ -106,6 +111,7 @@ if CONFIG["MOZ_APPLEMEDIA"]:
         "-framework AudioToolbox",
         "-framework CoreMedia",
         "-framework VideoToolbox",
+        "-framework VideoDecodeAcceleration",
         # For some unknown reason, the documented method of using weak_import
         # attribute doesn't work with VideoToolbox's functions.
         # We want to lazily load _VTRegisterSupplementalVideoDecoderIfAvailable
