diff --git a/dom/webauthn/MacOSWebAuthnService.mm b/dom/webauthn/MacOSWebAuthnService.mm
index 1c43967a5238..5f82592b0c13 100644
--- a/dom/webauthn/MacOSWebAuthnService.mm
+++ b/dom/webauthn/MacOSWebAuthnService.mm
@@ -1009,7 +1009,7 @@ MacOSWebAuthnService::GetAssertion(uint64_t aTransactionId,
                 credentials) {
           nsTArray<RefPtr<nsIWebAuthnAutoFillEntry>> autoFillEntries;
           for (NSUInteger i = 0; i < credentials.count; i++) {
-            const auto& credential = credentials[i];
+            const auto& credential = [credentials objectAtIndex:i];
             nsAutoString userName;
             nsCocoaUtils::GetStringForNSString(credential.name, userName);
             nsAutoString rpId;
