diff --git a/gfx/2d/NativeFontResourceMac.h b/gfx/2d/NativeFontResourceMac.h
index 97683a8a4f3a..e679472fcd0e 100644
--- a/gfx/2d/NativeFontResourceMac.h
+++ b/gfx/2d/NativeFontResourceMac.h
@@ -26,20 +26,15 @@ class NativeFontResourceMac final : public NativeFontResource {
       uint32_t aInstanceDataLength) final;
 
   ~NativeFontResourceMac() {
-    CFRelease(mFontDescRef);
     CFRelease(mFontRef);
   }
 
   static void RegisterMemoryReporter();
 
  private:
-  explicit NativeFontResourceMac(CTFontDescriptorRef aFontDescRef,
-                                 CGFontRef aFontRef, size_t aDataLength)
-      : NativeFontResource(aDataLength),
-        mFontDescRef(aFontDescRef),
-        mFontRef(aFontRef) {}
+  explicit NativeFontResourceMac(CGFontRef aFontRef, size_t aDataLength)
+      : NativeFontResource(aDataLength), mFontRef(aFontRef) {}
 
-  CTFontDescriptorRef mFontDescRef;
   CGFontRef mFontRef;
 };
 
