diff --git a/gfx/skia/skia/src/utils/mac/SkCTFontCreateExactCopy.cpp b/gfx/skia/skia/src/utils/mac/SkCTFontCreateExactCopy.cpp
index d99f9a9700e3..4f49a432d1e3 100644
--- a/gfx/skia/skia/src/utils/mac/SkCTFontCreateExactCopy.cpp
+++ b/gfx/skia/skia/src/utils/mac/SkCTFontCreateExactCopy.cpp
@@ -102,7 +102,8 @@ SkUniqueCFRef<CTFontRef> SkCTFontCreateExactCopy(CTFontRef baseFont, CGFloat tex
     // If we have a system font we need to use the CGFont APIs to avoid having the
     // underlying font change for us when using CTFontCreateCopyWithAttributes.
     CFDictionaryRef variations = nullptr;
-    if (IsInstalledFont(baseFont)) {
+    if (nsCocoaFeatures::OnSierraExactly() || 
+            (IsInstalledFont(baseFont) && nsCocoaFeatures::OnHighSierraOrLater())) {
         baseCGFont.reset(CTFontCopyGraphicsFont(baseFont, nullptr));
 
         // The last parameter (CTFontDescriptorRef attributes) *must* be nullptr.
