diff --git a/ipc/chromium/src/chrome/common/ipc_channel.cc b/ipc/chromium/src/chrome/common/ipc_channel.cc
index 9ecc883268bd..dcca3dc12f22 100644
--- a/ipc/chromium/src/chrome/common/ipc_channel.cc
+++ b/ipc/chromium/src/chrome/common/ipc_channel.cc
@@ -14,9 +14,12 @@
 #else
 #  include "chrome/common/ipc_channel_posix.h"
 #endif
+//it ain't workin. sorry nika
+/*
 #ifdef XP_DARWIN
 #  include "chrome/common/ipc_channel_mach.h"
 #endif
+*/
 
 namespace IPC {
 
@@ -37,7 +40,8 @@ already_AddRefed<Channel> Channel::Create(ChannelHandle pipe, Mode mode,
                                                 other_pid);
 #endif
   }
-#if XP_DARWIN
+//it ain't workin. sorry nika
+/*#if XP_DARWIN
   if (auto* receive = std::get_if<mozilla::UniqueMachReceiveRight>(&pipe)) {
     return mozilla::MakeAndAddRef<ChannelMach>(std::move(*receive), nullptr,
                                                mode, other_pid);
@@ -47,6 +51,7 @@ already_AddRefed<Channel> Channel::Create(ChannelHandle pipe, Mode mode,
                                                other_pid);
   }
 #endif
+*/
   MOZ_ASSERT_UNREACHABLE("unhandled pipe type");
   return nullptr;
 }
