diff --git a/ipc/chromium/src/chrome/common/ipc_channel_posix.cc b/ipc/chromium/src/chrome/common/ipc_channel_posix.cc
index ff65fadad3cb..6324c90e6677 100644
--- a/ipc/chromium/src/chrome/common/ipc_channel_posix.cc
+++ b/ipc/chromium/src/chrome/common/ipc_channel_posix.cc
@@ -1086,12 +1086,14 @@ bool ChannelPosix::AcceptMachPorts(Message& msg) {
 // required information for AcceptMachPorts to the message footer. See comment
 // above for details.
 bool ChannelPosix::TransferMachPorts(Message& msg) {
+//it ain't working sorry nika
+/*
   uint32_t num_receive_rights = msg.num_receive_rights();
   if (num_receive_rights != 0) {
     CHROMIUM_LOG(ERROR) << "ChannelPosix does not support receive rights";
     return false;
   }
-
+*/
   uint32_t num_send_rights = msg.num_send_rights();
   if (num_send_rights == 0) {
     return true;
