diff --git a/ipc/chromium/src/chrome/common/ipc_message.cc b/ipc/chromium/src/chrome/common/ipc_message.cc
index 20e34691569f..4f3769894549 100644
--- a/ipc/chromium/src/chrome/common/ipc_message.cc
+++ b/ipc/chromium/src/chrome/common/ipc_message.cc
@@ -188,6 +188,8 @@ uint32_t Message::num_send_rights() const {
   return attached_send_rights_.Length();
 }
 
+//it ain't working sorry nika
+/*
 bool Message::WriteMachReceiveRight(mozilla::UniqueMachReceiveRight port) {
   uint32_t index = attached_receive_rights_.Length();
   WriteUInt32(index);
@@ -216,6 +218,7 @@ bool Message::ConsumeMachReceiveRight(
 uint32_t Message::num_receive_rights() const {
   return attached_receive_rights_.Length();
 }
+*/
 #endif
 
 bool Message::WillBeRoutedExternally(
