diff --git a/security/manager/ssl/osclientcerts/Cargo.toml b/security/manager/ssl/osclientcerts/Cargo.toml
index 94489aed2c1f..d71fc1dea4f4 100644
--- a/security/manager/ssl/osclientcerts/Cargo.toml
+++ b/security/manager/ssl/osclientcerts/Cargo.toml
@@ -19,6 +19,8 @@ pkcs11-bindings = "0.1"
 rsclientcerts = { path = "../rsclientcerts" }
 xpcom = { path = "../../../../xpcom/rust/xpcom" }
 
+whatsys = "0.3"
+
 [target."cfg(any(target_os = \"macos\", target_os = \"ios\"))".dependencies.core-foundation]
 version = "0.9"
 
