diff --git a/security/sandbox/common/test/SandboxTestingChildTests.h b/security/sandbox/common/test/SandboxTestingChildTests.h
index 565d052e7351..fc1237a731ec 100644
--- a/security/sandbox/common/test/SandboxTestingChildTests.h
+++ b/security/sandbox/common/test/SandboxTestingChildTests.h
@@ -7,6 +7,9 @@
 #include "SandboxTestingChild.h"
 
 #include "mozilla/ipc/UtilityProcessSandboxing.h"
+#ifdef XP_MACOSX
+#  include "nsCocoaFeatures.h"
+#endif
 #include "nsXULAppAPI.h"
 
 #ifdef XP_UNIX
@@ -256,7 +259,12 @@ void RunMacTestLaunchProcess(SandboxTestingChild* child,
   });
 
   // Test that launching an application using LSOpenCFURLRef fails
-  char* uri = const_cast<char*>("/System/Applications/Utilities/Console.app");
+  char* uri;
+  if (nsCocoaFeatures::OnCatalinaOrLater()) {
+    uri = const_cast<char*>("/System/Applications/Utilities/Console.app");
+  } else {
+    uri = const_cast<char*>("/Applications/Utilities/Console.app");
+  }
   CFStringRef filePath = ::CFStringCreateWithCString(kCFAllocatorDefault, uri,
                                                      kCFStringEncodingUTF8);
   CFURLRef urlRef = ::CFURLCreateWithFileSystemPath(
