diff --git a/third_party/rust/cubeb-coreaudio/src/lib.rs b/third_party/rust/cubeb-coreaudio/src/lib.rs
index f146b866a316..6ab7a9c7b771 100644
--- a/third_party/rust/cubeb-coreaudio/src/lib.rs
+++ b/third_party/rust/cubeb-coreaudio/src/lib.rs
@@ -2,7 +2,7 @@
 //
 // This program is made available under an ISC-style license.  See the
 // accompanying file LICENSE for details.
-
+#![feature(result_option_inspect)]
 extern crate atomic;
 #[macro_use]
 extern crate bitflags;
