diff --git a/third_party/rust/metal/src/blitpass.rs b/third_party/rust/metal/src/blitpass.rs
index cb561868f789..1fb37497e244 100644
--- a/third_party/rust/metal/src/blitpass.rs
+++ b/third_party/rust/metal/src/blitpass.rs
@@ -35,7 +35,7 @@ impl BlitPassSampleBufferAttachmentDescriptorArrayRef {
         &self,
         index: NSUInteger,
     ) -> Option<&BlitPassSampleBufferAttachmentDescriptorRef> {
-        unsafe { msg_send![self, objectAtIndexedSubscript: index] }
+        unsafe { msg_send![self, objectAtIndex: index] }
     }
 
     pub fn set_object_at(
@@ -45,7 +45,7 @@ impl BlitPassSampleBufferAttachmentDescriptorArrayRef {
     ) {
         unsafe {
             msg_send![self, setObject:attachment
-                     atIndexedSubscript:index]
+                     atIndex:index]
         }
     }
 }
