diff --git a/third_party/wasm2c/src/wast-parser.cc b/third_party/wasm2c/src/wast-parser.cc
index e3606b7c7a06..80635cbd217f 100644
--- a/third_party/wasm2c/src/wast-parser.cc
+++ b/third_party/wasm2c/src/wast-parser.cc
@@ -3746,7 +3746,7 @@ void WastParser::TokenQueue::pop_front() {
 
 const Token& WastParser::TokenQueue::at(size_t n) const {
   assert(n <= 1);
-  return tokens[i ^ static_cast<bool>(n)].value();
+  return *tokens[i ^ static_cast<bool>(n)];
 }
 
 const Token& WastParser::TokenQueue::front() const {
