diff --git a/toolkit/xre/nsNativeAppSupportCocoa.mm b/toolkit/xre/nsNativeAppSupportCocoa.mm
index 6476bf214655..e556e28ab400 100644
--- a/toolkit/xre/nsNativeAppSupportCocoa.mm
+++ b/toolkit/xre/nsNativeAppSupportCocoa.mm
@@ -72,7 +72,7 @@ NS_IMETHODIMP nsNativeAppSupportCocoa::Start(bool* _retval) {
   // alert here.  But the alert's message and buttons would require custom
   // localization.  So (for now at least) we just log an English message
   // to the console before quitting.
-  if (major < 10 || (major == 10 && minor < 12)) {
+  if (major < 10 || (major == 10 && minor < 7)) {
     NSLog(@"Minimum OS version requirement not met!");
     return NS_OK;
   }
