diff --git a/widget/TextRecognition.cpp b/widget/TextRecognition.cpp
index aa8ff98bb968..1ee57e8e1bf6 100644
--- a/widget/TextRecognition.cpp
+++ b/widget/TextRecognition.cpp
@@ -10,6 +10,10 @@
 #include "nsTextNode.h"
 #include "imgIContainer.h"
 
+#ifdef XP_MACOSX
+#  include "nsCocoaFeatures.h"
+#endif
+
 using namespace mozilla::dom;
 
 namespace mozilla::widget {
@@ -119,7 +123,10 @@ auto TextRecognition::DoFindText(gfx::DataSourceSurface&,
 
 bool TextRecognition::IsSupported() {
 #ifdef XP_MACOSX
-  return true;
+  // Catalina (10.15) or higher is required because of the following API:
+  // VNRecognizeTextRequest - macOS 10.15+
+  // https://developer.apple.com/documentation/vision/vnrecognizetextrequest?language=objc
+  return nsCocoaFeatures::OnCatalinaOrLater();
 #else
   return false;
 #endif
