diff --git a/widget/cocoa/TextInputHandler.h b/widget/cocoa/TextInputHandler.h
index 4283103817d8..3bf6b0e284ea 100644
--- a/widget/cocoa/TextInputHandler.h
+++ b/widget/cocoa/TextInputHandler.h
@@ -18,7 +18,6 @@
 #include "nsTArray.h"
 #include "mozilla/BasicEvents.h"
 #include "mozilla/EventForwards.h"
-#include "mozilla/Maybe.h"
 #include "mozilla/TextEventDispatcherListener.h"
 #include "WritingModes.h"
 
@@ -29,6 +28,10 @@ namespace widget {
 
 // Key code constants
 enum {
+#if !defined(MAC_OS_X_VERSION_10_12) || MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_12
+  kVK_RightCommand = 0x36,  // right command key
+#endif
+
   kVK_PC_PrintScreen = kVK_F13,
   kVK_PC_ScrollLock = kVK_F14,
   kVK_PC_Pause = kVK_F15,
@@ -904,8 +907,8 @@ class TextInputHandlerBase : public TextEventDispatcherListener {
  * And also provides the methods which controls the current IME transaction of
  * the instance.
  *
- * Note that an nsCocoaWindow handles one or more NSView's events.  E.g., even
- * if a text editor on XUL panel element, the input events handled on the parent
+ * Note that an nsCocoaWindow handles one or more NSView's events.  E.g., even if
+ * a text editor on XUL panel element, the input events handled on the parent
  * (or its ancestor) widget handles it (the native focus is set to it).  The
  * actual focused view is notified by OnFocusChangeInGecko.
  */
@@ -1113,23 +1116,13 @@ class IMEInputHandler : public TextInputHandlerBase {
 
  private:
   // If mIsIMEComposing is true, the composition string is stored here.
-  NSString* mIMECompositionString = nullptr;
-  // Store the composition start offset which is considered before dispatching
-  // eCompositionStart.
-  Maybe<uint32_t> mIMECompositionStartBeforeStart;
-  // Store the composition start in content.  This may be different from
-  // mIMECompositionBeforeStart if the web app changed the text after
-  // dispatching eCompositionStart.
-  Maybe<uint32_t> mIMECompositionStartInContent;
+  NSString* mIMECompositionString;
+  // If mIsIMEComposing is true, the start offset of the composition string.
+  uint32_t mIMECompositionStart;
 
   NSRange mMarkedRange;
   NSRange mSelectedRange;
 
-  // Store the override of mSelectedRange during a composition.  For avoiding
-  // IME to be confused at text changes before the composition, this keeps the
-  // selected range as in the composition string.
-  Maybe<NSRange> mSelectedRangeOverride;
-
   NSRange mRangeForWritingMode;  // range within which mWritingMode applies
   mozilla::WritingMode mWritingMode;
 
