diff --git a/widget/cocoa/nsDragService.mm b/widget/cocoa/nsDragService.mm
index 2469b8c13dbe..59e25dff40a5 100644
--- a/widget/cocoa/nsDragService.mm
+++ b/widget/cocoa/nsDragService.mm
@@ -53,6 +53,19 @@ already_AddRefed<nsIDragSession> nsDragService::CreateDragSession() {
   return sess.forget();
 }
 
+
+#if !defined(MAC_OS_X_VERSION_10_8) || MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_8
+/* this imagewithSize, unlike the one in nsCocoaWindow.mm, 
+ * was added after 10.7 support was dropped. so, i am adding 
+ * it here because i saw an exception and it has to be here*/
+@interface NSImage (ImageCreationWithDrawingHandler)
++ (NSImage*)imageWithSize:(NSSize)size
+                  flipped:(BOOL)drawingHandlerShouldBeCalledWithFlippedContext
+           drawingHandler:(BOOL (^)(NSRect dstRect))drawingHandler;
+@end
+
+#endif
+
 NSImage* nsDragSession::ConstructDragImage(nsINode* aDOMNode,
                                            const Maybe<CSSIntRegion>& aRegion,
                                            NSPoint* aDragPoint) {
@@ -70,16 +83,31 @@ NSImage* nsDragSession::ConstructDragImage(nsINode* aDOMNode,
         nsCocoaUtils::DevPixelsToCocoaPoints(dragRect.width, scaleFactor);
     size.height =
         nsCocoaUtils::DevPixelsToCocoaPoints(dragRect.height, scaleFactor);
-    image = [NSImage imageWithSize:size
-                           flipped:YES
-                    drawingHandler:^BOOL(NSRect dstRect) {
-                      [[NSColor grayColor] set];
-                      NSBezierPath* path =
-                          [NSBezierPath bezierPathWithRect:dstRect];
-                      [path setLineWidth:2.0];
-                      [path stroke];
-                      return YES;
-                    }];
+    if(@available(macOS 10.8, *)) {
+      image = [NSImage imageWithSize:size
+                             flipped:YES
+                      drawingHandler:^BOOL(NSRect dstRect) {
+                        [[NSColor grayColor] set];
+                        NSBezierPath* path =
+                            [NSBezierPath bezierPathWithRect:dstRect];
+                        [path setLineWidth:2.0];
+                        [path stroke];
+                        return YES;
+                      }];
+    } else {
+      image = [[NSImage alloc] initWithSize:size];
+      [image lockFocus];
+      [[NSColor grayColor] set];
+      NSBezierPath* path = [NSBezierPath bezierPath];
+      [path setLineWidth:2.0];
+      [path moveToPoint:NSMakePoint(0, 0)];
+      [path lineToPoint:NSMakePoint(0, size.height)];
+      [path lineToPoint:NSMakePoint(size.width, size.height)];
+      [path lineToPoint:NSMakePoint(size.width, 0)];
+      [path lineToPoint:NSMakePoint(0, 0)];
+      [path stroke];
+      [image unlockFocus];
+    }
   }
 
   LayoutDeviceIntPoint pt(dragRect.x, dragRect.YMost());
