diff --git a/widget/cocoa/nsTouchBarInput.mm b/widget/cocoa/nsTouchBarInput.mm
index af638f52d300..0a5da5eca7e3 100644
--- a/widget/cocoa/nsTouchBarInput.mm
+++ b/widget/cocoa/nsTouchBarInput.mm
@@ -118,7 +118,7 @@
         TouchBarInput* convertedChild =
             [[TouchBarInput alloc] initWithXPCOM:child];
         if (convertedChild) {
-          orderedChildren[i] = convertedChild;
+          [orderedChildren insertObject:convertedChild atIndex:i];
         }
       }
       [self setChildren:orderedChildren];
