diff --git a/Cargo.toml b/Cargo.toml
index 2e10d666ae35d..4434ea4f4cad1 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -66,7 +66,7 @@ exclude = [
 resolver = "2"
 
 [workspace.package]
-rust-version = "1.87.0"
+rust-version = "1.73.0"
 
 [workspace.dependencies]
 # Shared across multiple UniFFI consumers.
@@ -218,7 +218,6 @@ js-sys = { path = "build/rust/dummy-web/js-sys" }
 wasm-bindgen = { path = "build/rust/dummy-web/wasm-bindgen" }
 web-sys = { path = "build/rust/dummy-web/web-sys" }
 
-
 # Upgrade `core-foundation` 0.9.* to 0.10.
 core-foundation = { path = "build/rust/core-foundation" }
 # Patch `core-graphics-types` 0.1.* to 0.2.
