diff --git a/dom/webauthn/MacOSWebAuthnService.mm b/dom/webauthn/MacOSWebAuthnService.mm
index 92ca19fafcc31..8918decc03ba6 100644
--- a/dom/webauthn/MacOSWebAuthnService.mm
+++ b/dom/webauthn/MacOSWebAuthnService.mm
@@ -1023,7 +1023,7 @@ nsresult BrowsingContextIdToNSWindow(uint64_t browsingContextId,
                 credentials) {
           nsTArray<RefPtr<nsIWebAuthnAutoFillEntry>> autoFillEntries;
           for (NSUInteger i = 0; i < credentials.count; i++) {
-            const auto& credential = credentials[i];
+            const auto& credential = [credentials objectAtIndex:i];
             nsAutoString userName;
             nsCocoaUtils::GetStringForNSString(credential.name, userName);
             nsAutoString rpId;
