diff --git a/gfx/gl/GLContext.h b/gfx/gl/GLContext.h
index 775100661e006..dbcace361ae13 100644
--- a/gfx/gl/GLContext.h
+++ b/gfx/gl/GLContext.h
@@ -54,6 +54,10 @@ constexpr inline std::array<ElemT, 1 + sizeof...(More)> make_array(
 #  include "mozilla/ProfilerLabels.h"
 #endif
 
+#if defined(MOZ_WIDGET_COCOA)
+#  include "nsCocoaFeatures.h"
+#endif
+
 namespace mozilla {
 
 namespace gl {
@@ -176,6 +180,7 @@ enum class GLRenderer {
   Tegra,
   AndroidEmulator,
   GalliumLlvmpipe,
+  IntelHD3000,
   MicrosoftBasicRenderDriver,
   SamsungXclipse,
   Other
