diff --git a/ipc/chromium/src/base/message_loop.h b/ipc/chromium/src/base/message_loop.h
index 3b3111364b514..b0edc2c0e9767 100644
--- a/ipc/chromium/src/base/message_loop.h
+++ b/ipc/chromium/src/base/message_loop.h
@@ -20,8 +20,10 @@
 // We need this to declare base::MessagePumpWin::Dispatcher, which we should
 // really just eliminate.
 #  include "base/message_pump_win.h"
-#elif defined(XP_DARWIN)
+//it ain't workin. sorry nika
+/*#elif defined(XP_DARWIN)
 #  include "base/message_pump_kqueue.h"
+*/
 #else
 #  include "base/message_pump_libevent.h"
 #endif
@@ -329,10 +331,12 @@ class MessageLoop : public base::MessagePump::Delegate {
   base::MessagePumpWin* pump_win() {
     return static_cast<base::MessagePumpWin*>(pump_.get());
   }
-#elif defined(XP_DARWIN)
+//it ain't workin' sorry nika
+/*#elif defined(XP_DARWIN)
   base::MessagePumpKqueue* pump_kqueue() {
     return static_cast<base::MessagePumpKqueue*>(pump_.get());
   }
+*/
 #else
   base::MessagePumpLibevent* pump_libevent() {
     return static_cast<base::MessagePumpLibevent*>(pump_.get());
@@ -531,6 +535,8 @@ class MessageLoopForIO : public MessageLoop {
     return static_cast<base::MessagePumpForIO*>(pump_.get());
   }
 
+//it ain't workin. sorry nika
+/*
 #elif defined(XP_DARWIN)
 
   typedef base::MessagePumpKqueue::Watcher Watcher;
@@ -552,7 +558,7 @@ class MessageLoopForIO : public MessageLoop {
   bool WatchMachReceivePort(mach_port_t port,
                             MachPortWatchController* controller,
                             MachPortWatcher* delegate);
-
+*/
 #else
   typedef base::MessagePumpLibevent::Watcher Watcher;
   typedef base::MessagePumpLibevent::FileDescriptorWatcher
