diff --git a/toolkit/xre/nsEmbedFunctions.cpp b/toolkit/xre/nsEmbedFunctions.cpp
index 612f0fed2e949..45a30d12248cc 100644
--- a/toolkit/xre/nsEmbedFunctions.cpp
+++ b/toolkit/xre/nsEmbedFunctions.cpp
@@ -440,12 +440,15 @@ nsresult XRE_InitChildProcess(int aArgc, char* aArgv[],
 
   Maybe<IPC::Channel::ChannelHandle> clientChannel =
       geckoargs::sIPCHandle.Get(aArgc, aArgv);
+//it ain't workin sorry nika
+/*
 #ifdef XP_DARWIN
   MOZ_ASSERT_IF(clientChannel, !geckoargs::sIPCPort.IsPresent(aArgc, aArgv));
   if (!clientChannel) {
     clientChannel = geckoargs::sIPCPort.Get(aArgc, aArgv);
   }
 #endif
+*/
   if (NS_WARN_IF(!clientChannel)) {
     return NS_ERROR_FAILURE;
   }
