diff --git a/xpcom/io/nsLocalFileUnix.cpp b/xpcom/io/nsLocalFileUnix.cpp
index 98572642528cf..d06456bb99610 100644
--- a/xpcom/io/nsLocalFileUnix.cpp
+++ b/xpcom/io/nsLocalFileUnix.cpp
@@ -1960,6 +1960,7 @@ nsLocalFile::IsExecutable(bool* aResult) {
     }
   }
 
+  // On OS X, then query Launch Services.
 #ifdef MOZ_WIDGET_COCOA
   // Certain Mac applications, such as Classic applications, which
   // run under Rosetta, might not have the +x mode bit but are still
@@ -1969,16 +1970,15 @@ nsLocalFile::IsExecutable(bool* aResult) {
     return NS_ERROR_FAILURE;
   }
 
-  CFBooleanRef isApp = NULL;
-  *aResult = ::CFURLCopyResourcePropertyForKey(url, kCFURLIsApplicationKey,
-                                               &isApp, NULL) &&
-             (isApp == kCFBooleanTrue);
+  LSRequestedInfo theInfoRequest = kLSRequestAllInfo;
+  LSItemInfoRecord theInfo;
+  OSStatus result = ::LSCopyItemInfoForURL(url, theInfoRequest, &theInfo);
   ::CFRelease(url);
-  if (isApp) {
-    ::CFRelease(isApp);
-  }
-  if (*aResult) {
-    return NS_OK;
+  if (result == noErr) {
+    if ((theInfo.flags & kLSItemInfoIsApplication) != 0) {
+      *aResult = true;
+      return NS_OK;
+    }
   }
 #endif
 
