diff --git a/accessible/mac/mozAccessible.mm b/accessible/mac/mozAccessible.mm
index c3b3fc6fd22aa..6955c364f1986 100644
--- a/accessible/mac/mozAccessible.mm
+++ b/accessible/mac/mozAccessible.mm
@@ -1119,7 +1119,7 @@ - (void)handleAccessibleEvent:(uint32_t)eventType {
           static_cast<MOXTextMarkerDelegate*>([self moxTextMarkerDelegate]);
       NSMutableDictionary* userInfo =
           [[[delegate selectionChangeInfo] mutableCopy] autorelease];
-      userInfo[@"AXTextChangeElement"] = self;
+      [userInfo setObject:self forKey:@"AXTextChangeElement"];
 
       mozAccessible* webArea = [self topWebArea];
       [webArea
