diff --git a/accessible/xpcom/xpcAccessibleMacInterface.mm b/accessible/xpcom/xpcAccessibleMacInterface.mm
index 11184f6ddf474..59b40b2f3c67a 100644
--- a/accessible/xpcom/xpcAccessibleMacInterface.mm
+++ b/accessible/xpcom/xpcAccessibleMacInterface.mm
@@ -88,13 +88,6 @@
   NS_OBJC_END_TRY_BLOCK_RETURN(NS_ERROR_FAILURE)
 }
 
-// Return a string that uniquely identifies a custom action.
-static NSString* GetCustomActionName(NSAccessibilityCustomAction* action) {
-  return [NSString stringWithFormat:@"Name:%@ Target:%@ Selector:%@",
-                                    [action name], [action target],
-                                    NSStringFromSelector([action selector])];
-}
-
 NS_IMETHODIMP
 xpcAccessibleMacInterface::GetActionNames(nsTArray<nsString>& aActionNames) {
   NS_OBJC_BEGIN_TRY_BLOCK_RETURN
@@ -109,15 +102,19 @@
     aActionNames.AppendElement(actionName);
   }
 
-  if (NSArray* customActions = [mNativeObject accessibilityCustomActions]) {
-    for (id action in customActions) {
-      nsAutoString actionName;
-      NSString* actionNameStr = GetCustomActionName(action);
-      nsCocoaUtils::GetStringForNSString(actionNameStr, actionName);
-      aActionNames.AppendElement(actionName);
+  // custom actions are only available on 10.13 and up
+  if (@available(macOS 10.13, *)) {
+    if (NSArray* customActions = [mNativeObject accessibilityCustomActions]) {
+      for (id action in customActions) {
+        nsAutoString actionName;
+        NSString* actionNameStr = [NSString stringWithFormat:@"Name:%@ Target:%@ Selector:%@",
+                           [action name], [action target],
+                           NSStringFromSelector([action selector])];
+        nsCocoaUtils::GetStringForNSString(actionNameStr, actionName);
+        aActionNames.AppendElement(actionName);
+      }
     }
   }
-
   return NS_OK;
 
   NS_OBJC_END_TRY_BLOCK_RETURN(NS_ERROR_FAILURE)
@@ -136,12 +133,17 @@
 
   // First search custom actions, since `accessibilityActionDescription` will
   // just return the provided name if no description is found.
-  if (NSArray* customActions = [mNativeObject accessibilityCustomActions]) {
-    for (id action in customActions) {
-      NSString* actionNameStr = GetCustomActionName(action);
-      if ([actionNameStr isEqualToString:actionName]) {
-        nsCocoaUtils::GetStringForNSString([action name], aDescription);
-        return NS_OK;
+  //custom actions are only available on 10.13 and up.
+  if (@available(macos 10.13, *)) {
+    if (NSArray* customActions = [mNativeObject accessibilityCustomActions]) {
+      for (id action in customActions) {
+        NSString* actionNameStr = [NSString stringWithFormat:@"Name:%@ Target:%@ Selector:%@",
+                           [action name], [action target],
+                           NSStringFromSelector([action selector])];
+        if ([actionNameStr isEqualToString:actionName]) {
+          nsCocoaUtils::GetStringForNSString([action name], aDescription);
+          return NS_OK;
+        }
       }
     }
   }
@@ -165,12 +167,17 @@
 
   // First search custom actions, since `accessibilityPerformAction` will
   // silently fail on unknown action names.
-  if (NSArray* customActions = [mNativeObject accessibilityCustomActions]) {
-    for (id action in customActions) {
-      NSString* actionNameStr = GetCustomActionName(action);
-      if ([actionNameStr isEqualToString:actionName]) {
-        [[action target] performSelector:[action selector]];
-        return NS_OK;
+  // custom actions are only available on 10.13 and up
+  if (@available(macOS 10.13, *)) {
+    if (NSArray* customActions = [mNativeObject accessibilityCustomActions]) {
+      for (id action in customActions) {
+        NSString* actionNameStr = [NSString stringWithFormat:@"Name:%@ Target:%@ Selector:%@",
+                           [action name], [action target],
+                           NSStringFromSelector([action selector])];
+        if ([actionNameStr isEqualToString:actionName]) {
+          [[action target] performSelector:[action selector]];
+          return NS_OK;
+        }
       }
     }
   }
@@ -318,7 +325,7 @@
       return NS_ERROR_FAILURE;
     }
     for (size_t i = 0; i < [objArr count]; ++i) {
-      nsresult rv = NSObjectToJsValue(objArr[i], aCx, v[i]);
+      nsresult rv = NSObjectToJsValue([objArr objectAtIndex:i], aCx, v[i]);
       NS_ENSURE_SUCCESS(rv, rv);
     }
 
@@ -356,7 +363,7 @@
 
                           NSMutableDictionary* attrRun =
                               [attributes mutableCopy];
-                          attrRun[@"string"] = str;
+                          [attrRun setObject:str forKey:@"string"];
 
                           [attrRunArray addObject:attrRun];
                         }];
@@ -587,7 +594,7 @@
       JS_GetPropertyById(aCx, object, ids[i], &currentValue);
       id unwrappedValue = JsValueToNSObject(currentValue, aCx, &rv);
       NS_ENSURE_SUCCESS(rv, nil);
-      dict[unwrappedKey] = unwrappedValue;
+      [dict setObject:unwrappedValue forKey:unwrappedKey];
     }
 
     *aResult = NS_OK;
