diff --git a/build/gyp.mozbuild b/build/gyp.mozbuild
index cd71ae9f966bc..39bf409d101cd 100644
--- a/build/gyp.mozbuild
+++ b/build/gyp.mozbuild
@@ -16,8 +16,8 @@ gyp_vars.update({
     ),
     "build_with_mozilla": 1,
     "build_with_chromium": 0,
-    "mac_sdk_min": "10.15",
-    "mac_deployment_target": "10.15",
+    "mac_sdk_min": "10.6",
+    "mac_deployment_target": "10.6",
     "use_official_google_api_keys": 0,
     "have_clock_monotonic": 1 if CONFIG["HAVE_CLOCK_MONOTONIC"] else 0,
     "have_ethtool_cmd_speed_hi": (
