diff --git a/dom/media/platforms/apple/AppleDecoderModule.cpp b/dom/media/platforms/apple/AppleDecoderModule.cpp
index e9ea4edef5ca8..731836b7990b4 100644
--- a/dom/media/platforms/apple/AppleDecoderModule.cpp
+++ b/dom/media/platforms/apple/AppleDecoderModule.cpp
@@ -10,11 +10,16 @@
 #include "AppleATDecoder.h"
 #include "AppleVTDecoder.h"
 #include "H265.h"
+#include "AppleVDADecoder.h"
+#include "AppleVDALinker.h"
+#include "AppleCMLinker.h"
+#include "AppleCVLinker.h"
+#include "AppleVTLinker.h"
+
 #include "MP4Decoder.h"
 #include "VPXDecoder.h"
 #include "VideoUtils.h"
 #include "mozilla/Logging.h"
-#include "mozilla/ScopeExit.h"
 #include "mozilla/StaticPrefs_media.h"
 #include "mozilla/gfx/gfxVars.h"
 
@@ -22,6 +27,8 @@ extern "C" {
 // Only exists from MacOS 11
 extern void VTRegisterSupplementalVideoDecoderIfAvailable(
     CMVideoCodecType codecType) __attribute__((weak_import));
+extern Boolean VTIsHardwareDecodeSupported(CMVideoCodecType codecType)
+    __attribute__((weak_import));
 }
 
 namespace mozilla {
@@ -31,6 +38,11 @@ using media::DecodeSupportSet;
 using media::MCSInfo;
 using media::MediaCodec;
 
+bool AppleDecoderModule::sIsCoreMediaAvailable = false;
+bool AppleDecoderModule::sIsCoreVideoAvailable = false;
+bool AppleDecoderModule::sIsVTAvailable = false;
+bool AppleDecoderModule::sIsVDAAvailable = false;
+
 static inline CMVideoCodecType GetCMVideoCodecType(const MediaCodec& aCodec) {
   switch (aCodec) {
     case MediaCodec::H264:
@@ -45,13 +57,18 @@ static inline CMVideoCodecType GetCMVideoCodecType(const MediaCodec& aCodec) {
       return static_cast<CMVideoCodecType>(0);
   }
 }
-
 /* static */
 void AppleDecoderModule::Init() {
   if (sInitialized) {
     return;
   }
 
+  //10.7.3 - > 10.7 need these (thanks jya)
+  sIsCoreMediaAvailable = AppleCMLinker::Link();
+  sIsCoreVideoAvailable = AppleCVLinker::Link();
+  sIsVDAAvailable = AppleVDALinker::Link();
+  sIsVTAvailable = AppleVTLinker::Link();
+  
   // Initialize all values to false first.
   for (auto& support : sCanUseHWDecoder) {
     support = false;
@@ -86,11 +103,19 @@ already_AddRefed<MediaDataDecoder> AppleDecoderModule::CreateVideoDecoder(
           .isEmpty()) {
     return nullptr;
   }
+  
   RefPtr<MediaDataDecoder> decoder;
-  if (IsVideoSupported(aParams.VideoConfig(), aParams.mOptions)) {
-    decoder = new AppleVTDecoder(aParams.VideoConfig(), aParams.mImageContainer,
-                                 aParams.mOptions, aParams.mKnowsCompositor,
-                                 aParams.mTrackingId);
+
+  if(__builtin_available(macOS 10.7, *)) {
+    if (IsVideoSupported(aParams.VideoConfig(), aParams.mOptions)) {
+      decoder = new AppleVTDecoder(aParams.VideoConfig(), aParams.mImageContainer,
+          aParams.mOptions, aParams.mKnowsCompositor,
+          aParams.mTrackingId);
+    }
+  } else {
+      decoder = new AppleVDADecoder(aParams.VideoConfig(), aParams.mImageContainer,
+          aParams.mOptions, aParams.mKnowsCompositor,
+          aParams.mTrackingId);
   }
   return decoder.forget();
 }
@@ -200,7 +225,7 @@ bool AppleDecoderModule::IsVideoSupported(
     return profile == 0;
   }
 
-  if (!VPXDecoder::IsVP9(aConfig.mMimeType) ||
+  if (!VPXDecoder::IsVP9(aConfig.mMimeType) || 
       !sCanUseHWDecoder[MediaCodec::VP9] ||
       aOptions.contains(
           CreateDecoderParams::Option::HardwareDecoderNotAllowed)) {
@@ -239,8 +264,13 @@ bool AppleDecoderModule::CanCreateHWDecoder(const MediaCodec& aCodec) {
     return false;
   }
 
-  if (!VTIsHardwareDecodeSupported(GetCMVideoCodecType(aCodec))) {
-    return false;
+  if (__builtin_available(macOS 10.13, *)) {
+      if (!VTIsHardwareDecodeSupported) {
+        return false;
+      }
+      if (!VTIsHardwareDecodeSupported(GetCMVideoCodecType(aCodec))) {
+        return false;
+      }
   }
 
   // H264 hardware decoding has been supported since macOS 10.6 on most Intel
@@ -249,7 +279,7 @@ bool AppleDecoderModule::CanCreateHWDecoder(const MediaCodec& aCodec) {
   if (aCodec == MediaCodec::H264) {
     return true;
   }
-
+    
   // Build up a fake extradata to create an actual decoder to verify
   VideoInfo info(1920, 1080);
   if (aCodec == MediaCodec::AV1) {
