diff --git a/dom/media/platforms/apple/AppleVTDecoder.h b/dom/media/platforms/apple/AppleVTDecoder.h
index 7ad9b4e6b8986..bc21e92b3bf81 100644
--- a/dom/media/platforms/apple/AppleVTDecoder.h
+++ b/dom/media/platforms/apple/AppleVTDecoder.h
@@ -9,6 +9,7 @@
 #include <CoreMedia/CoreMedia.h>          // For CMVideoFormatDescriptionRef
 #include <VideoToolbox/VideoToolbox.h>    // For VTDecompressionSessionRef
 
+#include "AppleVDADecoder.h"
 #include "AppleDecoderModule.h"
 #include "PerformanceRecorder.h"
 #include "PlatformDecoderModule.h"
