diff --git a/dom/system/mac/nsOSPermissionRequest.mm b/dom/system/mac/nsOSPermissionRequest.mm
index 3f0e714fa6958..626c3c1923e6d 100644
--- a/dom/system/mac/nsOSPermissionRequest.mm
+++ b/dom/system/mac/nsOSPermissionRequest.mm
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "nsOSPermissionRequest.h"
-
+#include "nsCocoaFeatures.h"
 #include "mozilla/dom/Promise.h"
 #include "nsCocoaUtils.h"
 
@@ -14,24 +14,41 @@
 NS_IMETHODIMP
 nsOSPermissionRequest::GetAudioCapturePermissionState(uint16_t* aAudio) {
   MOZ_ASSERT(aAudio);
-  return nsCocoaUtils::GetAudioCapturePermissionState(*aAudio);
+    if (!nsCocoaFeatures::OnMojaveOrLater()) {
+    return nsOSPermissionRequestBase::GetAudioCapturePermissionState(aAudio);
+    }
+    return nsCocoaUtils::GetAudioCapturePermissionState(*aAudio);
 }
 
 NS_IMETHODIMP
 nsOSPermissionRequest::GetVideoCapturePermissionState(uint16_t* aVideo) {
   MOZ_ASSERT(aVideo);
+ 
+  if (!nsCocoaFeatures::OnMojaveOrLater()) {
+    return nsOSPermissionRequestBase::GetVideoCapturePermissionState(aVideo);
+  }
+
+
   return nsCocoaUtils::GetVideoCapturePermissionState(*aVideo);
 }
 
 NS_IMETHODIMP
 nsOSPermissionRequest::GetScreenCapturePermissionState(uint16_t* aScreen) {
   MOZ_ASSERT(aScreen);
+  if (!nsCocoaFeatures::OnMojaveOrLater()) {
+    return nsOSPermissionRequestBase::GetScreenCapturePermissionState(aScreen);
+ }
+
   return nsCocoaUtils::GetScreenCapturePermissionState(*aScreen);
 }
 
 NS_IMETHODIMP
 nsOSPermissionRequest::RequestVideoCapturePermission(JSContext* aCx,
                                                      Promise** aPromiseOut) {
+  if (!nsCocoaFeatures::OnMojaveOrLater()) {
+    return nsOSPermissionRequestBase::RequestVideoCapturePermission(aCx, aPromiseOut);
+  }
+
   RefPtr<Promise> promiseHandle;
   nsresult rv = GetPromise(aCx, promiseHandle);
   if (NS_FAILED(rv)) {
@@ -46,6 +63,10 @@
 NS_IMETHODIMP
 nsOSPermissionRequest::RequestAudioCapturePermission(JSContext* aCx,
                                                      Promise** aPromiseOut) {
+
+  if (!nsCocoaFeatures::OnMojaveOrLater()) {
+    return nsOSPermissionRequestBase::RequestAudioCapturePermission(aCx, aPromiseOut);
+  }  
   RefPtr<Promise> promiseHandle;
   nsresult rv = GetPromise(aCx, promiseHandle);
   if (NS_FAILED(rv)) {
@@ -59,5 +80,8 @@
 
 NS_IMETHODIMP
 nsOSPermissionRequest::MaybeRequestScreenCapturePermission() {
+  if (!nsCocoaFeatures::OnCatalinaOrLater()) {
+    return nsOSPermissionRequestBase::MaybeRequestScreenCapturePermission();
+  }
   return nsCocoaUtils::MaybeRequestScreenCapturePermission();
 }
