diff --git a/gfx/gl/GLContext.h b/gfx/gl/GLContext.h
index 49cc484c81ffc..6d1fe19f10533 100644
--- a/gfx/gl/GLContext.h
+++ b/gfx/gl/GLContext.h
@@ -52,6 +52,10 @@ constexpr inline std::array<ElemT, 1 + sizeof...(More)> make_array(
 #  include "mozilla/ProfilerLabels.h"
 #endif
 
+#if defined(MOZ_WIDGET_COCOA)
+#  include "nsCocoaFeatures.h"
+#endif
+
 namespace mozilla {
 
 namespace gl {
@@ -174,6 +178,7 @@ enum class GLRenderer {
   Tegra,
   AndroidEmulator,
   GalliumLlvmpipe,
+  IntelHD3000,
   MicrosoftBasicRenderDriver,
   SamsungXclipse,
   Other
