diff --git a/gfx/layers/ipc/CompositorBridgeChild.h b/gfx/layers/ipc/CompositorBridgeChild.h
index e703ec6a31aff..8f38101baa3ff 100644
--- a/gfx/layers/ipc/CompositorBridgeChild.h
+++ b/gfx/layers/ipc/CompositorBridgeChild.h
@@ -143,6 +143,7 @@ class CompositorBridgeChild final : public PCompositorBridgeChild,
   static void ShutDown();
 
   FwdTransactionCounter& GetFwdTransactionCounter();
+  void WindowOverlayChanged() { mWindowOverlayChanged = true; }
 
   /**
    * Hold TextureClient ref until end of usage on host side if
@@ -222,6 +223,7 @@ class CompositorBridgeChild final : public PCompositorBridgeChild,
   // True until the beginning of the two-step shutdown sequence of this actor.
   bool mCanSend;
 
+  bool mWindowOverlayChanged;
   // False until the actor is destroyed.
   bool mActorDestroyed;
 
