diff --git a/gfx/layers/wr/WebRenderLayerManager.h b/gfx/layers/wr/WebRenderLayerManager.h
index 0ca0423349542..c0f229a397bbe 100644
--- a/gfx/layers/wr/WebRenderLayerManager.h
+++ b/gfx/layers/wr/WebRenderLayerManager.h
@@ -158,6 +158,7 @@ class WebRenderLayerManager final : public WindowRenderer {
   WebRenderScrollData& GetScrollData() { return mScrollData; }
 
   void WrUpdated();
+  void WindowOverlayChanged() { mWindowOverlayChanged = true; }
   nsIWidget* GetWidget() { return mWidget; }
 
   uint32_t StartFrameTimeRecording(int32_t aBufferSize) override;
@@ -232,6 +233,7 @@ class WebRenderLayerManager final : public WindowRenderer {
   // APZ to do it's job
   WebRenderScrollData mScrollData;
 
+  bool mWindowOverlayChanged;
   bool mNeedsComposite;
   bool mIsFirstPaint;
   bool mDestroyed;
