diff --git a/ipc/chromium/src/chrome/common/ipc_channel.cc b/ipc/chromium/src/chrome/common/ipc_channel.cc
index 777c67e14c410..c6ead25d28cb9 100644
--- a/ipc/chromium/src/chrome/common/ipc_channel.cc
+++ b/ipc/chromium/src/chrome/common/ipc_channel.cc
@@ -12,9 +12,12 @@
 #else
 #  include "chrome/common/ipc_channel_posix.h"
 #endif
+//it ain't workin. sorry nika
+/*
 #ifdef XP_DARWIN
 #  include "chrome/common/ipc_channel_mach.h"
 #endif
+*/
 
 namespace IPC {
 
@@ -35,7 +38,8 @@ already_AddRefed<Channel> Channel::Create(ChannelHandle pipe, Mode mode,
                                                 other_pid);
 #endif
   }
-#if XP_DARWIN
+//it ain't workin. sorry nika
+/*#if XP_DARWIN
   if (auto* receive = std::get_if<mozilla::UniqueMachReceiveRight>(&pipe)) {
     return mozilla::MakeAndAddRef<ChannelMach>(std::move(*receive), nullptr,
                                                mode, other_pid);
@@ -45,6 +49,7 @@ already_AddRefed<Channel> Channel::Create(ChannelHandle pipe, Mode mode,
                                                other_pid);
   }
 #endif
+*/
   MOZ_ASSERT_UNREACHABLE("unhandled pipe type");
   return nullptr;
 }
