diff --git a/ipc/chromium/src/chrome/common/ipc_message.cc b/ipc/chromium/src/chrome/common/ipc_message.cc
index feba44d036763..7e4caa9f1f188 100644
--- a/ipc/chromium/src/chrome/common/ipc_message.cc
+++ b/ipc/chromium/src/chrome/common/ipc_message.cc
@@ -131,8 +131,10 @@ void Message::SetAttachedFileHandles(
 bool Message::has_any_attachments() const {
   return !attached_ports_.IsEmpty() || !attached_handles_.IsEmpty()
 #if defined(XP_DARWIN)
-         || !attached_send_rights_.IsEmpty() ||
+/* sorry nika it ain't working
+      || !attached_send_rights_.IsEmpty() ||
          !attached_receive_rights_.IsEmpty()
+*/
 #endif
       ;
 }
@@ -195,6 +197,8 @@ uint32_t Message::num_send_rights() const {
   return attached_send_rights_.Length();
 }
 
+//it ain't working sorry nika
+/*
 bool Message::WriteMachReceiveRight(mozilla::UniqueMachReceiveRight port) {
   uint32_t index = attached_receive_rights_.Length();
   WriteUInt32(index);
@@ -223,6 +227,7 @@ bool Message::ConsumeMachReceiveRight(
 uint32_t Message::num_receive_rights() const {
   return attached_receive_rights_.Length();
 }
+*/
 #endif
 
 bool Message::WillBeRoutedExternally(
