diff --git a/ipc/chromium/src/chrome/common/mach_ipc_mac.h b/ipc/chromium/src/chrome/common/mach_ipc_mac.h
index 520b9a5d3b374..209ac15492a85 100644
--- a/ipc/chromium/src/chrome/common/mach_ipc_mac.h
+++ b/ipc/chromium/src/chrome/common/mach_ipc_mac.h
@@ -48,13 +48,12 @@ bool MachChildProcessCheckIn(
 //==============================================================================
 // Called by MacProcessLauncher to transfer ports to the child process, and
 // acquire the child process task port.
-using MachHandleProcessCheckInPromise =
-    mozilla::MozPromise<task_t, mozilla::ipc::LaunchError, true>;
-RefPtr<MachHandleProcessCheckInPromise> MachHandleProcessCheckIn(
-    mozilla::UniqueMachReceiveRight endpoint, pid_t child_pid,
-    mozilla::TimeDuration timeout,
-    std::vector<mozilla::UniqueMachSendRight> send_rights,
-    std::vector<mozilla::UniqueMachReceiveRight> receive_rights);
+mozilla::Result<mozilla::Ok, mozilla::ipc::LaunchError>
+MachHandleProcessCheckIn(
+    mach_port_t endpoint, pid_t child_pid, mach_msg_timeout_t timeout,
+    const std::vector<mozilla::UniqueMachSendRight>& send_rights,
+    std::vector<mozilla::UniqueMachReceiveRight>& receive_rights,
+    task_t* child_task);
 #endif
 
 #endif  // BASE_MACH_IPC_MAC_H_
