diff --git a/netwerk/protocol/http/nsHttpHandler.cpp b/netwerk/protocol/http/nsHttpHandler.cpp
index 781609fcb09a0..d74b52ee2f664 100644
--- a/netwerk/protocol/http/nsHttpHandler.cpp
+++ b/netwerk/protocol/http/nsHttpHandler.cpp
@@ -110,6 +110,7 @@
 
 #if defined(XP_MACOSX)
 #  include <CoreServices/CoreServices.h>
+#  include "nsCocoaFeatures.h"
 #endif
 
 //-----------------------------------------------------------------------------
@@ -1195,7 +1196,8 @@ void nsHttpHandler::InitUserAgentComponents() {
 #  endif
 
 #elif defined(XP_MACOSX)
-  mOscpu.AssignLiteral("Intel Mac OS X 10.15");
+   // Always return an "Intel" UA string, even on ARM64 macOS like Safari does.
+   mOscpu = nsPrintfCString("Intel Mac OS X 10.15");
 #elif defined(ANDROID)
   mOscpu.AssignLiteral("Linux armv81");
 #elif defined(XP_IOS)
