diff --git a/servo/components/style/values/specified/box.rs b/servo/components/style/values/specified/box.rs
index e70c1d592ebdb..1dc14e0cb0e9e 100644
--- a/servo/components/style/values/specified/box.rs
+++ b/servo/components/style/values/specified/box.rs
@@ -1692,6 +1692,11 @@ pub enum Appearance {
     Textfield,
     /// The dropdown button(s) that open up a dropdown list.
     MenulistButton,
+    /// <menu> and <menuitem> appearances
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    Menuitem,
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    Checkmenuitem,
     /// https://drafts.csswg.org/css-forms/#appearance
     #[parse(condition = "appearance_base_enabled")]
     Base,
@@ -1702,6 +1707,11 @@ pub enum Appearance {
     /// Menu Popup background.
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     Menupopup,
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    Menuseparator,
+    /// The meter bar's meter indicator.
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    Meterchunk,
     /// The "arrowed" part of the dropdown button that open up a dropdown list.
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     MozMenulistArrowButton,
@@ -1711,9 +1721,14 @@ pub enum Appearance {
     /// For HTML's <input type=password>
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     PasswordInput,
+    /// The progress bar's progress indicator
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    Progresschunk,
     /// nsRangeFrame and its subparts
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     Range,
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    RangeThumb,
     /// The scrollbar slider
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     ScrollbarHorizontal,
@@ -1738,12 +1753,21 @@ pub enum Appearance {
     /// The scroll corner
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     Scrollcorner,
+    /// A separator.  Can be horizontal or vertical.
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    Separator,
+    /// A spin control (up/down control for time/date pickers).
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    Spinner,
     /// The up button of a spin control.
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     SpinnerUpbutton,
     /// The down button of a spin control.
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     SpinnerDownbutton,
+    /// A status bar in a main application window.
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    Statusbar,
     /// A single toolbar button (with no associated dropdown).
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     Toolbarbutton,
@@ -1765,6 +1789,8 @@ pub enum Appearance {
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     MozMacWindow,
 
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    MozMacFullscreenButton,
     /// Windows themed window frame elements.
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     MozWindowButtonBox,
@@ -1782,11 +1808,24 @@ pub enum Appearance {
     MozWindowTitlebarMaximized,
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     MozWindowDecorations,
-
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    MozMacActiveSourceListSelection,
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     MozMacDisclosureButtonClosed,
     #[parse(condition = "ParserContext::chrome_rules_enabled")]
     MozMacDisclosureButtonOpen,
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    MozMacSourceList,
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    MozMacSourceListSelection,
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    MozMacVibrancyDark,
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    MozMacVibrancyLight,
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    MozMacVibrantTitlebarDark,
+    #[parse(condition = "ParserContext::chrome_rules_enabled")]
+    MozMacVibrantTitlebarLight,
 
     /// A themed focus outline (for outline:auto).
     ///
