diff --git a/toolkit/xre/MacApplicationDelegate.mm b/toolkit/xre/MacApplicationDelegate.mm
index b7db9890d967c..7a690dae56f6e 100644
--- a/toolkit/xre/MacApplicationDelegate.mm
+++ b/toolkit/xre/MacApplicationDelegate.mm
@@ -41,6 +41,7 @@
 #include "nsCommandLine.h"
 #include "nsStandaloneNativeMenu.h"
 #include "nsCocoaUtils.h"
+#include "nsCocoaFeatures.h"
 #include "nsMenuBarX.h"
 #include "mozilla/NeverDestroyed.h"
 
@@ -72,6 +73,7 @@ - (IBAction)copy:(id)aSender;
 
 static LaunchStatus sLaunchStatus = LaunchStatus::Initial;
 
+
 static nsTArray<nsCString>& StartupURLs() {
   static mozilla::NeverDestroyed<nsTArray<nsCString>> sStartupURLs;
   return *sStartupURLs;
@@ -128,8 +130,7 @@ void SetupMacApplicationDelegate(bool* gRestartedByOS) {
   MOZ_ASSERT(
       sLaunchStatus == LaunchStatus::Initial,
       "Launch status should be in intial state when setting up delegate");
-  sLaunchStatus = LaunchStatus::DelegateIsSetup;
-
+    sLaunchStatus = LaunchStatus::DelegateIsSetup;
   NS_OBJC_END_TRY_IGNORE_BLOCK;
 }
 
@@ -139,15 +140,14 @@ void SetupMacApplicationDelegate(bool* gRestartedByOS) {
 // (b) Collect URLs that were provided to the app at open time.
 void InitializeMacApp() {
   if (sLaunchStatus != LaunchStatus::DelegateIsSetup) {
-    // Delegate has not been set up or NSApp has been launched already.
+      // Delegate has not been set up or NSApp has been launched already.
     return;
   }
-
   sLaunchStatus = LaunchStatus::CollectingURLs;
   if (!gfxPlatform::IsHeadless()) {
     [NSApp run];
   }
-  sLaunchStatus = LaunchStatus::CollectedURLs;
+  sLaunchStatus = LaunchStatus::CollectedURLs; 
 }
 
 nsTArray<nsCString> TakeStartupURLs() { return std::move(StartupURLs()); }
@@ -162,6 +162,24 @@ - (id)init {
   NS_OBJC_BEGIN_TRY_BLOCK_RETURN;
 
   if ((self = [super init])) {
+    if (!nsCocoaFeatures::OnHighSierraOrLater()) {
+      NSAppleEventManager* aeMgr = [NSAppleEventManager sharedAppleEventManager];
+
+      [aeMgr setEventHandler:self
+                 andSelector:@selector(handleAppleEvent:withReplyEvent:)
+               forEventClass:kInternetEventClass
+                  andEventID:kAEGetURL];
+
+      [aeMgr setEventHandler:self
+                 andSelector:@selector(handleAppleEvent:withReplyEvent:)
+               forEventClass:'WWW!'
+                  andEventID:'OURL'];
+
+      [aeMgr setEventHandler:self
+                 andSelector:@selector(handleAppleEvent:withReplyEvent:)
+               forEventClass:kCoreEventClass
+                  andEventID:kAEOpenDocuments];
+   }
     if (![NSApp windowsMenu]) {
       // If the application has a windows menu, it will keep it up to date and
       // prepend the window list to the Dock menu automatically.
@@ -175,6 +193,23 @@ - (id)init {
   NS_OBJC_END_TRY_BLOCK_RETURN(nil);
 }
 
+- (void)dealloc {
+  if(!nsCocoaFeatures::OnHighSierraOrLater()) {
+    NS_OBJC_BEGIN_TRY_IGNORE_BLOCK;
+
+    NSAppleEventManager* aeMgr = [NSAppleEventManager sharedAppleEventManager];
+    [aeMgr removeEventHandlerForEventClass:kInternetEventClass
+                                andEventID:kAEGetURL];
+    [aeMgr removeEventHandlerForEventClass:'WWW!' andEventID:'OURL'];
+    [aeMgr removeEventHandlerForEventClass:kCoreEventClass
+                                andEventID:kAEOpenDocuments];
+    [super dealloc];
+
+    NS_OBJC_END_TRY_IGNORE_BLOCK;
+  }
+}
+
+
 // The method that NSApplication calls upon a request to reopen, such as when
 // the Dock icon is clicked and no windows are open. A "visible" window may be
 // miniaturized, so we can't skip nsCocoaNativeReOpen() if 'flag' is 'true'.
@@ -332,14 +367,19 @@ - (BOOL)application:(NSApplication*)application
 
 - (BOOL)application:(NSApplication*)application
     continueUserActivity:(NSUserActivity*)userActivity
+#if defined(MAC_OS_X_VERSION_10_14) && \
+    MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_14
       restorationHandler:
           (void (^)(NSArray<id<NSUserActivityRestoring>>*))restorationHandler {
-  if (![userActivity.activityType
-          isEqualToString:NSUserActivityTypeBrowsingWeb]) {
-    return NO;
+#else
+      restorationHandler:(void (^)(NSArray*))restorationHandler {
+#endif
+    if (![userActivity.activityType
+            isEqualToString:NSUserActivityTypeBrowsingWeb]) {
+      return NO;
   }
 
-  return [self openURLs:@[ userActivity.webpageURL ]];
+  return [self openURLs:@[userActivity.webpageURL]];
 }
 
 - (void)application:(NSApplication*)application
@@ -348,6 +388,56 @@ - (void)application:(NSApplication*)application
   NSLog(@"Failed to continue user activity %@: %@", userActivityType, error);
 }
 
+// opened. It will be called once for each selected document.
+- (BOOL)application:(NSApplication*)theApplication
+           openFile:(NSString*)filename {
+  if(nsCocoaFeatures::OnHighSierraOrLater()) {
+    return false; 
+  }
+
+  NS_OBJC_BEGIN_TRY_BLOCK_RETURN;
+  return [self openURLs:((NSArray<NSURL*>*) @[filename])];
+  NS_OBJC_END_TRY_BLOCK_RETURN(NO);
+}
+
+- (void)handleAppleEvent:(NSAppleEventDescriptor*)event
+          withReplyEvent:(NSAppleEventDescriptor*)replyEvent {
+  if (!event) return;
+
+  AutoAutoreleasePool pool;
+
+  bool isGetURLEvent = ([event eventClass] == kInternetEventClass &&
+                        [event eventID] == kAEGetURL);
+
+  if (isGetURLEvent ||
+      ([event eventClass] == 'WWW!' && [event eventID] == 'OURL')) {
+    NSString* urlString =
+        [[event paramDescriptorForKeyword:keyDirectObject] stringValue];
+    NSURL* url = [NSURL URLWithString:urlString];
+
+    [self openURLs:@[url]];
+  } else if ([event eventClass] == kCoreEventClass &&
+             [event eventID] == kAEOpenDocuments) {
+    NSAppleEventDescriptor* fileListDescriptor =
+        [event paramDescriptorForKeyword:keyDirectObject];
+    if (!fileListDescriptor) return;
+
+    // Descriptor list indexing is one-based...
+    NSInteger numberOfFiles = [fileListDescriptor numberOfItems];
+    for (NSInteger i = 1; i <= numberOfFiles; i++) {
+      NSString* urlString =
+          [[fileListDescriptor descriptorAtIndex:i] stringValue];
+      if (!urlString) continue;
+
+      // We need a path, not a URL
+      NSURL* url = [NSURL URLWithString:urlString];
+      if (!url) continue;
+
+      [self application:NSApp openFile:[url path]];
+    }
+  }
+}
+
 - (BOOL)openURLs:(NSArray<NSURL*>*)urls {
   nsTArray<const char*> args([urls count] * 2 + 2);
   // Placeholder for unused program name.
@@ -395,5 +485,4 @@ - (BOOL)openURLs:(NSArray<NSURL*>*)urls {
 
   return YES;
 }
-
 @end
