diff --git a/tools/profiler/core/platform.cpp b/tools/profiler/core/platform.cpp
index 83e7b5af52dfe..e5e4017cd170e 100644
--- a/tools/profiler/core/platform.cpp
+++ b/tools/profiler/core/platform.cpp
@@ -7810,6 +7810,13 @@ void profiler_mark_thread_awake() {
       currentPriority = threadInfo.Priority;
     }
   }
+#endif
+#if defined(GP_OS_darwin)
+  size_t qos_self_retval;
+  if(__builtin_available(macOS 10.10, *)) 
+    qos_self_retval = qos_class_self();
+  else
+    qos_self_retval = QOS_CLASS_UNSPECIFIED;
 #endif
   PROFILER_MARKER("Awake", OTHER, MarkerTiming::IntervalStart(), CPUAwakeMarker,
                   0 /* CPU time */
@@ -7819,7 +7826,7 @@ void profiler_mark_thread_awake() {
 #endif
 #if defined(GP_OS_darwin)
                   ,
-                  qos_class_self()
+                  qos_self_retval
 #endif
 #if defined(GP_OS_windows)
                       ,
