diff --git a/widget/cocoa/nsCocoaUtils.h b/widget/cocoa/nsCocoaUtils.h
index 69d5e33bdf7aa..68b87988eb4e2 100644
--- a/widget/cocoa/nsCocoaUtils.h
+++ b/widget/cocoa/nsCocoaUtils.h
@@ -41,6 +41,10 @@ extern NSString* const kMozFileUrlsPboardType;
 + (NSString*)stringFromPboardType:(NSString*)aType;
 @end
 
+#if !defined(MAC_OS_X_VERSION_10_8) || MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_8
+enum { NSEventPhaseMayBegin = 0x1 << 5 };
+#endif
+
 class nsITransferable;
 class nsIWidget;
 
@@ -91,6 +95,13 @@ class nsAutoreleasePool {
 - (BOOL)_isRunningModal;
 - (BOOL)_isRunningAppModal;
 
+// It's sometimes necessary to explicitly remove a window from the "window
+// cache" in order to deactivate it.  The "window cache" is an undocumented
+// subsystem, all of whose methods are included in the NSWindowCache category
+// of the NSApplication class (in header files generated using class-dump).
+// Present in all versions of OS X from (at least) 10.2.8 through 10.5.
+- (void)_removeWindowFromCache:(NSWindow *)aWindow;
+
 // Send an event to the current Cocoa app-modal session.  Present in all
 // versions of OS X from (at least) 10.2.8 through 10.5.
 - (void)_modalSession:(NSModalSession)aSession sendEvent:(NSEvent*)theEvent;
