diff --git a/widget/cocoa/nsNativeThemeCocoa.h b/widget/cocoa/nsNativeThemeCocoa.h
index b3445fdcb48fb..dd3f5f422c74c 100644
--- a/widget/cocoa/nsNativeThemeCocoa.h
+++ b/widget/cocoa/nsNativeThemeCocoa.h
@@ -16,7 +16,10 @@
 
 @class MOZCellDrawWindow;
 @class MOZCellDrawView;
+@class MOZSearchFieldCell;
+@class NSProgressBarCell;
 class nsDeviceContext;
+struct SegmentedControlRenderSettings;
 
 namespace mozilla {
 namespace gfx {
@@ -28,6 +31,13 @@ class nsNativeThemeCocoa : public mozilla::widget::ThemeCocoa {
   using ThemeCocoa = mozilla::widget::ThemeCocoa;
 
  public:
+  enum class MenuIcon : uint8_t {
+    eCheckmark,
+    eMenuArrow,
+    eMenuDownScrollArrow,
+    eMenuUpScrollArrow
+  };
+
   enum class CheckboxOrRadioState : uint8_t { eOff, eOn, eIndeterminate };
 
   enum class ButtonType : uint8_t {
@@ -40,6 +50,10 @@ class nsNativeThemeCocoa : public mozilla::widget::ThemeCocoa {
     eDisclosureButtonOpen
   };
 
+  enum class SpinButton : uint8_t { eUp, eDown };
+
+  enum class SegmentType : uint8_t { eToolbarButton };
+
   enum class OptimumState : uint8_t { eOptimum, eSubOptimum, eSubSubOptimum };
 
   struct ControlParams {
@@ -56,6 +70,27 @@ class nsNativeThemeCocoa : public mozilla::widget::ThemeCocoa {
     bool focused : 1;
     bool rtl : 1;
   };
+  
+  struct MenuBackgroundParams {
+    bool disabled = false;
+    bool submenuRightOfParent = false;
+  };
+  
+  struct MenuIconParams {
+    MenuIcon icon = MenuIcon::eCheckmark;
+    bool disabled = false;
+    bool insideActiveMenuItem = false;
+    bool centerHorizontally = false;
+    bool rtl = false;
+  };
+
+  struct MenuItemParams {
+    bool backgroundIsVibrant = false;
+    bool checked = false;
+    bool disabled = false;
+    bool selected = false;
+    bool rtl = false;
+  };
 
   struct CheckboxOrRadioParams {
     ControlParams controlParams;
@@ -74,6 +109,30 @@ class nsNativeThemeCocoa : public mozilla::widget::ThemeCocoa {
     bool editable = false;
   };
 
+  struct SpinButtonParams {
+    mozilla::Maybe<SpinButton> pressedButton;
+    bool disabled = false;
+    bool insideActiveWindow = false;
+  };
+
+  struct SegmentParams {
+    SegmentType segmentType = SegmentType::eToolbarButton;
+    bool insideActiveWindow = false;
+    bool pressed = false;
+    bool selected = false;
+    bool focused = false;
+    bool atLeftEnd = false;
+    bool atRightEnd = false;
+    bool drawsLeftSeparator = false;
+    bool drawsRightSeparator = false;
+    bool rtl = false;
+  };
+
+  struct UnifiedToolbarParams {
+    float unifiedHeight = 0.0f;
+    bool isMain = false;
+  };
+
   struct TextFieldParams {
     float verticalAlignFactor = 0.5f;
     bool insideToolbar = false;
@@ -82,16 +141,87 @@ class nsNativeThemeCocoa : public mozilla::widget::ThemeCocoa {
     bool rtl = false;
   };
 
+  struct ProgressParams {
+    double value = 0.0;
+    double max = 0.0;
+    float verticalAlignFactor = 0.5f;
+    bool insideActiveWindow = false;
+    bool indeterminate = false;
+    bool horizontal = false;
+    bool rtl = false;
+  };
+
+  struct MeterParams {
+    double value = 0;
+    double min = 0;
+    double max = 0;
+    OptimumState optimumState = OptimumState::eOptimum;
+    float verticalAlignFactor = 0.5f;
+    bool horizontal = true;
+    bool rtl = false;
+  };
+
+  struct ScaleParams {
+    int32_t value = 0;
+    int32_t min = 0;
+    int32_t max = 0;
+    bool insideActiveWindow = false;
+    bool disabled = false;
+    bool focused = false;
+    bool horizontal = true;
+    bool reverse = false;
+  };
+
   enum Widget : uint8_t {
-    eCheckbox,            // CheckboxOrRadioParams
-    eRadio,               // CheckboxOrRadioParams
-    eButton,              // ButtonParams
-    eDropdown,            // DropdownParams
+    eColorFill,       // mozilla::gfx::sRGBColor
+    eMenuBackground,  // MenuBackgroundParams
+    eMenuIcon,        // MenuIconParams
+    eMenuItem,        // MenuItemParams
+    eMenuSeparator,   // MenuItemParams
+    eTooltip,
+    eCheckbox,        // CheckboxOrRadioParams
+    eRadio,           // CheckboxOrRadioParams
+    eButton,          // ButtonParams
+    eDropdown,        // DropdownParams
+    eFocusOutline,
+    eSpinButtons,     // SpinButtonParams
+    eSpinButtonUp,    // SpinButtonParams
+    eSpinButtonDown,  // SpinButtonParams
+    eSegment,         // SegmentParams
+    eSeparator,
+    eNativeTitlebar,  // UnifiedToolbarParams
+    eStatusBar,  // bool
+    eGroupBox,
     eTextField,           // TextFieldParams
+    eSearchField,         // TextFieldParams
+    eProgressBar,         // ProgressParams
+    eMeter,               // MeterParams
+    eScale,               // ScaleParams
     eMultilineTextField,  // bool
+    eListBox,
+    eSourceList,                   // bool
+    eActiveSourceListSelection,    // bool
+    eInactiveSourceListSelection,  // bool
   };
 
   struct WidgetInfo {
+    static WidgetInfo ColorFill(const mozilla::gfx::sRGBColor& aParams) {
+      return WidgetInfo(Widget::eColorFill, aParams);
+    }
+    static WidgetInfo MenuBackground(const MenuBackgroundParams& aParams) {
+      return WidgetInfo(Widget::eMenuBackground, aParams);
+    }
+    static WidgetInfo MenuIcon(const MenuIconParams& aParams) {
+      return WidgetInfo(Widget::eMenuIcon, aParams);
+    }
+    static WidgetInfo MenuItem(const MenuItemParams& aParams) {
+      return WidgetInfo(Widget::eMenuItem, aParams);
+    }
+    static WidgetInfo MenuSeparator(const MenuItemParams& aParams) {
+      return WidgetInfo(Widget::eMenuSeparator, aParams);
+    }
+    static WidgetInfo Tooltip() { return WidgetInfo(Widget::eTooltip, false); }
+
     static WidgetInfo Checkbox(const CheckboxOrRadioParams& aParams) {
       return WidgetInfo(Widget::eCheckbox, aParams);
     }
@@ -104,12 +234,57 @@ class nsNativeThemeCocoa : public mozilla::widget::ThemeCocoa {
     static WidgetInfo Dropdown(const DropdownParams& aParams) {
       return WidgetInfo(Widget::eDropdown, aParams);
     }
+    static WidgetInfo FocusOutline() { return WidgetInfo(Widget::eFocusOutline, false); }
+    static WidgetInfo SpinButtons(const SpinButtonParams& aParams) {
+      return WidgetInfo(Widget::eSpinButtons, aParams);
+    }
+    static WidgetInfo SpinButtonUp(const SpinButtonParams& aParams) {
+      return WidgetInfo(Widget::eSpinButtonUp, aParams);
+    }
+    static WidgetInfo SpinButtonDown(const SpinButtonParams& aParams) {
+      return WidgetInfo(Widget::eSpinButtonDown, aParams);
+    }
+    static WidgetInfo Segment(const SegmentParams& aParams) {
+      return WidgetInfo(Widget::eSegment, aParams);
+    }
+    static WidgetInfo Separator() {
+      return WidgetInfo(Widget::eSeparator, false);
+    }
+    static WidgetInfo NativeTitlebar(const UnifiedToolbarParams& aParams) {
+      return WidgetInfo(Widget::eNativeTitlebar, aParams);
+    }  
+    static WidgetInfo StatusBar(bool aParams) {
+      return WidgetInfo(Widget::eStatusBar, aParams);
+    }
+    static WidgetInfo GroupBox() {
+      return WidgetInfo(Widget::eGroupBox, false);
+    }
     static WidgetInfo TextField(const TextFieldParams& aParams) {
       return WidgetInfo(Widget::eTextField, aParams);
     }
+    static WidgetInfo SearchField(const TextFieldParams& aParams) {
+      return WidgetInfo(Widget::eSearchField, aParams);
+    }
+    static WidgetInfo ProgressBar(const ProgressParams& aParams) {
+      return WidgetInfo(Widget::eProgressBar, aParams);
+    }
+    static WidgetInfo Meter(const MeterParams& aParams) {
+      return WidgetInfo(Widget::eMeter, aParams);
+    }
+    static WidgetInfo Scale(const ScaleParams& aParams) {
+      return WidgetInfo(Widget::eScale, aParams);
+    }
     static WidgetInfo MultilineTextField(bool aParams) {
       return WidgetInfo(Widget::eMultilineTextField, aParams);
     }
+    static WidgetInfo ListBox() { return WidgetInfo(Widget::eListBox, false); }
+    static WidgetInfo SourceList(bool aParams) { return WidgetInfo(Widget::eSourceList, aParams); }
+    static WidgetInfo ActiveSourceListSelection(bool aParams) {
+      return WidgetInfo(Widget::eActiveSourceListSelection, aParams);
+    }
+    static WidgetInfo InactiveSourceListSelection(bool aParams) {
+      return WidgetInfo(Widget::eInactiveSourceListSelection, aParams);
+    }
 
     template <typename T>
     T Params() const {
@@ -124,8 +299,10 @@ class nsNativeThemeCocoa : public mozilla::widget::ThemeCocoa {
     WidgetInfo(enum Widget aWidget, const T& aParams)
         : mVariant(aParams), mWidget(aWidget) {}
 
-    mozilla::Variant<mozilla::gfx::sRGBColor, CheckboxOrRadioParams,
-                     ButtonParams, DropdownParams, TextFieldParams, bool>
+    mozilla::Variant<mozilla::gfx::sRGBColor, MenuBackgroundParams, MenuIconParams, MenuItemParams,
+                     CheckboxOrRadioParams, ButtonParams, DropdownParams, SpinButtonParams,
+                     SegmentParams, UnifiedToolbarParams, TextFieldParams, ProgressParams,
+                     MeterParams, ScaleParams, bool>
         mVariant;
 
     enum Widget mWidget;
@@ -167,41 +344,96 @@ class nsNativeThemeCocoa : public mozilla::widget::ThemeCocoa {
   Transparency GetWidgetTransparency(nsIFrame*, StyleAppearance) override;
   mozilla::Maybe<WidgetInfo> ComputeWidgetInfo(nsIFrame*, StyleAppearance,
                                                const nsRect& aRect);
-
+  void DrawProgress(CGContextRef context, const HIRect& inBoxRect,
+                    const ProgressParams& aParams);
+  static void DrawNativeTitlebar(CGContextRef aContext, CGRect aTitlebarRect,
+                                 CGFloat aUnifiedHeight, BOOL aIsMain, BOOL aIsFlipped);
  protected:
   virtual ~nsNativeThemeCocoa();
 
   LayoutDeviceIntMargin DirectionAwareMargin(const LayoutDeviceIntMargin&,
                                              nsIFrame*);
+  nsIFrame* SeparatorResponsibility(nsIFrame* aBefore, nsIFrame* aAfter);
   ControlParams ComputeControlParams(nsIFrame*, mozilla::dom::ElementState);
+  MenuBackgroundParams ComputeMenuBackgroundParams(nsIFrame* aFrame,
+                                                   mozilla::dom::ElementState);
+  MenuIconParams ComputeMenuIconParams(nsIFrame* aParams, mozilla::dom::ElementState,
+                                       MenuIcon aIcon);
+  MenuItemParams ComputeMenuItemParams(nsIFrame* aFrame, mozilla::dom::ElementState,
+                                       bool aIsChecked);
+  SegmentParams ComputeSegmentParams(nsIFrame*, mozilla::dom::ElementState,
+                                     SegmentType);
   TextFieldParams ComputeTextFieldParams(nsIFrame*, mozilla::dom::ElementState);
+  ProgressParams ComputeProgressParams(nsIFrame*, mozilla::dom::ElementState,
+                                       bool aIsHorizontal);
+  MeterParams ComputeMeterParams(nsIFrame*);
+  mozilla::Maybe<ScaleParams> ComputeHTMLScaleParams(
+      nsIFrame*, mozilla::dom::ElementState);
 
   // HITheme drawing routines
+  void DrawMeter(CGContextRef context, const HIRect& inBoxRect,
+                 const MeterParams& aParams);
+  void DrawSegment(CGContextRef cgContext, const HIRect& inBoxRect,
+                   const SegmentParams& aParams);
+  void DrawSegmentBackground(CGContextRef cgContext, const HIRect& inBoxRect,
+                             const SegmentParams& aParams);
+  void DrawTabPanel(CGContextRef context, const HIRect& inBoxRect,
+                    bool aIsInsideActiveWindow);
+  void DrawScale(CGContextRef context, const HIRect& inBoxRect,
+                 const ScaleParams& aParams);
   void DrawCheckboxOrRadio(CGContextRef cgContext, bool inCheckbox,
-                           const NSRect& inBoxRect,
+                           const HIRect& inBoxRect,
                            const CheckboxOrRadioParams& aParams);
-  void DrawTextField(CGContextRef cgContext, const NSRect& inBoxRect,
+  void DrawSearchField(CGContextRef cgContext, const HIRect& inBoxRect,
+                       const TextFieldParams& aParams);
+  void DrawTextField(CGContextRef cgContext, const HIRect& inBoxRect,
                      const TextFieldParams& aParams);
-  void DrawPushButton(CGContextRef cgContext, const NSRect& inBoxRect,
+  void DrawPushButton(CGContextRef cgContext, const HIRect& inBoxRect,
                       ButtonType aButtonType, ControlParams aControlParams);
   void DrawSquareBezelPushButton(CGContextRef cgContext,
-                                 const NSRect& inBoxRect,
+                                 const HIRect& inBoxRect,
                                  ControlParams aControlParams);
-  void DrawHelpButton(CGContextRef cgContext, const NSRect& inBoxRect,
+  void DrawHelpButton(CGContextRef cgContext, const HIRect& inBoxRect,
                       ControlParams aControlParams);
-  void DrawDisclosureButton(CGContextRef cgContext, const NSRect& inBoxRect,
+  void DrawDisclosureButton(CGContextRef cgContext, const HIRect& inBoxRect,
                             ControlParams aControlParams,
-                            NSControlStateValue aState);
-  void DrawHIThemeButton(CGContextRef cgContext, const NSRect& aRect,
+                            NSCellStateValue aState);
+  void DrawMenuBackground(CGContextRef cgContext, const CGRect& inBoxRect,
+                          const MenuBackgroundParams& aParams);
+  NSString* GetMenuIconName(const MenuIconParams& aParams);
+  NSSize GetMenuIconSize(MenuIcon aIcon);
+  void DrawMenuIcon(CGContextRef cgContext, const CGRect& aRect, const MenuIconParams& aParams);
+  void DrawMenuItem(CGContextRef cgContext, const CGRect& inBoxRect, const MenuItemParams& aParams);
+  void DrawMenuSeparator(CGContextRef cgContext, const CGRect& inBoxRect,
+                         const MenuItemParams& aParams);
+  void DrawHIThemeButton(CGContextRef cgContext, const HIRect& aRect,
                          ThemeButtonKind aKind, ThemeButtonValue aValue,
                          ThemeDrawState aState, ThemeButtonAdornment aAdornment,
                          const ControlParams& aParams);
-  void DrawButton(CGContextRef context, const NSRect& inBoxRect,
+  void DrawButton(CGContextRef context, const HIRect& inBoxRect,
                   const ButtonParams& aParams);
-  void DrawDropdown(CGContextRef context, const NSRect& inBoxRect,
+  void DrawFocusOutline(CGContextRef cgContext, const HIRect& inBoxRect);
+  void DrawDropdown(CGContextRef context, const HIRect& inBoxRect,
                     const DropdownParams& aParams);
+  HIThemeButtonDrawInfo SpinButtonDrawInfo(ThemeButtonKind aKind,
+                                           const SpinButtonParams& aParams);
+  void DrawSpinButtons(CGContextRef context, const HIRect& inBoxRect,
+                       const SpinButtonParams& aParams);
+  void DrawSpinButton(CGContextRef context, const HIRect& inBoxRect,
+                      SpinButton aDrawnButton, const SpinButtonParams& aParams);
+  void DrawToolbar(CGContextRef cgContext, const CGRect& inBoxRect,
+                   bool aIsMain);
+  void DrawUnifiedToolbar(CGContextRef cgContext, const HIRect& inBoxRect,
+                          const UnifiedToolbarParams& aParams);
+  void DrawNativeTitlebar(CGContextRef aContext, CGRect aTitlebarRect,
+                          const UnifiedToolbarParams& aParams);
+  void DrawStatusBar(CGContextRef cgContext, const HIRect& inBoxRect,
+                     bool aIsMain);
   void DrawMultilineTextField(CGContextRef cgContext, const CGRect& inBoxRect,
                               bool aIsFocused);
+  void DrawSourceList(CGContextRef cgContext, const CGRect& inBoxRect, bool aIsActive);
+  void DrawSourceListSelection(CGContextRef aContext, const CGRect& aRect, bool aWindowIsActive,
+                               bool aSelectionIsActive);
   void RenderWidget(const WidgetInfo& aWidgetInfo, mozilla::ColorScheme,
                     mozilla::gfx::DrawTarget& aDrawTarget,
                     const mozilla::gfx::Rect& aWidgetRect,
@@ -214,8 +446,11 @@ class nsNativeThemeCocoa : public mozilla::widget::ThemeCocoa {
   NSButtonCell* mRadioButtonCell;
   NSButtonCell* mCheckboxCell;
   NSTextFieldCell* mTextFieldCell;
+  MOZSearchFieldCell* mSearchFieldCell;
   NSPopUpButtonCell* mDropdownCell;
   NSComboBoxCell* mComboBoxCell;
+  NSProgressBarCell* mProgressBarCell;
+  NSLevelIndicatorCell* mMeterBarCell;
   MOZCellDrawWindow* mCellDrawWindow = nil;
   MOZCellDrawView* mCellDrawView;
 };
