diff --git a/ipc/chromium/src/chrome/common/ipc_channel_mach.cc b/ipc/chromium/src/chrome/common/ipc_channel_mach.cc
index 4670299f4d1f3..2e8f833d97143 100644
--- a/ipc/chromium/src/chrome/common/ipc_channel_mach.cc
+++ b/ipc/chromium/src/chrome/common/ipc_channel_mach.cc
@@ -424,7 +424,7 @@ bool ChannelMach::ProcessIncomingMessage() {
     }
 #ifdef XP_MACOSX
     if (XRE_IsParentProcess() &&
-        audit_token_to_pid(trailer->msgh_audit) != other_pid) {
+        ((pid_t) trailer->msgh_audit.val[5]) != other_pid) {
       CHROMIUM_LOG(ERROR) << "audit token does not correspond to given pid";
       return false;
     }
