diff --git a/media/libvpx/libvpx/vpx_ports/vpx_timer.h b/media/libvpx/libvpx/vpx_ports/vpx_timer.h
index 55e7891e19f98..cc67527afac30 100644
--- a/media/libvpx/libvpx/vpx_ports/vpx_timer.h
+++ b/media/libvpx/libvpx/vpx_ports/vpx_timer.h
@@ -55,24 +55,92 @@ struct vpx_usec_timer {
 #endif
 };
 
+#if defined(__APPLE__) && defined(__MACH__)
+#if !defined(MAC_OS_VERSION_10_12) || MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_VERSION_10_12
+#  include <time.h>
+#  include <sys/time.h>
+#  include <sys/types.h>
+#  include <mach/mach_time.h>
+
+const uint64_t kNSperMS = 1000000;
+
+#  include <mach/mach.h>
+#  include <mach/clock.h>
+    //taken from https://github.com/ChisholmKyle/PosixMachTiming/
+    /* timing struct for osx */
+    typedef struct RoTimingMach {
+        mach_timebase_info_data_t timebase;
+        clock_serv_t cclock;
+    } RoTimingMach;
+
+    /* internal timing struct for osx */
+    static RoTimingMach ro_timing_mach_g;
+
+    /* mach clock port */
+    static mach_port_t clock_port;
+
+    /* emulate posix clock_gettime */
+    static inline int clock_gettime_missing (clockid_t id, struct timespec *tspec)
+    {
+        int retval = -1;
+        mach_timespec_t mts;
+        if (id == CLOCK_REALTIME) {
+            retval = clock_get_time (ro_timing_mach_g.cclock, &mts);
+            if (retval == 0 && tspec != NULL) {
+                tspec->tv_sec = mts.tv_sec;
+                tspec->tv_nsec = mts.tv_nsec;
+            }
+        } else if (id == CLOCK_MONOTONIC) {
+            retval = clock_get_time (clock_port, &mts);
+            if (retval == 0 && tspec != NULL) {
+                tspec->tv_sec = mts.tv_sec;
+                tspec->tv_nsec = mts.tv_nsec;
+            }
+        } else {}
+        return retval;
+    }
+
+
+#endif
+#endif
 static INLINE void vpx_usec_timer_start(struct vpx_usec_timer *t) {
 #if defined(_WIN32)
   QueryPerformanceCounter(&t->begin);
-#elif defined(CLOCK_MONOTONIC_RAW)
+#endif
+#ifdef XP_MACOSX
+  if(__builtin_available(macOS 10.12, *)) {
+#endif
+#if defined(CLOCK_MONOTONIC_RAW)
   clock_gettime(CLOCK_MONOTONIC_RAW, &t->begin);
-#else
+#elif !defined(_WIN32) // need this to avoid compiling for WIN32 
+                       // because we modified the original macro
   clock_gettime(CLOCK_MONOTONIC, &t->begin);
 #endif
+#ifdef XP_MACOSX
+  } else {
+    clock_gettime_missing(CLOCK_MONOTONIC, &t->begin);
+  }
+#endif
 }
 
 static INLINE void vpx_usec_timer_mark(struct vpx_usec_timer *t) {
 #if defined(_WIN32)
   QueryPerformanceCounter(&t->end);
-#elif defined(CLOCK_MONOTONIC_RAW)
+#endif
+#ifdef XP_MACOSX
+  if(__builtin_available(macOS 10.12, *)) {
+#endif
+#if defined(CLOCK_MONOTONIC_RAW)
   clock_gettime(CLOCK_MONOTONIC_RAW, &t->end);
-#else
+#elif !defined(_WIN32) // need this to avoid compiling for WIN32 
+                       // because we modified the original macro
   clock_gettime(CLOCK_MONOTONIC, &t->end);
 #endif
+#ifdef XP_MACOSX
+  } else {
+    clock_gettime_missing(CLOCK_MONOTONIC, &t->end);
+  }
+#endif
 }
 
 static INLINE int64_t vpx_usec_timer_elapsed(struct vpx_usec_timer *t) {
