diff --git a/python/mach/mach/telemetry.py b/python/mach/mach/telemetry.py
index bdc3b958b339b..ce69a35b556d1 100644
--- a/python/mach/mach/telemetry.py
+++ b/python/mach/mach/telemetry.py
@@ -92,10 +92,10 @@ def is_applicable_telemetry_environment():
 
 
 def is_telemetry_enabled(settings):
-    if os.environ.get("DISABLE_TELEMETRY") == "1":
-        return False
+    # if os.environ.get("DISABLE_TELEMETRY") == "1":
+    return False
 
-    return settings.mach_telemetry.is_enabled
+    # return settings.mach_telemetry.is_enabled
 
 
 def arcrc_path():
