diff --git a/third_party/rust/cubeb-coreaudio/src/backend/tests/interfaces.rs b/third_party/rust/cubeb-coreaudio/src/backend/tests/interfaces.rs
index 73e249973495e..f3587bb2ee483 100644
--- a/third_party/rust/cubeb-coreaudio/src/backend/tests/interfaces.rs
+++ b/third_party/rust/cubeb-coreaudio/src/backend/tests/interfaces.rs
@@ -1314,7 +1314,8 @@ fn test_ops_input_voice_stream_init_and_destroy() {
         let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
         assert_eq!(
             stm.core_stream_data.using_voice_processing_unit(),
-            macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+            (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
         );
     });
 }
@@ -1326,7 +1327,8 @@ fn test_ops_input_voice_stream_start() {
         let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
         assert_eq!(
             stm.core_stream_data.using_voice_processing_unit(),
-            macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+            (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
         );
     });
 }
@@ -1338,7 +1340,8 @@ fn test_ops_input_voice_stream_stop() {
         let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
         assert_eq!(
             stm.core_stream_data.using_voice_processing_unit(),
-            macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+            (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
         );
     });
 }
@@ -1349,7 +1352,8 @@ fn test_ops_duplex_voice_stream_init_and_destroy() {
         let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
         assert_eq!(
             stm.core_stream_data.using_voice_processing_unit(),
-            macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+            (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
         );
     });
 }
@@ -1361,7 +1365,8 @@ fn test_ops_duplex_voice_stream_start() {
         let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
         assert_eq!(
             stm.core_stream_data.using_voice_processing_unit(),
-            macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+            (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
         );
     });
 }
@@ -1373,7 +1378,8 @@ fn test_ops_duplex_voice_stream_stop() {
         let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
         assert_eq!(
             stm.core_stream_data.using_voice_processing_unit(),
-            macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+            (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
         );
     });
 }
@@ -1388,7 +1394,8 @@ fn test_ops_duplex_voice_stream_drain() {
             let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
             assert_eq!(
                 stm.core_stream_data.using_voice_processing_unit(),
-                macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+                (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
             );
             thread::sleep(Duration::from_millis(10));
         },
@@ -1398,7 +1405,8 @@ fn test_ops_duplex_voice_stream_drain() {
 #[test]
 #[ignore]
 fn test_ops_timing_sensitive_multiple_voice_stream_init_and_destroy() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1551,7 +1559,8 @@ fn test_ops_timing_sensitive_multiple_voice_stream_init_and_destroy() {
 #[test]
 #[ignore]
 fn test_ops_timing_sensitive_multiple_duplex_voice_stream_start() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1592,7 +1601,8 @@ fn test_ops_timing_sensitive_multiple_duplex_voice_stream_start() {
 #[test]
 #[ignore]
 fn test_ops_timing_sensitive_multiple_duplex_voice_stream_params() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1679,7 +1689,8 @@ fn test_ops_timing_sensitive_multiple_duplex_voice_stream_params() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_mute() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1695,7 +1706,8 @@ fn test_ops_duplex_voice_stream_set_input_mute() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_mute_before_start() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1715,7 +1727,8 @@ fn test_ops_duplex_voice_stream_set_input_mute_before_start() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_mute_before_start_with_reinit() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1754,7 +1767,8 @@ fn test_ops_duplex_voice_stream_set_input_mute_before_start_with_reinit() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_mute_after_start() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1771,7 +1785,8 @@ fn test_ops_duplex_voice_stream_set_input_mute_after_start() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_processing_params() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1791,7 +1806,8 @@ fn test_ops_duplex_voice_stream_set_input_processing_params() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_processing_params_before_start() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1815,7 +1831,8 @@ fn test_ops_duplex_voice_stream_set_input_processing_params_before_start() {
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_processing_params_before_start_with_reinit() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1879,7 +1896,8 @@ fn test_ops_duplex_voice_stream_set_input_processing_params_before_start_with_re
 
 #[test]
 fn test_ops_duplex_voice_stream_set_input_processing_params_after_start() {
-    if macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY {
+    if (macos_kernel_major_version().unwrap() == MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA) {
         // We disable VPIO on Monterey.
         return;
     }
@@ -1909,7 +1927,8 @@ fn test_ops_stereo_input_duplex_voice_stream_init_and_destroy() {
             let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
             assert_eq!(
                 stm.core_stream_data.using_voice_processing_unit(),
-                macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+                (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
             );
         },
     );
@@ -1923,7 +1942,8 @@ fn test_ops_stereo_input_duplex_voice_stream_start() {
             let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
             assert_eq!(
                 stm.core_stream_data.using_voice_processing_unit(),
-                macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+                (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
             );
             assert_eq!(unsafe { OPS.stream_start.unwrap()(stream) }, ffi::CUBEB_OK);
         },
@@ -1938,7 +1958,8 @@ fn test_ops_stereo_input_duplex_voice_stream_stop() {
             let stm = unsafe { &mut *(stream as *mut AudioUnitStream) };
             assert_eq!(
                 stm.core_stream_data.using_voice_processing_unit(),
-                macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY
+                (macos_kernel_major_version().unwrap() != MACOS_KERNEL_MAJOR_VERSION_MONTEREY) &&
+		(macos_kernel_major_version().unwrap() >= MACOS_KERNEL_MAJOR_VERSION_SIERRA)
             );
             assert_eq!(unsafe { OPS.stream_stop.unwrap()(stream) }, ffi::CUBEB_OK);
         },
