diff --git a/toolkit/library/moz.build b/toolkit/library/moz.build
index 2853f66c45a21..a3b9446f73588 100644
--- a/toolkit/library/moz.build
+++ b/toolkit/library/moz.build
@@ -89,7 +89,16 @@ def Libxul(name, output_category=None):
 
     OS_LIBS += CONFIG["LIBATOMIC_LIBS"]
 
+    # TouchBar-related classes are only available in the 10.12.2 SDK and later.
+    # We need to weak link these classes until we've upgraded our SDK to at
+    # least 10.12.2.
+    # We're also linking against libresolv to solve bug 1367932.
     if CONFIG["OS_ARCH"] == "Darwin":
+        LDFLAGS += ["-Wl,-U,_OBJC_CLASS_$_NSTouchBar"]
+        LDFLAGS += ["-Wl,-U,_OBJC_CLASS_$_NSSharingServicePickerTouchBarItem"]
+        LDFLAGS += ["-Wl,-U,_OBJC_METACLASS_$_NSTouchBar"]
+        LDFLAGS += ["-Wl,-U,_OBJC_CLASS_$_NSCustomTouchBarItem"]
+        LDFLAGS += ["-Wl,-U,_OBJC_CLASS_$_NSPopoverTouchBarItem"]
         LDFLAGS += ["-lresolv"]
         LDFLAGS += ["-Wl,-rpath,@executable_path/../Frameworks/ChannelPrefs.framework"]
 
@@ -202,36 +211,35 @@ if CONFIG["MOZ_WEBRTC"]:
             "wininet",
         ]
 
-if CONFIG["OS_ARCH"] == "Darwin":
-    OS_LIBS += [
-        "-framework AudioToolbox",
-        "-framework AVFoundation",
-        "-framework CoreFoundation",
-        "-framework CoreMedia",
-        "-framework CoreVideo",
-        "-framework Foundation",
-        "-framework MediaPlayer",
-        "-framework Metal",
-        "-framework QuartzCore",
-        "-framework Security",
-    ]
-
 if CONFIG["MOZ_WIDGET_TOOLKIT"] == "cocoa":
     OS_LIBS += [
-        "-framework AddressBook",
+        "-framework Accessibility",
         "-framework AppKit",
-        "-framework ApplicationServices",
-        "-framework AudioUnit",
+        "-framework Foundation",
+        "-framework CoreFoundation",
+        "-framework CoreLocation",
+        "-framework QuartzCore",
         "-framework Carbon",
         "-framework CoreAudio",
-        "-framework CoreLocation",
-        "-framework CoreMIDI",
-        "-framework CoreServices",
-        "-framework IOKit",
+        "-framework CoreVideo",
+        "-framework AudioToolbox",
+        "-framework AudioUnit",
+        "-framework AddressBook",
         "-framework OpenGL",
+        "-framework Security",
         "-framework ServiceManagement",
+        "-framework CoreServices",
+        "-framework ApplicationServices",
+        "-framework AppKit",
+        "-framework CoreMIDI",
         "-framework SystemConfiguration",
+        "-framework AVFoundation",
+        "-framework CoreMedia",
+        "-framework IOKit",
         "-weak_framework ScreenCaptureKit",
+        "-F%s" % CONFIG["MACOS_PRIVATE_FRAMEWORKS_DIR"],
+        "-framework CoreUI",
+        "-framework CoreSymbolication",
         "cups",
     ]
     USE_LIBS += [
@@ -240,9 +248,17 @@ if CONFIG["MOZ_WIDGET_TOOLKIT"] == "cocoa":
 
 if CONFIG["MOZ_WIDGET_TOOLKIT"] == "uikit":
     OS_LIBS += [
+        "-framework Foundation",
+        "-framework CoreFoundation",
         "-framework CoreGraphics",
         "-framework CoreText",
+        "-framework AVFoundation",
+        "-framework AudioToolbox",
+        "-framework CoreMedia",
+        "-framework CoreVideo",
         "-framework OpenGLES",
+        "-framework QuartzCore",
+        "-framework Security",
     ]
 
 if CONFIG["MOZ_WMF"]:
@@ -333,6 +349,14 @@ if CONFIG["OS_ARCH"] == "FreeBSD":
         "util",
     ]
 
+if CONFIG["OS_ARCH"] == "Darwin":
+    OS_LIBS += [
+        # Link to Metal as required by the Metal gfx-hal backend
+        "-weak_framework Metal",
+        "-weak_framework MediaPlayer",
+    ]
+
+
 if CONFIG["OS_ARCH"] == "WINNT":
     OS_LIBS += [
         "shell32",
