diff --git a/widget/cocoa/nsMacSharingService.mm b/widget/cocoa/nsMacSharingService.mm
index c2e432e297541..e72fc279c8198 100644
--- a/widget/cocoa/nsMacSharingService.mm
+++ b/widget/cocoa/nsMacSharingService.mm
@@ -12,6 +12,7 @@
 #include "js/PropertyAndElement.h"  // JS_SetElement, JS_SetProperty
 #include "nsCocoaUtils.h"
 #include "mozilla/MacStringHelpers.h"
+#include "SDKDeclarations.h"
 
 NS_IMPL_ISUPPORTS(nsMacSharingService, nsIMacSharingService)
 
@@ -30,6 +31,13 @@
 NSString* const openSharingSubpaneProtocolKey = @"protocol";
 NSString* const openSharingSubpaneProtocolValue = @"com.apple.share-services";
 
+
+#if !defined(MAC_OS_X_VERSION_10_10) || \
+    MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_10
+NSString* const NSUserActivityTypeBrowsingWeb =
+    @"NSUserActivityTypeBrowsingWeb";
+#endif  // MAC_OS_X_VERSION_10_10
+
 // Expose the id so we can pass reference through to JS and back
 @interface NSSharingService (ExposeName)
 - (id)name;
@@ -114,11 +122,11 @@ static void SetStrAttribute(JSContext* aCx, JS::Rooted<JSObject*>& aObj,
   NS_OBJC_BEGIN_TRY_BLOCK_RETURN;
 
   NSURL* url = nsCocoaUtils::ToNSURL(aPageUrl);
-  if (!url) {
+  if (!url || !nsCocoaFeatures::OnMountainLionOrLater()) {
     // aPageUrl is not a valid URL.
     return NS_ERROR_FAILURE;
   }
-
+  
   NSArray* sharingService = [NSSharingService sharingServicesForItems:@[ url ]];
   int32_t serviceCount = 0;
   JS::Rooted<JSObject*> array(aCx, JS::NewArrayObject(aCx, 0));
@@ -179,43 +187,53 @@ static void SetStrAttribute(JSContext* aCx, JS::Rooted<JSObject*>& aObj,
                               const nsAString& aPageUrl,
                               const nsAString& aPageTitle) {
   NS_OBJC_BEGIN_TRY_BLOCK_RETURN;
-
+  
   NSString* serviceName = nsCocoaUtils::ToNSString(aServiceName);
-  NSSharingService* service =
-      [NSSharingService sharingServiceNamed:serviceName];
-  if (!service) {
-    return NS_ERROR_FAILURE;
-  }
-
-  NSString* pageTitle = nsCocoaUtils::ToNSString(aPageTitle);
-  [service setSubject:pageTitle];
+  if(nsCocoaFeatures::OnMountainLionOrLater()) {
+    NSSharingService* service =
+        [NSSharingService sharingServiceNamed:serviceName];
+    if (!service) {
+      return NS_ERROR_FAILURE;
+    }
 
-  NSURL* pageUrl = nsCocoaUtils::ToNSURL(aPageUrl);
-  if (!pageUrl) {
-    return NS_ERROR_FAILURE;
-  }
+    NSString* pageTitle = nsCocoaUtils::ToNSString(aPageTitle);  
+    [service setSubject:pageTitle];
 
-  // Reminders fetch data from an activity, not the share data
-  if ([serviceName isEqual:oldRemindersServiceName] ||
-      [serviceName isEqual:newRemindersServiceName]) {
-    NSUserActivity* shareActivity = [[[NSUserActivity alloc]
-        initWithActivityType:NSUserActivityTypeBrowsingWeb] autorelease];
+    NSURL* pageUrl = nsCocoaUtils::ToNSURL(aPageUrl);
+    if (!pageUrl) {
+      return NS_ERROR_FAILURE;
+    }
 
-    if ([pageUrl.scheme hasPrefix:@"http"]) {
-      [shareActivity setWebpageURL:pageUrl];
+    // Reminders fetch data from an activity, not the share data
+    if(nsCocoaFeatures::OnYosemiteOrLater()) {
+      if ([serviceName isEqual:oldRemindersServiceName] ||
+          [serviceName isEqual:newRemindersServiceName]) {
+        NSUserActivity* shareActivity = [[[NSUserActivity alloc]
+            initWithActivityType:NSUserActivityTypeBrowsingWeb] autorelease];
+
+        if ([pageUrl.scheme hasPrefix:@"http"]) {
+          [shareActivity setWebpageURL:pageUrl];
+        }
+
+        [shareActivity setEligibleForHandoff:NO];
+        [shareActivity setTitle:pageTitle];
+        [shareActivity becomeCurrent];
+
+        SharingServiceDelegate* shareDelegate =
+            [[SharingServiceDelegate alloc] initWithActivity:shareActivity];
+        [service setDelegate:shareDelegate];  // weak reference
+      }
     }
 
-    [shareActivity setEligibleForHandoff:NO];
-    [shareActivity setTitle:pageTitle];
-    [shareActivity becomeCurrent];
+    // Twitter likes the the title as an additional share item
+    NSArray* toShare = [[service name] isEqual:NSSharingServiceNamePostOnTwitter]
+                          ? @[ pageUrl, pageTitle ]
+                          : @[ pageUrl ];
 
-    SharingServiceDelegate* shareDelegate =
-        [[SharingServiceDelegate alloc] initWithActivity:shareActivity];
-    [service setDelegate:shareDelegate];  // weak reference
+    [service setSubject:pageTitle];
+    [service performWithItems:toShare];
   }
 
-  [service performWithItems:@[ pageUrl ]];
-
   return NS_OK;
 
   NS_OBJC_END_TRY_BLOCK_RETURN(NS_ERROR_FAILURE);
